// @refinedev/antd/dist/index.mjs
"use client"
var __defProp = Object.defineProperty;
var __name = (target, value) => __defProp(target, "name", { value, configurable: true });

// src/hooks/form/useForm.ts
import React from "react";
import {
  Form
} from "antd";
import { useForm as useFormSF } from "sunflower-antd";
import {
  flattenObjectKeys,
  propertyPathToArray,
  useForm as useFormCore,
  useWarnAboutChange,
  useTranslate,
  useRefineContext
} from "@refinedev/core";
var useForm = /* @__PURE__ */ __name(({
  action,
  resource,
  onMutationSuccess: onMutationSuccessProp,
  onMutationError: onMutationErrorProp,
  autoSave,
  submitOnEnter = false,
  warnWhenUnsavedChanges: warnWhenUnsavedChangesProp,
  redirect,
  successNotification,
  errorNotification,
  meta,
  queryMeta,
  mutationMeta,
  liveMode,
  liveParams,
  mutationMode,
  dataProviderName,
  onLiveEvent,
  invalidates,
  undoableTimeout,
  queryOptions,
  createMutationOptions,
  updateMutationOptions,
  id: idFromProps,
  overtimeOptions,
  optimisticUpdateMap,
  defaultFormValues,
  disableServerSideValidation: disableServerSideValidationProp = false
} = {}) => {
  var _a, _b;
  const { options } = useRefineContext();
  const disableServerSideValidation = (options == null ? void 0 : options.disableServerSideValidation) || disableServerSideValidationProp;
  const translate = useTranslate();
  const [formAnt] = Form.useForm();
  const formSF = useFormSF({
    form: formAnt,
    defaultFormValues
  });
  const { form } = formSF;
  const useFormCoreResult = useFormCore({
    onMutationSuccess: onMutationSuccessProp ? onMutationSuccessProp : void 0,
    onMutationError: async (error, _variables, _context) => {
      if (disableServerSideValidation) {
        onMutationErrorProp == null ? void 0 : onMutationErrorProp(error, _variables, _context);
        return;
      }
      let parsedErrors = [];
      const fieldsValue = form.getFieldsValue();
      const fields = Object.keys(flattenObjectKeys(fieldsValue));
      parsedErrors = fields.map((field) => {
        return {
          name: propertyPathToArray(field),
          errors: void 0
        };
      });
      form.setFields(parsedErrors);
      const errors = error == null ? void 0 : error.errors;
      for (const key in errors) {
        const fieldError = errors[key];
        let newError = [];
        if (Array.isArray(fieldError)) {
          newError = fieldError;
        }
        if (typeof fieldError === "string") {
          newError = [fieldError];
        }
        if (typeof fieldError === "boolean" && fieldError) {
          newError = ["Field is not valid."];
        }
        if (typeof fieldError === "object" && "key" in fieldError) {
          const translatedMessage = translate(
            fieldError.key,
            fieldError.message
          );
          newError = [translatedMessage];
        }
        parsedErrors.push({
          name: propertyPathToArray(key),
          errors: newError
        });
      }
      form.setFields([...parsedErrors]);
      onMutationErrorProp == null ? void 0 : onMutationErrorProp(error, _variables, _context);
    },
    redirect,
    action,
    resource,
    successNotification,
    errorNotification,
    meta,
    queryMeta,
    mutationMeta,
    liveMode,
    liveParams,
    mutationMode,
    dataProviderName,
    onLiveEvent,
    invalidates,
    undoableTimeout,
    queryOptions,
    createMutationOptions,
    updateMutationOptions,
    id: idFromProps,
    overtimeOptions,
    optimisticUpdateMap,
    autoSave
  });
  const { formLoading, onFinish, query, id, onFinishAutoSave } = useFormCoreResult;
  const { warnWhenUnsavedChanges: warnWhenUnsavedChangesRefine, setWarnWhen } = useWarnAboutChange();
  const warnWhenUnsavedChanges = warnWhenUnsavedChangesProp ?? warnWhenUnsavedChangesRefine;
  React.useEffect(() => {
    form.resetFields();
  }, [(_a = query == null ? void 0 : query.data) == null ? void 0 : _a.data, id]);
  const onKeyUp = /* @__PURE__ */ __name((event) => {
    if (submitOnEnter && event.key === "Enter") {
      form.submit();
    }
  }, "onKeyUp");
  const onValuesChange = /* @__PURE__ */ __name((changeValues, allValues) => {
    if (changeValues && warnWhenUnsavedChanges) {
      setWarnWhen(true);
    }
    if (autoSave == null ? void 0 : autoSave.enabled) {
      setWarnWhen(false);
      const onFinishFromProps = (autoSave == null ? void 0 : autoSave.onFinish) ?? ((values) => values);
      return onFinishAutoSave(onFinishFromProps(allValues)).catch(
        (error) => error
      );
    }
    return changeValues;
  }, "onValuesChange");
  const saveButtonProps = {
    disabled: formLoading,
    onClick: () => {
      form.submit();
    }
  };
  return {
    form: formSF.form,
    formProps: {
      ...formSF.formProps,
      onFinish: (values) => onFinish(values).catch((error) => error),
      onKeyUp,
      onValuesChange,
      initialValues: (_b = query == null ? void 0 : query.data) == null ? void 0 : _b.data
    },
    saveButtonProps,
    defaultFormValuesLoading: formSF.defaultFormValuesLoading,
    ...useFormCoreResult,
    onFinish: async (values) => {
      return await onFinish(values ?? formSF.form.getFieldsValue(true));
    }
  };
}, "useForm");

// src/hooks/form/useModalForm/useModalForm.ts
import React2, { useCallback } from "react";
import {
  useTranslate as useTranslate2,
  useWarnAboutChange as useWarnAboutChange2,
  useUserFriendlyName,
  useResourceParams,
  useParsed,
  useGo,
  useInvalidate
} from "@refinedev/core";

// src/hooks/modal/useModal/index.tsx
import {
  useModal as useCoreModal
} from "@refinedev/core";
var useModal = /* @__PURE__ */ __name(({
  modalProps = {}
} = {}) => {
  const { show, close, visible } = useCoreModal({
    defaultVisible: modalProps.open
  });
  return {
    modalProps: {
      ...modalProps,
      onCancel: (e) => {
        var _a;
        (_a = modalProps.onCancel) == null ? void 0 : _a.call(modalProps, e);
        close();
      },
      open: visible,
      visible
    },
    show,
    close
  };
}, "useModal");

// src/hooks/form/useModalForm/useModalForm.ts
var useModalForm = /* @__PURE__ */ __name(({
  syncWithLocation,
  defaultVisible = false,
  autoSubmitClose = true,
  autoResetForm = true,
  autoResetFormWhenClose = true,
  autoSave,
  invalidates,
  ...rest
}) => {
  var _a;
  const [initiallySynced, setInitiallySynced] = React2.useState(false);
  const invalidate = useInvalidate();
  const {
    resource,
    action: actionFromParams,
    identifier
  } = useResourceParams({ resource: rest.resource });
  const parsed = useParsed();
  const go = useGo();
  const getUserFriendlyName = useUserFriendlyName();
  const action = rest.action ?? actionFromParams ?? "";
  const syncingId = !(typeof syncWithLocation === "object" && (syncWithLocation == null ? void 0 : syncWithLocation.syncId) === false);
  const syncWithLocationKey = typeof syncWithLocation === "object" && "key" in syncWithLocation ? syncWithLocation.key : resource && action && syncWithLocation ? `modal-${identifier}-${action}` : void 0;
  const useFormProps = useForm({
    meta: {
      ...syncWithLocationKey ? { [syncWithLocationKey]: void 0 } : {},
      ...rest.meta
    },
    autoSave,
    invalidates,
    ...rest
  });
  const { form, formProps, id, setId, formLoading, onFinish, autoSaveProps } = useFormProps;
  const translate = useTranslate2();
  const { warnWhen, setWarnWhen } = useWarnAboutChange2();
  const { show, close, modalProps } = useModal({
    modalProps: {
      open: defaultVisible
    }
  });
  const visible = modalProps.open || false;
  const sunflowerUseModal = {
    modalProps,
    form,
    formLoading,
    formProps,
    formResult: void 0,
    formValues: form.getFieldsValue,
    defaultFormValuesLoading: false,
    initialValues: {},
    submit: onFinish,
    close,
    open: modalProps.open || false,
    show
  };
  React2.useEffect(() => {
    var _a2, _b, _c, _d;
    if (initiallySynced === false && syncWithLocationKey) {
      const openStatus = (_b = (_a2 = parsed == null ? void 0 : parsed.params) == null ? void 0 : _a2[syncWithLocationKey]) == null ? void 0 : _b.open;
      if (typeof openStatus === "boolean") {
        if (openStatus) {
          show();
        }
      } else if (typeof openStatus === "string") {
        if (openStatus === "true") {
          show();
        }
      }
      if (syncingId) {
        const idFromParams = (_d = (_c = parsed == null ? void 0 : parsed.params) == null ? void 0 : _c[syncWithLocationKey]) == null ? void 0 : _d.id;
        if (idFromParams) {
          setId == null ? void 0 : setId(idFromParams);
        }
      }
      setInitiallySynced(true);
    }
  }, [syncWithLocationKey, parsed, syncingId, setId]);
  React2.useEffect(() => {
    var _a2;
    if (initiallySynced === true) {
      if (visible && syncWithLocationKey) {
        go({
          query: {
            [syncWithLocationKey]: {
              ...(_a2 = parsed == null ? void 0 : parsed.params) == null ? void 0 : _a2[syncWithLocationKey],
              open: true,
              ...syncingId && id && { id }
            }
          },
          options: { keepQuery: true },
          type: "replace"
        });
      } else if (syncWithLocationKey && !visible) {
        go({
          query: {
            [syncWithLocationKey]: void 0
          },
          options: { keepQuery: true },
          type: "replace"
        });
      }
    }
  }, [id, visible, show, syncWithLocationKey, syncingId]);
  const saveButtonPropsSF = {
    disabled: formLoading,
    loading: formLoading,
    onClick: () => {
      form.submit();
    }
  };
  const handleClose = useCallback(() => {
    if (autoSaveProps.status === "success" && (autoSave == null ? void 0 : autoSave.invalidateOnClose)) {
      invalidate({
        id,
        invalidates: invalidates || ["list", "many", "detail"],
        dataProviderName: rest.dataProviderName,
        resource: identifier
      });
    }
    if (warnWhen) {
      const warnWhenConfirm = window.confirm(
        translate(
          "warnWhenUnsavedChanges",
          "Are you sure you want to leave? You have unsaved changes."
        )
      );
      if (warnWhenConfirm) {
        setWarnWhen(false);
      } else {
        return;
      }
    }
    setId == null ? void 0 : setId(void 0);
    sunflowerUseModal.close();
    if (autoResetFormWhenClose) {
      form.resetFields();
    }
  }, [warnWhen, autoSaveProps.status]);
  const handleShow = useCallback(
    (showId) => {
      if (typeof showId !== "undefined") {
        setId == null ? void 0 : setId(showId);
      }
      const needsIdToOpen = action === "edit" || action === "clone";
      const hasId = typeof showId !== "undefined" || typeof id !== "undefined";
      if (needsIdToOpen ? hasId : true) {
        sunflowerUseModal.show();
      }
    },
    [id]
  );
  const { visible: _visible, ...otherModalProps } = modalProps;
  const newModalProps = { open: _visible, ...otherModalProps };
  return {
    ...useFormProps,
    ...sunflowerUseModal,
    show: handleShow,
    close: handleClose,
    open: visible,
    formProps: {
      ...formProps,
      ...useFormProps.formProps,
      onValuesChange: formProps == null ? void 0 : formProps.onValuesChange,
      onKeyUp: formProps == null ? void 0 : formProps.onKeyUp,
      onFinish: async (values) => {
        await onFinish(values);
        if (autoSubmitClose) {
          close();
        }
        if (autoResetForm) {
          form.resetFields();
        }
      }
    },
    modalProps: {
      ...newModalProps,
      width: "1000px",
      okButtonProps: saveButtonPropsSF,
      title: translate(
        `${identifier}.titles.${rest.action}`,
        `${getUserFriendlyName(
          `${rest.action} ${((_a = resource == null ? void 0 : resource.meta) == null ? void 0 : _a.label) ?? identifier}`,
          "singular"
        )}`
      ),
      okText: translate("buttons.save", "Save"),
      cancelText: translate("buttons.cancel", "Cancel"),
      onCancel: handleClose,
      forceRender: true
    },
    formLoading
  };
}, "useModalForm");

// src/hooks/form/useDrawerForm/useDrawerForm.ts
import React3, { useCallback as useCallback2 } from "react";
import {
  useTranslate as useTranslate3,
  useWarnAboutChange as useWarnAboutChange3,
  useResourceParams as useResourceParams2,
  useParsed as useParsed2,
  useGo as useGo2,
  useInvalidate as useInvalidate2
} from "@refinedev/core";

// src/hooks/drawer/useDrawer/index.tsx
import { useModal as useModal2 } from "@refinedev/core";
var useDrawer = /* @__PURE__ */ __name(({
  drawerProps = {}
} = {}) => {
  const { show, close, visible } = useModal2({
    defaultVisible: drawerProps.open
  });
  return {
    drawerProps: {
      ...drawerProps,
      onClose: (e) => {
        var _a;
        (_a = drawerProps.onClose) == null ? void 0 : _a.call(drawerProps, e);
        close();
      },
      open: visible
    },
    show,
    close
  };
}, "useDrawer");

// src/hooks/form/useDrawerForm/useDrawerForm.ts
var useDrawerForm = /* @__PURE__ */ __name(({
  syncWithLocation,
  defaultVisible = false,
  autoSubmitClose = true,
  autoResetForm = true,
  autoSave,
  invalidates,
  ...rest
}) => {
  const invalidate = useInvalidate2();
  const [initiallySynced, setInitiallySynced] = React3.useState(false);
  const { show, close, drawerProps } = useDrawer({
    drawerProps: {
      open: defaultVisible
    }
  });
  const visible = drawerProps.open || false;
  const {
    resource,
    action: actionFromParams,
    identifier
  } = useResourceParams2({
    resource: rest.resource
  });
  const parsed = useParsed2();
  const go = useGo2();
  const action = rest.action ?? actionFromParams ?? "";
  const syncingId = !(typeof syncWithLocation === "object" && (syncWithLocation == null ? void 0 : syncWithLocation.syncId) === false);
  const syncWithLocationKey = typeof syncWithLocation === "object" && "key" in syncWithLocation ? syncWithLocation.key : resource && action && syncWithLocation ? `drawer-${(resource == null ? void 0 : resource.identifier) ?? (resource == null ? void 0 : resource.name)}-${action}` : void 0;
  const useFormProps = useForm({
    meta: {
      ...syncWithLocationKey ? { [syncWithLocationKey]: void 0 } : {},
      ...rest.meta
    },
    autoSave,
    invalidates,
    ...rest
  });
  const { form, formProps, formLoading, id, setId, onFinish, autoSaveProps } = useFormProps;
  React3.useEffect(() => {
    var _a, _b, _c, _d;
    if (initiallySynced === false && syncWithLocationKey) {
      const openStatus = (_b = (_a = parsed == null ? void 0 : parsed.params) == null ? void 0 : _a[syncWithLocationKey]) == null ? void 0 : _b.open;
      if (typeof openStatus === "boolean") {
        openStatus ? show() : close();
      } else if (typeof openStatus === "string") {
        if (openStatus === "true") {
          show();
        }
      }
      if (syncingId) {
        const idFromParams = (_d = (_c = parsed == null ? void 0 : parsed.params) == null ? void 0 : _c[syncWithLocationKey]) == null ? void 0 : _d.id;
        if (idFromParams) {
          setId == null ? void 0 : setId(idFromParams);
        }
      }
      setInitiallySynced(true);
    }
  }, [syncWithLocationKey, parsed, syncingId, setId, initiallySynced]);
  React3.useEffect(() => {
    var _a;
    if (initiallySynced === true) {
      if (visible && syncWithLocationKey) {
        go({
          query: {
            [syncWithLocationKey]: {
              ...(_a = parsed == null ? void 0 : parsed.params) == null ? void 0 : _a[syncWithLocationKey],
              open: true,
              ...syncingId && id && { id }
            }
          },
          options: { keepQuery: true },
          type: "replace"
        });
      } else if (syncWithLocationKey && !visible) {
        go({
          query: {
            [syncWithLocationKey]: void 0
          },
          options: { keepQuery: true },
          type: "replace"
        });
      }
    }
  }, [
    id,
    visible,
    show,
    close,
    syncWithLocationKey,
    syncingId,
    initiallySynced
  ]);
  const translate = useTranslate3();
  const { warnWhen, setWarnWhen } = useWarnAboutChange3();
  const saveButtonProps = {
    disabled: formLoading,
    onClick: () => {
      form.submit();
    },
    loading: formLoading
  };
  const deleteButtonProps = {
    recordItemId: id,
    onSuccess: () => {
      setId == null ? void 0 : setId(void 0);
      close();
    }
  };
  const handleClose = useCallback2(() => {
    if (autoSaveProps.status === "success" && (autoSave == null ? void 0 : autoSave.invalidateOnClose)) {
      invalidate({
        id,
        invalidates: invalidates || ["list", "many", "detail"],
        dataProviderName: rest.dataProviderName,
        resource: identifier
      });
    }
    if (warnWhen) {
      const warnWhenConfirm = window.confirm(
        translate(
          "warnWhenUnsavedChanges",
          "Are you sure you want to leave? You have unsaved changes."
        )
      );
      if (warnWhenConfirm) {
        setWarnWhen(false);
      } else {
        return;
      }
    }
    close();
    setId == null ? void 0 : setId(void 0);
  }, [warnWhen]);
  const handleShow = useCallback2(
    (showId) => {
      if (typeof showId !== "undefined") {
        setId == null ? void 0 : setId(showId);
      }
      const needsIdToOpen = action === "edit" || action === "clone";
      const hasId = typeof showId !== "undefined" || typeof id !== "undefined";
      if (needsIdToOpen ? hasId : true) {
        show();
      }
    },
    [id]
  );
  return {
    ...useFormProps,
    show: handleShow,
    close: handleClose,
    formProps: {
      form,
      ...useFormProps.formProps,
      onValuesChange: formProps == null ? void 0 : formProps.onValuesChange,
      onKeyUp: formProps == null ? void 0 : formProps.onKeyUp,
      onFinish: async (values) => {
        await onFinish(values);
        if (autoSubmitClose) {
          close();
        }
        if (autoResetForm) {
          form.resetFields();
        }
      }
    },
    drawerProps: {
      ...drawerProps,
      width: "500px",
      onClose: handleClose,
      open: visible,
      forceRender: true
    },
    saveButtonProps,
    deleteButtonProps,
    formLoading
  };
}, "useDrawerForm");

// src/hooks/form/useStepsForm/useStepsForm.ts
import {
  useStepsForm as useStepsFormSF
} from "sunflower-antd";
var useStepsForm = /* @__PURE__ */ __name((props = {}) => {
  const useFormProps = useForm({
    ...props
  });
  const { form, formProps } = useFormProps;
  const stepsPropsSunflower = useStepsFormSF({
    isBackValidate: false,
    form,
    submit: (values) => {
      var _a;
      (_a = formProps == null ? void 0 : formProps.onFinish) == null ? void 0 : _a.call(formProps, values);
    },
    ...props
  });
  return {
    ...useFormProps,
    ...stepsPropsSunflower,
    formLoading: useFormProps.formLoading,
    formProps: {
      ...stepsPropsSunflower.formProps,
      ...useFormProps.formProps,
      onValuesChange: formProps == null ? void 0 : formProps.onValuesChange,
      onKeyUp: formProps == null ? void 0 : formProps.onKeyUp
    },
    saveButtonProps: {
      ...useFormProps.saveButtonProps,
      onClick: () => stepsPropsSunflower.submit()
    }
  };
}, "useStepsForm");

// src/hooks/table/useTable/useTable.ts
import React5, { Children, createElement, Fragment } from "react";
import {
  Grid,
  Form as Form2
} from "antd";
import { useForm as useFormSF2 } from "sunflower-antd";
import {
  useLiveMode,
  useTable as useTableCore,
  useSyncWithLocation
} from "@refinedev/core";

// src/definitions/table/index.ts
import {
  getDefaultFilter as getDefaultFilterCore,
  getDefaultSortOrder as getDefaultSortOrderCore
} from "@refinedev/core";
var getDefaultSortOrder = /* @__PURE__ */ __name((columnName, sorter) => {
  const sort = getDefaultSortOrderCore(columnName, sorter);
  if (sort) {
    return `${sort}end`;
  }
  return void 0;
}, "getDefaultSortOrder");
var getDefaultFilter = /* @__PURE__ */ __name((columnName, filters, operatorType = "eq") => {
  return getDefaultFilterCore(columnName, filters, operatorType);
}, "getDefaultFilter");
var mapAntdSorterToCrudSorting = /* @__PURE__ */ __name((sorter) => {
  const crudSorting = [];
  if (Array.isArray(sorter)) {
    sorter.sort((a, b) => {
      var _a, _b;
      return (((_a = a.column) == null ? void 0 : _a.sorter).multiple ?? 0) < (((_b = b.column) == null ? void 0 : _b.sorter).multiple ?? 0) ? -1 : 0;
    }).map((item) => {
      if (item.field && item.order) {
        const field = Array.isArray(item.field) ? item.field.join(".") : `${item.field}`;
        crudSorting.push({
          field: `${item.columnKey ?? field}`,
          order: item.order.replace("end", "")
        });
      }
    });
  } else {
    if (sorter.field && sorter.order) {
      const field = Array.isArray(sorter.field) ? sorter.field.join(".") : `${sorter.field}`;
      crudSorting.push({
        field: `${sorter.columnKey ?? field}`,
        order: sorter.order.replace("end", "")
      });
    }
  }
  return crudSorting;
}, "mapAntdSorterToCrudSorting");
var mapAntdFilterToCrudFilter = /* @__PURE__ */ __name((tableFilters, prevFilters, initialFilters) => {
  const crudFilters = [];
  const mapInitialFilter = (initialFilters ?? []).reduce((acc, item) => {
    const field = item.key || item.field;
    return { ...acc, [field]: item };
  }, {});
  Object.keys(tableFilters).map((field) => {
    var _a, _b;
    const value = tableFilters[field];
    const operator = ((_a = prevFilters.filter((i) => i.operator !== "or").find((p) => p.field === field)) == null ? void 0 : _a.operator) || ((_b = mapInitialFilter[field]) == null ? void 0 : _b.operator);
    if (operator !== "or" && operator !== "and") {
      crudFilters.push({
        field,
        operator: operator ?? (Array.isArray(value) ? "in" : "eq"),
        value
      });
    }
  });
  return crudFilters;
}, "mapAntdFilterToCrudFilter");

// src/hooks/table/useTable/paginationLink.tsx
import { useLink } from "@refinedev/core";
import React4 from "react";
var PaginationLink = /* @__PURE__ */ __name(({ to, element }) => {
  const Link = useLink();
  return /* @__PURE__ */ React4.createElement(
    Link,
    {
      to,
      replace: false,
      onClick: (e) => {
        e.preventDefault();
      }
    },
    element
  );
}, "PaginationLink");

// src/hooks/table/useTable/useTable.ts
var useTable = /* @__PURE__ */ __name(({
  onSearch,
  pagination: paginationFromProp,
  filters: filtersFromProp,
  sorters: sortersFromProp,
  syncWithLocation,
  resource,
  successNotification,
  errorNotification,
  queryOptions,
  liveMode: liveModeFromProp,
  onLiveEvent,
  liveParams,
  meta,
  dataProviderName
} = {}) => {
  const {
    tableQuery,
    currentPage,
    setCurrentPage,
    pageSize,
    setPageSize,
    filters,
    setFilters,
    sorters,
    setSorters,
    createLinkForSyncWithLocation,
    pageCount,
    overtime,
    result
  } = useTableCore({
    pagination: paginationFromProp,
    filters: filtersFromProp,
    sorters: sortersFromProp,
    syncWithLocation,
    resource,
    successNotification,
    errorNotification,
    queryOptions,
    liveMode: liveModeFromProp,
    onLiveEvent,
    liveParams,
    meta,
    dataProviderName
  });
  const { syncWithLocation: defaultSyncWithLocation } = useSyncWithLocation();
  const shouldSyncWithLocation = syncWithLocation ?? defaultSyncWithLocation;
  const breakpoint = Grid.useBreakpoint();
  const [form] = Form2.useForm();
  const formSF = useFormSF2({
    form
  });
  const liveMode = useLiveMode(liveModeFromProp);
  const isPaginationEnabled = (paginationFromProp == null ? void 0 : paginationFromProp.mode) !== "off";
  const preferredInitialFilters = filtersFromProp == null ? void 0 : filtersFromProp.initial;
  const { data, isFetched, isLoading } = tableQuery;
  React5.useEffect(() => {
    if (shouldSyncWithLocation) {
      const registeredFields = formSF.form.getFieldsValue();
      const filterFilterMap = Object.keys(registeredFields).reduce(
        (acc, curr) => {
          const filter = filters.find(
            (filter2) => "field" in filter2 && filter2.field === curr
          );
          if (filter) {
            acc[curr] = filter == null ? void 0 : filter.value;
          }
          return acc;
        },
        {}
      );
      formSF.form.setFieldsValue(filterFilterMap);
    }
  }, [shouldSyncWithLocation]);
  const onChange = /* @__PURE__ */ __name((paginationState, tableFilters, sorter) => {
    if (tableFilters && Object.keys(tableFilters).length > 0) {
      const crudFilters = mapAntdFilterToCrudFilter(
        tableFilters,
        filters,
        preferredInitialFilters
      );
      setFilters(crudFilters);
    }
    if (sorter && Object.keys(sorter).length > 0) {
      const crudSorting = mapAntdSorterToCrudSorting(sorter);
      setSorters(crudSorting);
    }
    if (isPaginationEnabled) {
      setCurrentPage == null ? void 0 : setCurrentPage(paginationState.current || 1);
      setPageSize == null ? void 0 : setPageSize(paginationState.pageSize || 10);
    }
  }, "onChange");
  const onFinish = /* @__PURE__ */ __name(async (value) => {
    if (onSearch) {
      const searchFilters = await onSearch(value);
      setFilters(searchFilters);
      if (isPaginationEnabled) {
        setCurrentPage == null ? void 0 : setCurrentPage(1);
      }
    }
  }, "onFinish");
  const antdPagination = /* @__PURE__ */ __name(() => {
    if (isPaginationEnabled) {
      return {
        itemRender: (page, type, element) => {
          var _a;
          const link = createLinkForSyncWithLocation({
            pagination: {
              pageSize,
              currentPage: page
            },
            sorters,
            filters
          });
          if (type === "page") {
            return createElement(PaginationLink, {
              to: link,
              element: `${page}`
            });
          }
          if (type === "next" || type === "prev") {
            return createElement(PaginationLink, {
              to: link,
              element
            });
          }
          if (type === "jump-next" || type === "jump-prev") {
            const elementChildren = (_a = element == null ? void 0 : element.props) == null ? void 0 : _a.children;
            return createElement(PaginationLink, {
              to: link,
              element: Children.count(elementChildren) > 1 ? createElement(Fragment, {}, elementChildren) : elementChildren
            });
          }
          return element;
        },
        pageSize,
        current: currentPage,
        simple: !breakpoint.sm,
        position: !breakpoint.sm ? ["bottomCenter"] : ["bottomRight"],
        total: data == null ? void 0 : data.total
      };
    }
    return false;
  }, "antdPagination");
  return {
    searchFormProps: {
      ...formSF.formProps,
      onFinish
    },
    tableProps: {
      dataSource: data == null ? void 0 : data.data,
      loading: liveMode === "auto" ? isLoading : !isFetched,
      onChange,
      pagination: antdPagination(),
      scroll: { x: true }
    },
    tableQuery,
    sorters,
    filters,
    setSorters,
    setFilters,
    currentPage,
    setCurrentPage,
    pageSize,
    setPageSize,
    pageCount,
    createLinkForSyncWithLocation,
    overtime,
    result
  };
}, "useTable");

// src/hooks/table/useEditableTable/useEditableTable.ts
var useEditableTable = /* @__PURE__ */ __name(({
  autoSubmitClose = true,
  ...props
} = {}) => {
  const table = useTable({
    ...props,
    successNotification: void 0,
    errorNotification: void 0
  });
  const edit = useForm({
    ...props,
    action: "edit",
    redirect: false
  });
  const { id: editId, setId, saveButtonProps } = edit;
  const cancelButtonProps = {
    onClick: () => {
      setId(void 0);
    }
  };
  const editButtonProps = /* @__PURE__ */ __name((id) => {
    return {
      onClick: () => setId(id)
    };
  }, "editButtonProps");
  const isEditing = /* @__PURE__ */ __name((id) => id === editId, "isEditing");
  return {
    ...table,
    ...edit,
    formProps: {
      ...edit.formProps,
      onFinish: async (values) => {
        try {
          const result = await edit.onFinish(values);
          if (autoSubmitClose) {
            setId(void 0);
          }
          return result;
        } catch (error) {
          return error;
        }
      }
    },
    saveButtonProps,
    cancelButtonProps,
    editButtonProps,
    isEditing
  };
}, "useEditableTable");

// src/hooks/fields/useSelect/index.ts
import {
  useSelect as useSelectCore
} from "@refinedev/core";
var useSelect = /* @__PURE__ */ __name((props) => {
  const { query, defaultValueQuery, onSearch, options } = useSelectCore(props);
  return {
    selectProps: {
      options,
      onSearch,
      loading: defaultValueQuery.query.isFetching,
      showSearch: true,
      filterOption: false
    },
    query,
    defaultValueQuery: defaultValueQuery.query
  };
}, "useSelect");

// src/hooks/fields/useCheckboxGroup/index.ts
import {
  useSelect as useSelect2
} from "@refinedev/core";
var useCheckboxGroup = /* @__PURE__ */ __name(({
  resource,
  sorters,
  filters,
  optionLabel,
  optionValue,
  queryOptions,
  pagination,
  liveMode,
  defaultValue,
  selectedOptionsOrder,
  onLiveEvent,
  liveParams,
  meta,
  dataProviderName,
  ...rest
}) => {
  const { query, options } = useSelect2({
    resource,
    sorters,
    filters,
    optionLabel,
    optionValue,
    queryOptions,
    pagination,
    liveMode,
    defaultValue,
    selectedOptionsOrder,
    onLiveEvent,
    liveParams,
    meta,
    dataProviderName,
    ...rest
  });
  return {
    checkboxGroupProps: {
      options,
      defaultValue
    },
    query
  };
}, "useCheckboxGroup");

// src/hooks/fields/useRadioGroup/index.ts
import {
  useSelect as useSelect3
} from "@refinedev/core";
var useRadioGroup = /* @__PURE__ */ __name(({
  resource,
  sorters,
  filters,
  optionLabel,
  optionValue,
  queryOptions,
  pagination,
  liveMode,
  defaultValue,
  selectedOptionsOrder,
  onLiveEvent,
  liveParams,
  meta,
  dataProviderName,
  ...rest
}) => {
  const { query, options } = useSelect3({
    resource,
    sorters,
    filters,
    optionLabel,
    optionValue,
    queryOptions,
    pagination,
    liveMode,
    defaultValue,
    selectedOptionsOrder,
    onLiveEvent,
    liveParams,
    meta,
    dataProviderName,
    ...rest
  });
  return {
    radioGroupProps: {
      options,
      defaultValue
    },
    query
  };
}, "useRadioGroup");

// src/hooks/import/index.tsx
import React6 from "react";
import {
  notification,
  Progress
} from "antd";
import {
  useTranslate as useTranslate4,
  useResourceParams as useResourceParams3,
  useImport as useImportCore
} from "@refinedev/core";
var useImport = /* @__PURE__ */ __name(({
  resource: resourceFromProp,
  mapData = /* @__PURE__ */ __name((item) => item, "mapData"),
  paparseOptions,
  batchSize = Number.MAX_SAFE_INTEGER,
  onFinish,
  meta,
  dataProviderName,
  onProgress: onProgressFromProp
} = {}) => {
  const t = useTranslate4();
  const { resource } = useResourceParams3({ resource: resourceFromProp });
  const { mutationResult, isLoading, handleChange } = useImportCore({
    resource: (resource == null ? void 0 : resource.identifier) ?? (resource == null ? void 0 : resource.name),
    mapData,
    paparseOptions,
    batchSize,
    meta,
    dataProviderName,
    onFinish,
    onProgress: onProgressFromProp ?? (({ totalAmount, processedAmount }) => {
      if (totalAmount > 0 && processedAmount > 0) {
        const description = /* @__PURE__ */ React6.createElement(
          "div",
          {
            style: {
              display: "flex",
              alignItems: "center",
              justifyContent: "space-between",
              marginTop: "-7px"
            }
          },
          /* @__PURE__ */ React6.createElement(
            Progress,
            {
              type: "circle",
              percent: Math.floor(processedAmount / totalAmount * 100),
              size: 50,
              strokeColor: "#1890ff",
              status: "normal"
            }
          ),
          /* @__PURE__ */ React6.createElement("span", { style: { marginLeft: 8, width: "100%" } }, t(
            "notifications.importProgress",
            {
              processed: processedAmount,
              total: totalAmount
            },
            `Importing: ${processedAmount}/${totalAmount}`
          ))
        );
        notification.open({
          description,
          message: null,
          key: `${resource}-import`,
          duration: 0
        });
        if (processedAmount >= totalAmount) {
        }
        if (processedAmount === totalAmount) {
          setTimeout(() => {
            notification.destroy(`${resource}-import`);
          }, 4500);
        }
      }
    })
  });
  return {
    uploadProps: {
      onChange: handleChange,
      beforeUpload: () => false,
      showUploadList: false,
      accept: ".csv"
    },
    buttonProps: {
      type: "default",
      loading: isLoading
    },
    mutationResult,
    isLoading
  };
}, "useImport");

// src/hooks/useSimpleList/useSimpleList.ts
import { Children as Children2, createElement as createElement2, Fragment as Fragment2 } from "react";
import { Form as Form3, Grid as Grid2 } from "antd";
import {
  useTable as useTableCore2
} from "@refinedev/core";
import { useLiveMode as useLiveMode2 } from "@refinedev/core";
var useSimpleList = /* @__PURE__ */ __name(({
  resource,
  pagination: paginationFromProp,
  filters: filtersFromProp,
  sorters: sortersFromProp,
  onSearch,
  queryOptions,
  syncWithLocation,
  successNotification,
  errorNotification,
  liveMode: liveModeFromProp,
  onLiveEvent,
  liveParams,
  meta,
  dataProviderName
} = {}) => {
  const {
    sorters,
    filters,
    currentPage,
    pageSize,
    pageCount,
    setFilters,
    setCurrentPage,
    setPageSize,
    setSorters,
    createLinkForSyncWithLocation,
    tableQuery,
    overtime,
    result
  } = useTableCore2({
    resource,
    pagination: paginationFromProp,
    filters: filtersFromProp,
    sorters: sortersFromProp,
    queryOptions,
    successNotification,
    errorNotification,
    liveMode: liveModeFromProp,
    onLiveEvent,
    liveParams,
    meta,
    syncWithLocation,
    dataProviderName
  });
  const isPaginationEnabled = (paginationFromProp == null ? void 0 : paginationFromProp.mode) !== "off";
  const breakpoint = Grid2.useBreakpoint();
  const liveMode = useLiveMode2(liveModeFromProp);
  const [form] = Form3.useForm();
  const { data, isFetched, isLoading } = tableQuery;
  const onChange = /* @__PURE__ */ __name((page, pageSize2) => {
    if (isPaginationEnabled) {
      setCurrentPage(page);
      setPageSize(pageSize2 || 10);
    }
  }, "onChange");
  const onFinish = /* @__PURE__ */ __name(async (values) => {
    if (onSearch) {
      const searchFilters = await onSearch(values);
      if (isPaginationEnabled) {
        setCurrentPage == null ? void 0 : setCurrentPage(1);
      }
      return setFilters(searchFilters);
    }
  }, "onFinish");
  const antdPagination = /* @__PURE__ */ __name(() => {
    if (isPaginationEnabled) {
      return {
        itemRender: (page, type, element) => {
          var _a;
          const link = createLinkForSyncWithLocation({
            pagination: {
              pageSize,
              currentPage: page
            },
            sorters,
            filters
          });
          if (type === "page") {
            return createElement2(PaginationLink, {
              to: link,
              element: `${page}`
            });
          }
          if (type === "next" || type === "prev") {
            return createElement2(PaginationLink, {
              to: link,
              element
            });
          }
          if (type === "jump-next" || type === "jump-prev") {
            const elementChildren = (_a = element == null ? void 0 : element.props) == null ? void 0 : _a.children;
            return createElement2(PaginationLink, {
              to: link,
              element: Children2.count(elementChildren) > 1 ? createElement2(Fragment2, {}, elementChildren) : elementChildren
            });
          }
          return element;
        },
        pageSize,
        current: currentPage,
        simple: !breakpoint.sm,
        total: data == null ? void 0 : data.total,
        onChange
      };
    }
    return false;
  }, "antdPagination");
  return {
    searchFormProps: {
      form,
      onFinish
    },
    listProps: {
      dataSource: data == null ? void 0 : data.data,
      loading: liveMode === "auto" ? isLoading : !isFetched,
      pagination: antdPagination()
    },
    query: tableQuery,
    filters,
    setFilters,
    sorters,
    setSorters,
    currentPage,
    setCurrentPage,
    pageSize,
    setPageSize,
    pageCount,
    createLinkForSyncWithLocation,
    overtime,
    result
  };
}, "useSimpleList");

// src/hooks/useFileUploadState/index.ts
import { useCallback as useCallback3, useMemo, useState } from "react";
var useFileUploadState = /* @__PURE__ */ __name(() => {
  const [isLoading, setIsloading] = useState(false);
  const onChange = useCallback3((info) => {
    const fileListLoadings = mapStatusToLoading(info.fileList);
    if (fileListLoadings.includes(true)) {
      setIsloading(true);
    } else {
      setIsloading(false);
    }
  }, []);
  return useMemo(() => ({ isLoading, onChange }), [isLoading]);
}, "useFileUploadState");
var mapStatusToLoading = /* @__PURE__ */ __name((files) => {
  return files.map((file) => {
    switch (file.status) {
      case "uploading":
        return true;
      default:
        return false;
    }
  });
}, "mapStatusToLoading");

// src/hooks/useThemedLayoutContext/index.ts
import { useContext } from "react";

// src/contexts/themedLayoutContext/index.tsx
import React7, { useState as useState2 } from "react";
var ThemedLayoutContext = React7.createContext({
  siderCollapsed: false,
  mobileSiderOpen: false,
  setSiderCollapsed: () => void 0,
  setMobileSiderOpen: () => void 0
});
var ThemedLayoutContextProvider = /* @__PURE__ */ __name(({ children, initialSiderCollapsed, onSiderCollapsed }) => {
  const [siderCollapsed, setSiderCollapsedState] = useState2(
    initialSiderCollapsed ?? false
  );
  const [mobileSiderOpen, setMobileSiderOpen] = useState2(false);
  const setSiderCollapsed = /* @__PURE__ */ __name((collapsed) => {
    setSiderCollapsedState(collapsed);
    if (onSiderCollapsed) {
      onSiderCollapsed(collapsed);
    }
  }, "setSiderCollapsed");
  return /* @__PURE__ */ React7.createElement(
    ThemedLayoutContext.Provider,
    {
      value: {
        siderCollapsed,
        mobileSiderOpen,
        setSiderCollapsed,
        setMobileSiderOpen
      }
    },
    children
  );
}, "ThemedLayoutContextProvider");

// src/hooks/useThemedLayoutContext/index.ts
var useThemedLayoutContext = /* @__PURE__ */ __name(() => {
  const {
    mobileSiderOpen,
    siderCollapsed,
    setMobileSiderOpen,
    setSiderCollapsed
  } = useContext(ThemedLayoutContext);
  return {
    mobileSiderOpen,
    siderCollapsed,
    setMobileSiderOpen,
    setSiderCollapsed
  };
}, "useThemedLayoutContext");

// src/providers/notificationProvider/index.tsx
import { App, notification as staticNotification } from "antd";
import React9 from "react";

// src/components/undoableNotification/index.tsx
import React8 from "react";
import { Button, Progress as Progress2 } from "antd";
import { UndoOutlined } from "@ant-design/icons";
var UndoableNotification = /* @__PURE__ */ __name(({
  message,
  cancelMutation,
  undoableTimeout
}) => /* @__PURE__ */ React8.createElement(
  "div",
  {
    style: {
      display: "flex",
      alignItems: "center",
      justifyContent: "space-between",
      marginTop: "-7px"
    }
  },
  /* @__PURE__ */ React8.createElement(
    Progress2,
    {
      type: "circle",
      percent: (undoableTimeout ?? 0) * 20,
      format: (time) => time && time / 20,
      size: 50,
      strokeColor: "#1890ff",
      status: "normal"
    }
  ),
  /* @__PURE__ */ React8.createElement("span", { style: { marginLeft: 8, width: "100%" } }, message),
  /* @__PURE__ */ React8.createElement(
    Button,
    {
      style: { flexShrink: 0 },
      onClick: cancelMutation,
      disabled: undoableTimeout === 0,
      icon: /* @__PURE__ */ React8.createElement(UndoOutlined, null)
    }
  )
), "UndoableNotification");

// src/providers/notificationProvider/index.tsx
var useNotificationProvider = /* @__PURE__ */ __name(() => {
  const { notification: notificationFromContext } = App.useApp();
  const notification2 = "open" in notificationFromContext ? notificationFromContext : staticNotification;
  const notificationProvider = {
    open: ({
      key,
      message,
      description,
      type,
      cancelMutation,
      undoableTimeout
    }) => {
      if (type === "progress") {
        notification2.open({
          key,
          description: /* @__PURE__ */ React9.createElement(
            UndoableNotification,
            {
              notificationKey: key,
              message,
              cancelMutation: () => {
                cancelMutation == null ? void 0 : cancelMutation();
                notification2.destroy(key ?? "");
              },
              undoableTimeout
            }
          ),
          message: null,
          duration: 0,
          closeIcon: /* @__PURE__ */ React9.createElement(React9.Fragment, null)
        });
      } else {
        notification2.open({
          key,
          description: message,
          message: description ?? null,
          type
        });
      }
    },
    close: (key) => notification2.destroy(key)
  };
  return notificationProvider;
}, "useNotificationProvider");

// src/components/themedLayout/index.tsx
import React12 from "react";
import { Grid as Grid4, Layout as AntdLayout2 } from "antd";

// src/components/themedLayout/sider/index.tsx
import React10, { useContext as useContext2 } from "react";
import {
  Layout,
  Menu,
  Grid as Grid3,
  Drawer,
  Button as Button2,
  theme,
  ConfigProvider
} from "antd";
import {
  LogoutOutlined,
  UnorderedListOutlined,
  BarsOutlined,
  LeftOutlined,
  RightOutlined
} from "@ant-design/icons";
import {
  useTranslate as useTranslate5,
  useLogout,
  CanAccess,
  useIsExistAuthentication,
  useMenu,
  useLink as useLink2,
  useWarnAboutChange as useWarnAboutChange4
} from "@refinedev/core";

// src/components/themedLayout/sider/styles.ts
var drawerButtonStyles = {
  borderStartStartRadius: 0,
  borderEndStartRadius: 0,
  position: "fixed",
  top: 64,
  zIndex: 999
};

// src/components/themedLayout/sider/index.tsx
var ThemedSider = /* @__PURE__ */ __name(({
  Title: TitleFromProps,
  render,
  meta,
  fixed,
  activeItemDisabled = false,
  siderItemsAreCollapsed = true
}) => {
  var _a;
  const { token } = theme.useToken();
  const {
    siderCollapsed,
    setSiderCollapsed,
    mobileSiderOpen,
    setMobileSiderOpen
  } = useThemedLayoutContext();
  const isExistAuthentication = useIsExistAuthentication();
  const direction = (_a = useContext2(ConfigProvider.ConfigContext)) == null ? void 0 : _a.direction;
  const Link = useLink2();
  const { warnWhen, setWarnWhen } = useWarnAboutChange4();
  const translate = useTranslate5();
  const { menuItems, selectedKey, defaultOpenKeys } = useMenu({ meta });
  const breakpoint = Grid3.useBreakpoint();
  const { mutate: mutateLogout } = useLogout();
  const isMobile = typeof breakpoint.lg === "undefined" ? false : !breakpoint.lg;
  const RenderToTitle = TitleFromProps ?? ThemedTitle;
  const renderTreeView = /* @__PURE__ */ __name((tree, selectedKey2) => {
    return tree.map((item) => {
      const { key, name, children, meta: meta2, list } = item;
      const parentName = meta2 == null ? void 0 : meta2.parent;
      const label = (item == null ? void 0 : item.label) ?? (meta2 == null ? void 0 : meta2.label) ?? name;
      const icon = meta2 == null ? void 0 : meta2.icon;
      const route = list;
      if (children.length > 0) {
        return /* @__PURE__ */ React10.createElement(
          CanAccess,
          {
            key: item.key,
            resource: name,
            action: "list",
            params: {
              resource: item
            }
          },
          /* @__PURE__ */ React10.createElement(
            Menu.SubMenu,
            {
              key: item.key,
              icon: icon ?? /* @__PURE__ */ React10.createElement(UnorderedListOutlined, null),
              title: label
            },
            renderTreeView(children, selectedKey2)
          )
        );
      }
      const isSelected = key === selectedKey2;
      const isRoute = !(parentName !== void 0 && children.length === 0);
      const linkStyle = activeItemDisabled && isSelected ? { pointerEvents: "none" } : {};
      return /* @__PURE__ */ React10.createElement(
        CanAccess,
        {
          key: item.key,
          resource: name,
          action: "list",
          params: {
            resource: item
          }
        },
        /* @__PURE__ */ React10.createElement(
          Menu.Item,
          {
            key: item.key,
            icon: icon ?? (isRoute && /* @__PURE__ */ React10.createElement(UnorderedListOutlined, null)),
            style: linkStyle
          },
          /* @__PURE__ */ React10.createElement(Link, { to: route ?? "", style: linkStyle }, label),
          !siderCollapsed && isSelected && /* @__PURE__ */ React10.createElement("div", { className: "ant-menu-tree-arrow" })
        )
      );
    });
  }, "renderTreeView");
  const handleLogout = /* @__PURE__ */ __name(() => {
    if (warnWhen) {
      const confirm = window.confirm(
        translate(
          "warnWhenUnsavedChanges",
          "Are you sure you want to leave? You have unsaved changes."
        )
      );
      if (confirm) {
        setWarnWhen(false);
        mutateLogout();
      }
    } else {
      mutateLogout();
    }
  }, "handleLogout");
  const logout = isExistAuthentication && /* @__PURE__ */ React10.createElement(
    Menu.Item,
    {
      key: "logout",
      onClick: () => handleLogout(),
      icon: /* @__PURE__ */ React10.createElement(LogoutOutlined, null)
    },
    translate("buttons.logout", "Logout")
  );
  const defaultExpandMenuItems = (() => {
    if (siderItemsAreCollapsed)
      return [];
    return menuItems.map(({ key }) => key);
  })();
  const items = renderTreeView(menuItems, selectedKey);
  const renderSider = /* @__PURE__ */ __name(() => {
    if (render) {
      return render({
        items,
        logout,
        collapsed: siderCollapsed
      });
    }
    return [...items, logout].filter(Boolean);
  }, "renderSider");
  const renderMenu = /* @__PURE__ */ __name(() => {
    return /* @__PURE__ */ React10.createElement(
      Menu,
      {
        selectedKeys: selectedKey ? [selectedKey] : [],
        defaultOpenKeys: [...defaultOpenKeys, ...defaultExpandMenuItems],
        mode: "inline",
        style: {
          paddingTop: "8px",
          border: "none",
          overflow: "auto",
          height: "calc(100% - 72px)"
        },
        onClick: () => {
          setMobileSiderOpen(false);
        }
      },
      renderSider()
    );
  }, "renderMenu");
  const renderDrawerSider = /* @__PURE__ */ __name(() => {
    return /* @__PURE__ */ React10.createElement(React10.Fragment, null, /* @__PURE__ */ React10.createElement(
      Drawer,
      {
        open: mobileSiderOpen,
        onClose: () => setMobileSiderOpen(false),
        placement: direction === "rtl" ? "right" : "left",
        closable: false,
        width: 200,
        styles: {
          body: {
            padding: 0
          }
        },
        maskClosable: true
      },
      /* @__PURE__ */ React10.createElement(Layout, null, /* @__PURE__ */ React10.createElement(
        Layout.Sider,
        {
          style: {
            height: "100vh",
            backgroundColor: token.colorBgContainer,
            borderRight: `1px solid ${token.colorBgElevated}`
          }
        },
        /* @__PURE__ */ React10.createElement(
          "div",
          {
            style: {
              width: "200px",
              padding: "0 16px",
              display: "flex",
              justifyContent: "flex-start",
              alignItems: "center",
              height: "64px",
              backgroundColor: token.colorBgElevated
            }
          },
          /* @__PURE__ */ React10.createElement(RenderToTitle, { collapsed: false })
        ),
        renderMenu()
      ))
    ), /* @__PURE__ */ React10.createElement(
      Button2,
      {
        style: drawerButtonStyles,
        size: "large",
        onClick: () => setMobileSiderOpen(true),
        icon: /* @__PURE__ */ React10.createElement(BarsOutlined, null)
      }
    ));
  }, "renderDrawerSider");
  if (isMobile) {
    return renderDrawerSider();
  }
  const siderStyles = {
    backgroundColor: token.colorBgContainer,
    borderRight: `1px solid ${token.colorBgElevated}`
  };
  if (fixed) {
    siderStyles.position = "fixed";
    siderStyles.top = 0;
    siderStyles.height = "100vh";
    siderStyles.zIndex = 999;
  }
  const renderClosingIcons = /* @__PURE__ */ __name(() => {
    const iconProps = { style: { color: token.colorPrimary } };
    const OpenIcon = direction === "rtl" ? RightOutlined : LeftOutlined;
    const CollapsedIcon = direction === "rtl" ? LeftOutlined : RightOutlined;
    const IconComponent = siderCollapsed ? CollapsedIcon : OpenIcon;
    return /* @__PURE__ */ React10.createElement(IconComponent, { ...iconProps });
  }, "renderClosingIcons");
  return /* @__PURE__ */ React10.createElement(React10.Fragment, null, fixed && /* @__PURE__ */ React10.createElement(
    "div",
    {
      style: {
        width: siderCollapsed ? "80px" : "200px",
        transition: "all 0.2s"
      }
    }
  ), /* @__PURE__ */ React10.createElement(
    Layout.Sider,
    {
      style: siderStyles,
      collapsible: true,
      collapsed: siderCollapsed,
      onCollapse: (collapsed, type) => {
        if (type === "clickTrigger") {
          setSiderCollapsed(collapsed);
        }
      },
      collapsedWidth: 80,
      breakpoint: "lg",
      trigger: /* @__PURE__ */ React10.createElement(
        Button2,
        {
          type: "text",
          style: {
            borderRadius: 0,
            height: "100%",
            width: "100%",
            backgroundColor: token.colorBgElevated
          }
        },
        renderClosingIcons()
      )
    },
    /* @__PURE__ */ React10.createElement(
      "div",
      {
        style: {
          width: siderCollapsed ? "80px" : "200px",
          padding: siderCollapsed ? "0" : "0 16px",
          display: "flex",
          justifyContent: siderCollapsed ? "center" : "flex-start",
          alignItems: "center",
          height: "64px",
          backgroundColor: token.colorBgElevated,
          fontSize: "14px"
        }
      },
      /* @__PURE__ */ React10.createElement(RenderToTitle, { collapsed: siderCollapsed })
    ),
    renderMenu()
  ));
}, "ThemedSider");

// src/components/themedLayout/header/index.tsx
import React11 from "react";
import { Layout as AntdLayout, Typography, Avatar, Space, theme as theme2 } from "antd";
import { useGetIdentity } from "@refinedev/core";
var ThemedHeader = /* @__PURE__ */ __name(({
  sticky
}) => {
  const { token } = theme2.useToken();
  const { data: user } = useGetIdentity();
  const shouldRenderHeader = user && (user.name || user.avatar);
  if (!shouldRenderHeader) {
    return null;
  }
  const headerStyles = {
    backgroundColor: token.colorBgElevated,
    display: "flex",
    justifyContent: "flex-end",
    alignItems: "center",
    padding: "0px 24px",
    height: "64px"
  };
  if (sticky) {
    headerStyles.position = "sticky";
    headerStyles.top = 0;
    headerStyles.zIndex = 1;
  }
  return /* @__PURE__ */ React11.createElement(AntdLayout.Header, { style: headerStyles }, /* @__PURE__ */ React11.createElement(Space, null, /* @__PURE__ */ React11.createElement(Space, { size: "middle" }, (user == null ? void 0 : user.name) && /* @__PURE__ */ React11.createElement(Typography.Text, { strong: true }, user.name), (user == null ? void 0 : user.avatar) && /* @__PURE__ */ React11.createElement(Avatar, { src: user == null ? void 0 : user.avatar, alt: user == null ? void 0 : user.name }))));
}, "ThemedHeader");

// src/components/themedLayout/index.tsx
var ThemedLayout = /* @__PURE__ */ __name(({
  children,
  Header,
  Sider,
  Title,
  Footer,
  OffLayoutArea,
  initialSiderCollapsed,
  onSiderCollapsed
}) => {
  const breakpoint = Grid4.useBreakpoint();
  const SiderToRender = Sider ?? ThemedSider;
  const HeaderToRender = Header ?? ThemedHeader;
  const isSmall = typeof breakpoint.sm === "undefined" ? true : breakpoint.sm;
  const hasSider = !!SiderToRender({ Title });
  return /* @__PURE__ */ React12.createElement(
    ThemedLayoutContextProvider,
    {
      initialSiderCollapsed,
      onSiderCollapsed
    },
    /* @__PURE__ */ React12.createElement(AntdLayout2, { style: { minHeight: "100vh" }, hasSider }, /* @__PURE__ */ React12.createElement(SiderToRender, { Title }), /* @__PURE__ */ React12.createElement(AntdLayout2, null, /* @__PURE__ */ React12.createElement(HeaderToRender, null), /* @__PURE__ */ React12.createElement(AntdLayout2.Content, null, /* @__PURE__ */ React12.createElement(
      "div",
      {
        style: {
          minHeight: 360,
          padding: isSmall ? 24 : 12
        }
      },
      children
    ), OffLayoutArea && /* @__PURE__ */ React12.createElement(OffLayoutArea, null)), Footer && /* @__PURE__ */ React12.createElement(Footer, null)))
  );
}, "ThemedLayout");

// src/components/themedLayout/title/index.tsx
import React13 from "react";
import { useLink as useLink3, useRefineOptions } from "@refinedev/core";
import { Typography as Typography2, theme as theme3, Space as Space2 } from "antd";
var ThemedTitle = /* @__PURE__ */ __name(({
  collapsed,
  icon: iconFromProps,
  text: textFromProps,
  wrapperStyles
}) => {
  const {
    title: { icon: defaultIcon, text: defaultText } = {}
  } = useRefineOptions();
  const icon = typeof iconFromProps === "undefined" ? defaultIcon : iconFromProps;
  const text = typeof textFromProps === "undefined" ? defaultText : textFromProps;
  const { token } = theme3.useToken();
  const Link = useLink3();
  return /* @__PURE__ */ React13.createElement(
    Link,
    {
      to: "/",
      style: {
        display: "inline-block",
        textDecoration: "none"
      }
    },
    /* @__PURE__ */ React13.createElement(
      Space2,
      {
        style: {
          display: "flex",
          alignItems: "center",
          fontSize: "inherit",
          ...wrapperStyles
        }
      },
      /* @__PURE__ */ React13.createElement(
        "div",
        {
          style: {
            height: "24px",
            width: "24px",
            color: token.colorPrimary
          }
        },
        icon
      ),
      !collapsed && /* @__PURE__ */ React13.createElement(
        Typography2.Title,
        {
          style: {
            fontSize: "inherit",
            marginBottom: 0,
            fontWeight: 700
          }
        },
        text
      )
    )
  );
}, "ThemedTitle");

// src/components/buttons/create/index.tsx
import React14 from "react";
import { Button as Button3 } from "antd";
import { PlusSquareOutlined } from "@ant-design/icons";
import { useCreateButton } from "@refinedev/core";
import {
  RefineButtonClassNames,
  RefineButtonTestIds
} from "@refinedev/ui-types";
var CreateButton = /* @__PURE__ */ __name(({
  resource: resourceNameFromProps,
  hideText = false,
  accessControl,
  meta,
  children,
  onClick,
  ...rest
}) => {
  const { hidden, disabled, label, title, LinkComponent, to } = useCreateButton(
    {
      resource: resourceNameFromProps,
      meta,
      accessControl
    }
  );
  const isDisabled = disabled || rest.disabled;
  const isHidden = hidden || rest.hidden;
  if (isHidden)
    return null;
  return /* @__PURE__ */ React14.createElement(
    LinkComponent,
    {
      to,
      replace: false,
      onClick: (e) => {
        if (isDisabled) {
          e.preventDefault();
          return;
        }
        if (onClick) {
          e.preventDefault();
          onClick(e);
        }
      }
    },
    /* @__PURE__ */ React14.createElement(
      Button3,
      {
        icon: /* @__PURE__ */ React14.createElement(PlusSquareOutlined, null),
        disabled: isDisabled,
        title,
        
        className: RefineButtonClassNames.CreateButton,
        type: "primary",
        ...rest
      },
      !hideText && (children ?? label)
    )
  );
}, "CreateButton");

// src/components/buttons/edit/index.tsx
import React15 from "react";
import { Button as Button4 } from "antd";
import { EditOutlined } from "@ant-design/icons";
import { useEditButton } from "@refinedev/core";
import {
  RefineButtonClassNames as RefineButtonClassNames2,
  RefineButtonTestIds as RefineButtonTestIds2
} from "@refinedev/ui-types";
var EditButton = /* @__PURE__ */ __name(({
  resource: resourceNameFromProps,
  recordItemId,
  hideText = false,
  accessControl,
  meta,
  children,
  onClick,
  ...rest
}) => {
  const { to, label, title, hidden, disabled, LinkComponent } = useEditButton({
    resource: resourceNameFromProps,
    id: recordItemId,
    meta,
    accessControl
  });
  const isDisabled = disabled || rest.disabled;
  const isHidden = hidden || rest.hidden;
  if (isHidden)
    return null;
  return /* @__PURE__ */ React15.createElement(
    LinkComponent,
    {
      to,
      replace: false,
      onClick: (e) => {
        if (isDisabled) {
          e.preventDefault();
          return;
        }
        if (onClick) {
          e.preventDefault();
          onClick(e);
        }
      }
    },
    /* @__PURE__ */ React15.createElement(
      Button4,
      {
        icon: /* @__PURE__ */ React15.createElement(EditOutlined, null),
        disabled: isDisabled,
        title,
        
        className: RefineButtonClassNames2.EditButton,
        ...rest
      },
      !hideText && (children ?? label)
    )
  );
}, "EditButton");

// src/components/buttons/delete/index.tsx
import React16 from "react";
import { Button as Button5, Popconfirm } from "antd";
import { DeleteOutlined } from "@ant-design/icons";
import { useDeleteButton } from "@refinedev/core";
import {
  RefineButtonClassNames as RefineButtonClassNames3,
  RefineButtonTestIds as RefineButtonTestIds3
} from "@refinedev/ui-types";
var DeleteButton = /* @__PURE__ */ __name(({
  resource: resourceNameFromProps,
  recordItemId,
  onSuccess,
  mutationMode: mutationModeProp,
  children,
  successNotification,
  errorNotification,
  hideText = false,
  accessControl,
  meta,
  dataProviderName,
  confirmTitle,
  confirmOkText,
  confirmCancelText,
  invalidates,
  ...rest
}) => {
  const {
    title,
    label,
    hidden,
    disabled,
    loading,
    confirmTitle: defaultConfirmTitle,
    confirmOkLabel: defaultConfirmOkLabel,
    cancelLabel: defaultCancelLabel,
    onConfirm
  } = useDeleteButton({
    resource: resourceNameFromProps,
    id: recordItemId,
    dataProviderName,
    invalidates,
    meta,
    onSuccess,
    mutationMode: mutationModeProp,
    errorNotification,
    successNotification,
    accessControl
  });
  const isDisabled = disabled || rest.disabled;
  const isHidden = hidden || rest.hidden;
  if (isHidden)
    return null;
  return /* @__PURE__ */ React16.createElement(
    Popconfirm,
    {
      key: "delete",
      okText: confirmOkText ?? defaultConfirmOkLabel,
      cancelText: confirmCancelText ?? defaultCancelLabel,
      okType: "danger",
      title: confirmTitle ?? defaultConfirmTitle,
      okButtonProps: { disabled: loading },
      onConfirm,
      disabled: isDisabled
    },
    /* @__PURE__ */ React16.createElement(
      Button5,
      {
        danger: true,
        loading,
        icon: /* @__PURE__ */ React16.createElement(DeleteOutlined, null),
        title,
        disabled: isDisabled,
        
        className: RefineButtonClassNames3.DeleteButton,
        ...rest
      },
      !hideText && (children ?? label)
    )
  );
}, "DeleteButton");

// src/components/buttons/refresh/index.tsx
import React17 from "react";
import { Button as Button6 } from "antd";
import { RedoOutlined } from "@ant-design/icons";
import { useRefreshButton } from "@refinedev/core";
import {
  RefineButtonClassNames as RefineButtonClassNames4,
  RefineButtonTestIds as RefineButtonTestIds4
} from "@refinedev/ui-types";
var RefreshButton = /* @__PURE__ */ __name(({
  resource: resourceNameFromProps,
  recordItemId,
  hideText = false,
  dataProviderName,
  children,
  ...rest
}) => {
  const { onClick, label, loading } = useRefreshButton({
    resource: resourceNameFromProps,
    id: recordItemId,
    dataProviderName
  });
  return /* @__PURE__ */ React17.createElement(
    Button6,
    {
      onClick,
      icon: /* @__PURE__ */ React17.createElement(RedoOutlined, { spin: loading }),
      
      className: RefineButtonClassNames4.RefreshButton,
      ...rest
    },
    !hideText && (children ?? label)
  );
}, "RefreshButton");

// src/components/buttons/show/index.tsx
import React18 from "react";
import { Button as Button7 } from "antd";
import { EyeOutlined } from "@ant-design/icons";
import { useShowButton } from "@refinedev/core";
import {
  RefineButtonClassNames as RefineButtonClassNames5,
  RefineButtonTestIds as RefineButtonTestIds5
} from "@refinedev/ui-types";
var ShowButton = /* @__PURE__ */ __name(({
  resource: resourceNameFromProps,
  recordItemId,
  hideText = false,
  accessControl,
  meta,
  children,
  onClick,
  ...rest
}) => {
  const { to, label, title, hidden, disabled, LinkComponent } = useShowButton({
    resource: resourceNameFromProps,
    id: recordItemId,
    accessControl,
    meta
  });
  const isDisabled = disabled || rest.disabled;
  const isHidden = hidden || rest.hidden;
  if (isHidden)
    return null;
  return /* @__PURE__ */ React18.createElement(
    LinkComponent,
    {
      to,
      replace: false,
      onClick: (e) => {
        if (isDisabled) {
          e.preventDefault();
          return;
        }
        if (onClick) {
          e.preventDefault();
          onClick(e);
        }
      }
    },
    /* @__PURE__ */ React18.createElement(
      Button7,
      {
        disabled: isDisabled,
        icon: /* @__PURE__ */ React18.createElement(EyeOutlined, null),
        title,
        
        className: RefineButtonClassNames5.ShowButton,
        ...rest
      },
      !hideText && (children ?? label)
    )
  );
}, "ShowButton");

// src/components/buttons/list/index.tsx
import React19 from "react";
import { Button as Button8 } from "antd";
import { BarsOutlined as BarsOutlined2 } from "@ant-design/icons";
import { useListButton } from "@refinedev/core";
import {
  RefineButtonClassNames as RefineButtonClassNames6,
  RefineButtonTestIds as RefineButtonTestIds6
} from "@refinedev/ui-types";
var ListButton = /* @__PURE__ */ __name(({
  resource: resourceNameFromProps,
  hideText = false,
  accessControl,
  meta,
  children,
  onClick,
  ...rest
}) => {
  const { to, label, title, hidden, disabled, LinkComponent } = useListButton({
    resource: resourceNameFromProps,
    meta,
    accessControl
  });
  const isDisabled = disabled || rest.disabled;
  const isHidden = hidden || rest.hidden;
  if (isHidden)
    return null;
  return /* @__PURE__ */ React19.createElement(
    LinkComponent,
    {
      to,
      replace: false,
      onClick: (e) => {
        if (isDisabled) {
          e.preventDefault();
          return;
        }
        if (onClick) {
          e.preventDefault();
          onClick(e);
        }
      }
    },
    /* @__PURE__ */ React19.createElement(
      Button8,
      {
        icon: /* @__PURE__ */ React19.createElement(BarsOutlined2, null),
        disabled: isDisabled,
        title,
        
        className: RefineButtonClassNames6.ListButton,
        ...rest
      },
      !hideText && (children ?? label)
    )
  );
}, "ListButton");

// src/components/buttons/export/index.tsx
import React20 from "react";
import { Button as Button9 } from "antd";
import { ExportOutlined } from "@ant-design/icons";
import { useExportButton } from "@refinedev/core";
import {
  RefineButtonClassNames as RefineButtonClassNames7,
  RefineButtonTestIds as RefineButtonTestIds7
} from "@refinedev/ui-types";
var ExportButton = /* @__PURE__ */ __name(({
  hideText = false,
  children,
  ...rest
}) => {
  const { label } = useExportButton();
  return /* @__PURE__ */ React20.createElement(
    Button9,
    {
      type: "default",
      icon: /* @__PURE__ */ React20.createElement(ExportOutlined, null),
      
      className: RefineButtonClassNames7.ExportButton,
      ...rest
    },
    !hideText && (children ?? label)
  );
}, "ExportButton");

// src/components/buttons/save/index.tsx
import React21 from "react";
import { Button as Button10 } from "antd";
import { SaveOutlined } from "@ant-design/icons";
import { useSaveButton } from "@refinedev/core";
import {
  RefineButtonClassNames as RefineButtonClassNames8,
  RefineButtonTestIds as RefineButtonTestIds8
} from "@refinedev/ui-types";
var SaveButton = /* @__PURE__ */ __name(({
  hideText = false,
  children,
  ...rest
}) => {
  const { label } = useSaveButton();
  return /* @__PURE__ */ React21.createElement(
    Button10,
    {
      type: "primary",
      icon: /* @__PURE__ */ React21.createElement(SaveOutlined, null),
      
      className: RefineButtonClassNames8.SaveButton,
      ...rest
    },
    !hideText && (children ?? label)
  );
}, "SaveButton");

// src/components/buttons/clone/index.tsx
import React22 from "react";
import { Button as Button11 } from "antd";
import { PlusSquareOutlined as PlusSquareOutlined2 } from "@ant-design/icons";
import { useCloneButton } from "@refinedev/core";
import {
  RefineButtonTestIds as RefineButtonTestIds9,
  RefineButtonClassNames as RefineButtonClassNames9
} from "@refinedev/ui-types";
var CloneButton = /* @__PURE__ */ __name(({
  resource: resourceNameFromProps,
  recordItemId,
  hideText = false,
  accessControl,
  meta,
  children,
  onClick,
  ...rest
}) => {
  const { to, LinkComponent, label, disabled, hidden, title } = useCloneButton({
    id: recordItemId,
    resource: resourceNameFromProps,
    accessControl,
    meta
  });
  const isDisabled = disabled || rest.disabled;
  const isHidden = hidden || rest.hidden;
  if (isHidden)
    return null;
  return /* @__PURE__ */ React22.createElement(
    LinkComponent,
    {
      to,
      replace: false,
      onClick: (e) => {
        if (isDisabled) {
          e.preventDefault();
          return;
        }
        if (onClick) {
          e.preventDefault();
          onClick(e);
        }
      }
    },
    /* @__PURE__ */ React22.createElement(
      Button11,
      {
        icon: /* @__PURE__ */ React22.createElement(PlusSquareOutlined2, null),
        disabled: isDisabled,
        title,
        
        className: RefineButtonClassNames9.CloneButton,
        ...rest
      },
      !hideText && (children ?? label)
    )
  );
}, "CloneButton");

// src/components/buttons/import/index.tsx
import React23 from "react";
import { Button as Button12, Upload } from "antd";
import { ImportOutlined } from "@ant-design/icons";
import { useImportButton } from "@refinedev/core";
import {
  RefineButtonClassNames as RefineButtonClassNames10,
  RefineButtonTestIds as RefineButtonTestIds10
} from "@refinedev/ui-types";
var ImportButton = /* @__PURE__ */ __name(({
  uploadProps,
  buttonProps,
  hideText = false,
  children,
  loading,
  ...rest
}) => {
  const { label } = useImportButton();
  return /* @__PURE__ */ React23.createElement(Upload, { ...uploadProps }, /* @__PURE__ */ React23.createElement(
    Button12,
    {
      icon: /* @__PURE__ */ React23.createElement(ImportOutlined, null),
      
      className: RefineButtonClassNames10.ImportButton,
      loading,
      ...buttonProps,
      ...rest
    },
    !hideText && (children ?? label)
  ));
}, "ImportButton");

// src/components/crud/list/index.tsx
import React24 from "react";
import { Space as Space3 } from "antd";
import {
  useTranslate as useTranslate6,
  useUserFriendlyName as useUserFriendlyName2,
  useRefineContext as useRefineContext2,
  useResourceParams as useResourceParams4
} from "@refinedev/core";
var List = /* @__PURE__ */ __name(({
  canCreate,
  title,
  children,
  createButtonProps: createButtonPropsFromProps,
  resource: resourceFromProps,
  wrapperProps,
  contentProps,
  headerProps,
  breadcrumb: breadcrumbFromProps,
  headerButtonProps,
  headerButtons
}) => {
  var _a;
  const translate = useTranslate6();
  const {
    options: { breadcrumb: globalBreadcrumb } = {}
  } = useRefineContext2();
  const getUserFriendlyName = useUserFriendlyName2();
  const { resource, identifier } = useResourceParams4({
    resource: resourceFromProps
  });
  const isCreateButtonVisible = canCreate ?? (!!(resource == null ? void 0 : resource.create) || !!createButtonPropsFromProps);
  const breadcrumb = typeof breadcrumbFromProps === "undefined" ? globalBreadcrumb : breadcrumbFromProps;
  const createButtonProps = isCreateButtonVisible ? {
    size: "middle",
    resource: identifier,
    ...createButtonPropsFromProps
  } : void 0;
  const defaultExtra = isCreateButtonVisible ? /* @__PURE__ */ React24.createElement(CreateButton, { ...createButtonProps }) : null;
  return /* @__PURE__ */ React24.createElement("div", { ...wrapperProps ?? {} }, /* @__PURE__ */ React24.createElement(
    PageHeader,
    {
      title: title ?? translate(
        `${identifier}.titles.list`,
        getUserFriendlyName(((_a = resource == null ? void 0 : resource.meta) == null ? void 0 : _a.label) ?? identifier, "plural")
      ),
      extra: headerButtons ? /* @__PURE__ */ React24.createElement(Space3, { wrap: true, ...headerButtonProps }, typeof headerButtons === "function" ? headerButtons({
        defaultButtons: defaultExtra,
        createButtonProps
      }) : headerButtons) : defaultExtra,
      breadcrumb: typeof breadcrumb !== "undefined" ? /* @__PURE__ */ React24.createElement(React24.Fragment, null, breadcrumb) : /* @__PURE__ */ React24.createElement(Breadcrumb, null),
      ...headerProps ?? {}
    },
    /* @__PURE__ */ React24.createElement("div", { ...contentProps ?? {} }, children)
  ));
}, "List");

// src/components/crud/create/index.tsx
import React25 from "react";
import { Card, Space as Space4, Spin } from "antd";
import {
  useTranslate as useTranslate7,
  useUserFriendlyName as useUserFriendlyName3,
  useRefineContext as useRefineContext3,
  useResourceParams as useResourceParams5,
  useBack
} from "@refinedev/core";
var Create = /* @__PURE__ */ __name(({
  title,
  saveButtonProps: saveButtonPropsFromProps,
  children,
  resource: resourceFromProps,
  isLoading = false,
  breadcrumb: breadcrumbFromProps,
  wrapperProps,
  headerProps,
  contentProps,
  headerButtonProps,
  headerButtons,
  footerButtonProps,
  footerButtons,
  goBack: goBackFromProps
}) => {
  var _a;
  const translate = useTranslate7();
  const {
    options: { breadcrumb: globalBreadcrumb } = {}
  } = useRefineContext3();
  const back = useBack();
  const getUserFriendlyName = useUserFriendlyName3();
  const { resource, identifier } = useResourceParams5({
    resource: resourceFromProps
  });
  const breadcrumb = typeof breadcrumbFromProps === "undefined" ? globalBreadcrumb : breadcrumbFromProps;
  const saveButtonProps = {
    ...isLoading ? { disabled: true } : {},
    ...saveButtonPropsFromProps,
    htmlType: "submit"
  };
  const defaultFooterButtons = /* @__PURE__ */ React25.createElement(React25.Fragment, null, /* @__PURE__ */ React25.createElement(SaveButton, { ...saveButtonProps }));
  return /* @__PURE__ */ React25.createElement("div", { ...wrapperProps ?? {} }, /* @__PURE__ */ React25.createElement(
    PageHeader,
    {
      backIcon: goBackFromProps,
      onBack: back,
      title: title ?? translate(
        `${identifier}.titles.create`,
        `Create ${getUserFriendlyName(
          ((_a = resource == null ? void 0 : resource.meta) == null ? void 0 : _a.label) ?? identifier,
          "singular"
        )}`
      ),
      breadcrumb: typeof breadcrumb !== "undefined" ? /* @__PURE__ */ React25.createElement(React25.Fragment, null, breadcrumb) : /* @__PURE__ */ React25.createElement(Breadcrumb, null),
      extra: /* @__PURE__ */ React25.createElement(Space4, { wrap: true, ...headerButtonProps ?? {} }, headerButtons ? typeof headerButtons === "function" ? headerButtons({
        defaultButtons: null
      }) : headerButtons : null),
      ...headerProps ?? {}
    },
    /* @__PURE__ */ React25.createElement(Spin, { spinning: isLoading }, /* @__PURE__ */ React25.createElement(
      Card,
      {
        variant: "borderless",
        actions: [
          /* @__PURE__ */ React25.createElement(
            Space4,
            {
              key: "action-buttons",
              style: { float: "right", marginRight: 24 },
              ...footerButtonProps ?? {}
            },
            footerButtons ? typeof footerButtons === "function" ? footerButtons({
              defaultButtons: defaultFooterButtons,
              saveButtonProps
            }) : footerButtons : defaultFooterButtons
          )
        ],
        ...contentProps ?? {}
      },
      children
    ))
  ));
}, "Create");

// src/components/crud/edit/index.tsx
import React26 from "react";
import { Card as Card2, Space as Space5, Spin as Spin2 } from "antd";
import {
  useMutationMode,
  useTranslate as useTranslate8,
  useUserFriendlyName as useUserFriendlyName4,
  useRefineContext as useRefineContext4,
  useBack as useBack2,
  useResourceParams as useResourceParams6,
  useGo as useGo3,
  useToPath
} from "@refinedev/core";
var Edit = /* @__PURE__ */ __name(({
  title,
  saveButtonProps: saveButtonPropsFromProps,
  mutationMode: mutationModeProp,
  recordItemId,
  children,
  deleteButtonProps: deleteButtonPropsFromProps,
  canDelete,
  resource: resourceFromProps,
  isLoading = false,
  dataProviderName,
  breadcrumb: breadcrumbFromProps,
  wrapperProps,
  headerProps,
  contentProps,
  headerButtonProps,
  headerButtons,
  footerButtonProps,
  footerButtons,
  goBack: goBackFromProps,
  autoSaveProps
}) => {
  var _a, _b;
  const translate = useTranslate8();
  const {
    options: { breadcrumb: globalBreadcrumb } = {}
  } = useRefineContext4();
  const { mutationMode: mutationModeContext } = useMutationMode();
  const mutationMode = mutationModeProp ?? mutationModeContext;
  const back = useBack2();
  const go = useGo3();
  const getUserFriendlyName = useUserFriendlyName4();
  const {
    resource,
    action,
    id: idFromParams,
    identifier
  } = useResourceParams6({
    resource: resourceFromProps
  });
  const goListPath = useToPath({
    resource,
    action: "list"
  });
  const id = recordItemId ?? idFromParams;
  const breadcrumb = typeof breadcrumbFromProps === "undefined" ? globalBreadcrumb : breadcrumbFromProps;
  const hasList = (resource == null ? void 0 : resource.list) && !recordItemId;
  const isDeleteButtonVisible = canDelete ?? (((_a = resource == null ? void 0 : resource.meta) == null ? void 0 : _a.canDelete) || deleteButtonPropsFromProps);
  const listButtonProps = hasList ? {
    ...isLoading ? { disabled: true } : {},
    resource: identifier
  } : void 0;
  const refreshButtonProps = {
    ...isLoading ? { disabled: true } : {},
    resource: identifier,
    recordItemId: id,
    dataProviderName
  };
  const deleteButtonProps = isDeleteButtonVisible ? {
    ...isLoading ? { disabled: true } : {},
    resource: identifier,
    mutationMode,
    onSuccess: () => {
      go({ to: goListPath });
    },
    recordItemId: id,
    dataProviderName,
    ...deleteButtonPropsFromProps
  } : void 0;
  const saveButtonProps = {
    ...isLoading ? { disabled: true } : {},
    ...saveButtonPropsFromProps
  };
  const defaultHeaderButtons = /* @__PURE__ */ React26.createElement(React26.Fragment, null, autoSaveProps && /* @__PURE__ */ React26.createElement(AutoSaveIndicator, { ...autoSaveProps }), hasList && /* @__PURE__ */ React26.createElement(ListButton, { ...listButtonProps }), /* @__PURE__ */ React26.createElement(RefreshButton, { ...refreshButtonProps }));
  const defaultFooterButtons = /* @__PURE__ */ React26.createElement(React26.Fragment, null, isDeleteButtonVisible && /* @__PURE__ */ React26.createElement(DeleteButton, { ...deleteButtonProps }), /* @__PURE__ */ React26.createElement(SaveButton, { ...saveButtonProps }));
  return /* @__PURE__ */ React26.createElement("div", { ...wrapperProps ?? {} }, /* @__PURE__ */ React26.createElement(
    PageHeader,
    {
      backIcon: goBackFromProps,
      onBack: action !== "list" && typeof action !== "undefined" ? back : void 0,
      title: title ?? translate(
        `${identifier}.titles.edit`,
        `Edit ${getUserFriendlyName(
          ((_b = resource == null ? void 0 : resource.meta) == null ? void 0 : _b.label) ?? identifier,
          "singular"
        )}`
      ),
      extra: /* @__PURE__ */ React26.createElement(Space5, { wrap: true, ...headerButtonProps ?? {} }, headerButtons ? typeof headerButtons === "function" ? headerButtons({
        defaultButtons: defaultHeaderButtons,
        listButtonProps,
        refreshButtonProps
      }) : headerButtons : defaultHeaderButtons),
      breadcrumb: typeof breadcrumb !== "undefined" ? /* @__PURE__ */ React26.createElement(React26.Fragment, null, breadcrumb) : /* @__PURE__ */ React26.createElement(Breadcrumb, null),
      ...headerProps ?? {}
    },
    /* @__PURE__ */ React26.createElement(Spin2, { spinning: isLoading }, /* @__PURE__ */ React26.createElement(
      Card2,
      {
        variant: "borderless",
        actions: [
          /* @__PURE__ */ React26.createElement(
            Space5,
            {
              key: "footer-buttons",
              wrap: true,
              style: {
                float: "right",
                marginRight: 24
              },
              ...footerButtonProps ?? {}
            },
            footerButtons ? typeof footerButtons === "function" ? footerButtons({
              defaultButtons: defaultFooterButtons,
              deleteButtonProps,
              saveButtonProps
            }) : footerButtons : defaultFooterButtons
          )
        ],
        ...contentProps ?? {}
      },
      children
    ))
  ));
}, "Edit");

// src/components/crud/show/index.tsx
import React27 from "react";
import { Card as Card3, Space as Space6, Spin as Spin3 } from "antd";
import {
  useTranslate as useTranslate9,
  useUserFriendlyName as useUserFriendlyName5,
  useRefineContext as useRefineContext5,
  useResourceParams as useResourceParams7,
  useToPath as useToPath2,
  useBack as useBack3,
  useGo as useGo4
} from "@refinedev/core";
var Show = /* @__PURE__ */ __name(({
  title,
  canEdit,
  canDelete,
  deleteButtonProps: deleteButtonPropsFromProps,
  isLoading = false,
  children,
  resource: resourceFromProps,
  recordItemId,
  dataProviderName,
  breadcrumb: breadcrumbFromProps,
  contentProps,
  headerProps,
  wrapperProps,
  headerButtons,
  footerButtons,
  footerButtonProps,
  headerButtonProps,
  goBack: goBackFromProps
}) => {
  var _a, _b;
  const translate = useTranslate9();
  const {
    options: { breadcrumb: globalBreadcrumb } = {}
  } = useRefineContext5();
  const back = useBack3();
  const go = useGo4();
  const getUserFriendlyName = useUserFriendlyName5();
  const {
    resource,
    action,
    id: idFromParams,
    identifier
  } = useResourceParams7({
    resource: resourceFromProps
  });
  const goListPath = useToPath2({
    resource,
    action: "list"
  });
  const id = recordItemId ?? idFromParams;
  const breadcrumb = typeof breadcrumbFromProps === "undefined" ? globalBreadcrumb : breadcrumbFromProps;
  const hasList = (resource == null ? void 0 : resource.list) && !recordItemId;
  const isDeleteButtonVisible = canDelete ?? (((_a = resource == null ? void 0 : resource.meta) == null ? void 0 : _a.canDelete) || deleteButtonPropsFromProps);
  const isEditButtonVisible = canEdit ?? !!(resource == null ? void 0 : resource.edit);
  const listButtonProps = hasList ? {
    resource: identifier
  } : void 0;
  const editButtonProps = isEditButtonVisible ? {
    ...isLoading ? { disabled: true } : {},
    type: "primary",
    resource: identifier,
    recordItemId: id
  } : void 0;
  const deleteButtonProps = isDeleteButtonVisible ? {
    ...isLoading ? { disabled: true } : {},
    resource: identifier,
    recordItemId: id,
    onSuccess: () => {
      go({ to: goListPath });
    },
    dataProviderName,
    ...deleteButtonPropsFromProps
  } : void 0;
  const refreshButtonProps = {
    ...isLoading ? { disabled: true } : {},
    resource: identifier,
    recordItemId: id,
    dataProviderName
  };
  const defaultHeaderButtons = /* @__PURE__ */ React27.createElement(React27.Fragment, null, hasList && /* @__PURE__ */ React27.createElement(ListButton, { ...listButtonProps }), isEditButtonVisible && /* @__PURE__ */ React27.createElement(EditButton, { ...editButtonProps }), isDeleteButtonVisible && /* @__PURE__ */ React27.createElement(DeleteButton, { ...deleteButtonProps }), /* @__PURE__ */ React27.createElement(RefreshButton, { ...refreshButtonProps }));
  return /* @__PURE__ */ React27.createElement("div", { ...wrapperProps ?? {} }, /* @__PURE__ */ React27.createElement(
    PageHeader,
    {
      backIcon: goBackFromProps,
      onBack: action !== "list" && typeof action !== "undefined" ? back : void 0,
      title: title ?? translate(
        `${identifier}.titles.show`,
        `Show ${getUserFriendlyName(
          ((_b = resource == null ? void 0 : resource.meta) == null ? void 0 : _b.label) ?? identifier,
          "singular"
        )}`
      ),
      extra: /* @__PURE__ */ React27.createElement(Space6, { key: "extra-buttons", wrap: true, ...headerButtonProps ?? {} }, headerButtons ? typeof headerButtons === "function" ? headerButtons({
        defaultButtons: defaultHeaderButtons,
        deleteButtonProps,
        editButtonProps,
        listButtonProps,
        refreshButtonProps
      }) : headerButtons : defaultHeaderButtons),
      breadcrumb: typeof breadcrumb !== "undefined" ? /* @__PURE__ */ React27.createElement(React27.Fragment, null, breadcrumb) : /* @__PURE__ */ React27.createElement(Breadcrumb, null),
      ...headerProps ?? {}
    },
    /* @__PURE__ */ React27.createElement(Spin3, { spinning: isLoading }, /* @__PURE__ */ React27.createElement(
      Card3,
      {
        variant: "borderless",
        actions: footerButtons ? [
          /* @__PURE__ */ React27.createElement(Space6, { key: "footer-buttons", wrap: true, ...footerButtonProps }, typeof footerButtons === "function" ? footerButtons({
            defaultButtons: null
          }) : footerButtons)
        ] : void 0,
        ...contentProps ?? {}
      },
      children
    ))
  ));
}, "Show");

// src/components/fields/text/index.tsx
import React28 from "react";
import { Typography as Typography3 } from "antd";
var TextField = /* @__PURE__ */ __name(({ value, ...rest }) => {
  return /* @__PURE__ */ React28.createElement(Typography3.Text, { ...rest }, value);
}, "TextField");

// src/components/fields/tag/index.tsx
import React29 from "react";
import { Tag } from "antd";
var TagField = /* @__PURE__ */ __name(({ value, ...rest }) => {
  return /* @__PURE__ */ React29.createElement(Tag, { ...rest }, value == null ? void 0 : value.toString());
}, "TagField");

// src/components/fields/email/index.tsx
import React30 from "react";
import { Typography as Typography4 } from "antd";
var EmailField = /* @__PURE__ */ __name(({ value, ...rest }) => {
  return /* @__PURE__ */ React30.createElement(Typography4.Link, { href: `mailto:${value}`, ...rest }, value);
}, "EmailField");

// src/components/fields/image/index.tsx
import React31 from "react";
import { Image } from "antd";
var ImageField = /* @__PURE__ */ __name(({
  value,
  imageTitle,
  ...rest
}) => {
  return /* @__PURE__ */ React31.createElement(Image, { ...rest, src: value, title: imageTitle });
}, "ImageField");

// src/components/fields/boolean/index.tsx
import React32 from "react";
import { Tooltip } from "antd";
import { CheckOutlined, CloseOutlined } from "@ant-design/icons";
var BooleanField = /* @__PURE__ */ __name(({
  value,
  valueLabelTrue = "true",
  valueLabelFalse = "false",
  trueIcon = /* @__PURE__ */ React32.createElement(CheckOutlined, null),
  falseIcon = /* @__PURE__ */ React32.createElement(CloseOutlined, null),
  ...rest
}) => {
  return /* @__PURE__ */ React32.createElement(Tooltip, { title: value ? valueLabelTrue : valueLabelFalse, ...rest }, value ? /* @__PURE__ */ React32.createElement("span", null, trueIcon) : /* @__PURE__ */ React32.createElement("span", null, falseIcon));
}, "BooleanField");

// src/components/fields/date/index.tsx
import React33 from "react";
import dayjs from "dayjs";
import { Typography as Typography5 } from "antd";
import LocalizedFormat from "dayjs/plugin/localizedFormat.js";
var DateField = /* @__PURE__ */ __name(({
  value,
  locales,
  format: dateFormat = "L",
  ...rest
}) => {
  dayjs.extend(LocalizedFormat);
  const defaultLocale = dayjs.locale();
  return /* @__PURE__ */ React33.createElement(Typography5.Text, { ...rest }, value ? dayjs(value).locale(locales || defaultLocale).format(dateFormat) : "");
}, "DateField");

// src/components/fields/file/index.tsx
import React34 from "react";
var FileField = /* @__PURE__ */ __name(({
  title,
  src,
  ...rest
}) => {
  return /* @__PURE__ */ React34.createElement(UrlField, { value: src, title, ...rest }, title ?? src);
}, "FileField");

// src/components/fields/url/index.tsx
import React35 from "react";
import { Typography as Typography6 } from "antd";
var UrlField = /* @__PURE__ */ __name(({
  children,
  value,
  ...rest
}) => {
  return /* @__PURE__ */ React35.createElement(Typography6.Link, { href: value, ...rest }, children ?? value);
}, "UrlField");

// src/components/fields/number/index.tsx
import React36 from "react";
import { Typography as Typography7 } from "antd";
function toLocaleStringSupportsOptions() {
  return !!(typeof Intl === "object" && Intl && typeof Intl.NumberFormat === "function");
}
__name(toLocaleStringSupportsOptions, "toLocaleStringSupportsOptions");
var NumberField = /* @__PURE__ */ __name(({
  value,
  locale,
  options,
  ...rest
}) => {
  const number = Number(value);
  return /* @__PURE__ */ React36.createElement(Typography7.Text, { ...rest }, toLocaleStringSupportsOptions() ? number.toLocaleString(locale, options) : number);
}, "NumberField");

// src/components/fields/markdown/index.tsx
import React37 from "react";
import ReactMarkdown from "react-markdown";
import gfm from "remark-gfm";
var MarkdownField = /* @__PURE__ */ __name(({
  value = ""
}) => {
  return /* @__PURE__ */ React37.createElement(
    ReactMarkdown,
    {
      remarkPlugins: [gfm]
    },
    value
  );
}, "MarkdownField");

// src/components/table/components/filterDropdown/index.tsx
import React38 from "react";
import { Button as Button13, Space as Space7 } from "antd";
import dayjs2 from "dayjs";
import { FilterOutlined } from "@ant-design/icons";
import { useTranslate as useTranslate10 } from "@refinedev/core";
var FilterDropdown = /* @__PURE__ */ __name((props) => {
  const {
    setSelectedKeys,
    confirm,
    clearFilters,
    mapValue = /* @__PURE__ */ __name((value) => value, "mapValue"),
    selectedKeys,
    children
  } = props;
  const translate = useTranslate10();
  const clearFilter = /* @__PURE__ */ __name(() => {
    if (clearFilters) {
      clearFilters();
    }
  }, "clearFilter");
  const onFilter = /* @__PURE__ */ __name(() => {
    let keys;
    if (typeof selectedKeys === "number") {
      keys = `${selectedKeys}`;
    } else if (dayjs2.isDayjs(selectedKeys)) {
      keys = [selectedKeys.toISOString()];
    } else {
      keys = selectedKeys;
    }
    setSelectedKeys(keys);
    confirm == null ? void 0 : confirm();
  }, "onFilter");
  const onChange = /* @__PURE__ */ __name((e) => {
    if (typeof e === "object") {
      if (Array.isArray(e)) {
        const mappedValue3 = mapValue(e, "onChange");
        return setSelectedKeys(mappedValue3);
      }
      const changeEvent = !e || !e.target || dayjs2.isDayjs(e) ? { target: { value: e } } : e;
      const { target } = changeEvent;
      const mappedValue2 = mapValue(target.value, "onChange");
      setSelectedKeys(mappedValue2);
      return;
    }
    const mappedValue = mapValue(e, "onChange");
    setSelectedKeys(mappedValue);
  }, "onChange");
  const childrenWithProps = React38.Children.map(children, (child) => {
    if (React38.isValidElement(child)) {
      return React38.cloneElement(child, {
        onChange,
        value: mapValue(selectedKeys, "value")
      });
    }
    return child;
  });
  return /* @__PURE__ */ React38.createElement(
    "div",
    {
      style: {
        padding: 10,
        display: "flex",
        flexDirection: "column",
        alignItems: "flex-end"
      }
    },
    /* @__PURE__ */ React38.createElement("div", { style: { marginBottom: 15 } }, childrenWithProps),
    /* @__PURE__ */ React38.createElement(Space7, null, /* @__PURE__ */ React38.createElement(Button13, { type: "primary", size: "small", onClick: () => onFilter() }, /* @__PURE__ */ React38.createElement(FilterOutlined, null), " ", translate("buttons.filter", "Filter")), /* @__PURE__ */ React38.createElement(Button13, { danger: true, size: "small", onClick: () => clearFilter() }, translate("buttons.clear", "Clear")))
  );
}, "FilterDropdown");

// src/components/pages/error/index.tsx
import React39, { useEffect, useState as useState3 } from "react";
import { useGo as useGo5, useResourceParams as useResourceParams8 } from "@refinedev/core";
import { Button as Button14, Result, Typography as Typography8, Space as Space8, Tooltip as Tooltip2 } from "antd";
import { InfoCircleOutlined } from "@ant-design/icons";
import { useTranslate as useTranslate11 } from "@refinedev/core";
var ErrorComponent = /* @__PURE__ */ __name(() => {
  const [errorMessage, setErrorMessage] = useState3();
  const translate = useTranslate11();
  const go = useGo5();
  const { resource, action } = useResourceParams8();
  useEffect(() => {
    if (resource) {
      if (action) {
        setErrorMessage(
          translate(
            "pages.error.info",
            {
              action,
              resource: resource == null ? void 0 : resource.name
            },
            `You may have forgotten to add the "${action}" component to "${resource == null ? void 0 : resource.name}" resource.`
          )
        );
      }
    }
  }, [resource, action]);
  return /* @__PURE__ */ React39.createElement(
    Result,
    {
      status: "404",
      title: "404",
      extra: /* @__PURE__ */ React39.createElement(Space8, { direction: "vertical", size: "large" }, /* @__PURE__ */ React39.createElement(Space8, null, /* @__PURE__ */ React39.createElement(Typography8.Text, null, translate(
        "pages.error.404",
        "Sorry, the page you visited does not exist."
      )), errorMessage && /* @__PURE__ */ React39.createElement(Tooltip2, { title: errorMessage }, /* @__PURE__ */ React39.createElement(InfoCircleOutlined, { }))), /* @__PURE__ */ React39.createElement(Button14, { type: "primary", onClick: () => go({ to: "/" }) }, translate("pages.error.backHome", "Back Home")))
    }
  );
}, "ErrorComponent");

// src/components/pages/welcome/index.tsx
import React40 from "react";
import { WelcomePage as WelcomePageFromCore } from "@refinedev/core";
var WelcomePage = /* @__PURE__ */ __name(() => {
  return /* @__PURE__ */ React40.createElement(WelcomePageFromCore, null);
}, "WelcomePage");

// src/components/pages/auth/index.tsx
import React45 from "react";

// src/components/pages/auth/components/login/index.tsx
import React41 from "react";
import {
  useLink as useLink4
} from "@refinedev/core";
import {
  Row,
  Col,
  Layout as Layout2,
  Card as Card4,
  Typography as Typography9,
  Form as Form4,
  Input,
  Button as Button15,
  Checkbox,
  Divider,
  theme as theme4
} from "antd";
import { useLogin, useTranslate as useTranslate12 } from "@refinedev/core";

// src/components/pages/auth/components/styles.ts
var layoutStyles = {};
var containerStyles = {
  maxWidth: "400px",
  margin: "auto",
  padding: "32px",
  boxShadow: "0px 2px 4px rgba(0, 0, 0, 0.02), 0px 1px 6px -1px rgba(0, 0, 0, 0.02), 0px 1px 2px rgba(0, 0, 0, 0.03)"
};
var headStyles = {
  borderBottom: 0,
  padding: 0
};
var bodyStyles = { padding: 0, marginTop: "32px" };
var titleStyles = {
  textAlign: "center",
  marginBottom: 0,
  fontSize: "24px",
  lineHeight: "32px",
  fontWeight: 700,
  overflowWrap: "break-word",
  hyphens: "manual",
  textOverflow: "unset",
  whiteSpace: "pre-wrap"
};

// src/components/pages/auth/components/login/index.tsx
var LoginPage = /* @__PURE__ */ __name(({
  providers,
  registerLink,
  forgotPasswordLink,
  rememberMe,
  contentProps,
  wrapperProps,
  renderContent,
  formProps,
  title,
  hideForm,
  mutationVariables
}) => {
  const { token } = theme4.useToken();
  const [form] = Form4.useForm();
  const translate = useTranslate12();
  const Link = useLink4();
  const { mutate: login, isPending } = useLogin();
  const PageTitle = title === false ? null : /* @__PURE__ */ React41.createElement(
    "div",
    {
      style: {
        display: "flex",
        justifyContent: "center",
        marginBottom: "32px",
        fontSize: "20px"
      }
    },
    title ?? /* @__PURE__ */ React41.createElement(ThemedTitle, { collapsed: false })
  );
  const CardTitle = /* @__PURE__ */ React41.createElement(
    Typography9.Title,
    {
      level: 3,
      style: {
        color: token.colorPrimaryTextHover,
        ...titleStyles
      }
    },
    translate("pages.login.title", "Sign in to your account")
  );
  const renderProviders = /* @__PURE__ */ __name(() => {
    if (providers && providers.length > 0) {
      return /* @__PURE__ */ React41.createElement(React41.Fragment, null, providers.map((provider) => {
        return /* @__PURE__ */ React41.createElement(
          Button15,
          {
            key: provider.name,
            type: "default",
            block: true,
            icon: provider.icon,
            style: {
              display: "flex",
              justifyContent: "center",
              alignItems: "center",
              width: "100%",
              marginBottom: "8px"
            },
            onClick: () => login({
              ...mutationVariables,
              providerName: provider.name
            })
          },
          provider.label
        );
      }), !hideForm && /* @__PURE__ */ React41.createElement(Divider, null, /* @__PURE__ */ React41.createElement(
        Typography9.Text,
        {
          style: {
            color: token.colorTextLabel
          }
        },
        translate("pages.login.divider", "or")
      )));
    }
    return null;
  }, "renderProviders");
  const CardContent = /* @__PURE__ */ React41.createElement(
    Card4,
    {
      title: CardTitle,
      styles: {
        header: headStyles,
        body: bodyStyles
      },
      style: {
        ...containerStyles,
        backgroundColor: token.colorBgElevated
      },
      ...contentProps ?? {}
    },
    renderProviders(),
    !hideForm && /* @__PURE__ */ React41.createElement(
      Form4,
      {
        layout: "vertical",
        form,
        onFinish: (values) => login({ ...values, ...mutationVariables }),
        requiredMark: false,
        initialValues: {
          remember: false
        },
        ...formProps
      },
      /* @__PURE__ */ React41.createElement(
        Form4.Item,
        {
          name: "email",
          label: translate("pages.login.fields.email", "Email"),
          rules: [
            {
              required: true,
              message: translate(
                "pages.login.errors.requiredEmail",
                "Email is required"
              )
            },
            {
              type: "email",
              message: translate(
                "pages.login.errors.validEmail",
                "Invalid email address"
              )
            }
          ]
        },
        /* @__PURE__ */ React41.createElement(
          Input,
          {
            size: "large",
            placeholder: translate("pages.login.fields.email", "Email")
          }
        )
      ),
      /* @__PURE__ */ React41.createElement(
        Form4.Item,
        {
          name: "password",
          label: translate("pages.login.fields.password", "Password"),
          rules: [
            {
              required: true,
              message: translate(
                "pages.login.errors.requiredPassword",
                "Password is required"
              )
            }
          ]
        },
        /* @__PURE__ */ React41.createElement(
          Input,
          {
            type: "password",
            autoComplete: "current-password",
            placeholder: "\u25CF\u25CF\u25CF\u25CF\u25CF\u25CF\u25CF\u25CF",
            size: "large"
          }
        )
      ),
      /* @__PURE__ */ React41.createElement(
        "div",
        {
          style: {
            display: "flex",
            justifyContent: "space-between",
            marginBottom: "24px"
          }
        },
        rememberMe ?? /* @__PURE__ */ React41.createElement(Form4.Item, { name: "remember", valuePropName: "checked", noStyle: true }, /* @__PURE__ */ React41.createElement(
          Checkbox,
          {
            style: {
              fontSize: "12px"
            }
          },
          translate("pages.login.buttons.rememberMe", "Remember me")
        )),
        forgotPasswordLink ?? /* @__PURE__ */ React41.createElement(
          Link,
          {
            style: {
              color: token.colorPrimaryTextHover,
              fontSize: "12px",
              marginLeft: "auto"
            },
            to: "/forgot-password"
          },
          translate(
            "pages.login.buttons.forgotPassword",
            "Forgot password?"
          )
        )
      ),
      !hideForm && /* @__PURE__ */ React41.createElement(Form4.Item, null, /* @__PURE__ */ React41.createElement(
        Button15,
        {
          type: "primary",
          size: "large",
          htmlType: "submit",
          loading: isPending,
          block: true
        },
        translate("pages.login.signin", "Sign in")
      ))
    ),
    registerLink ?? /* @__PURE__ */ React41.createElement(
      "div",
      {
        style: {
          marginTop: hideForm ? 16 : 8
        }
      },
      /* @__PURE__ */ React41.createElement(Typography9.Text, { style: { fontSize: 12 } }, translate(
        "pages.login.buttons.noAccount",
        "Don\u2019t have an account?"
      ), " ", /* @__PURE__ */ React41.createElement(
        Link,
        {
          to: "/register",
          style: {
            fontWeight: "bold",
            color: token.colorPrimaryTextHover
          }
        },
        translate("pages.login.signup", "Sign up")
      ))
    )
  );
  return /* @__PURE__ */ React41.createElement(Layout2, { style: layoutStyles, ...wrapperProps ?? {} }, /* @__PURE__ */ React41.createElement(
    Row,
    {
      justify: "center",
      align: hideForm ? "top" : "middle",
      style: {
        padding: "16px 0",
        minHeight: "100dvh",
        paddingTop: hideForm ? "15dvh" : "16px"
      }
    },
    /* @__PURE__ */ React41.createElement(Col, { xs: 22 }, renderContent ? renderContent(CardContent, PageTitle) : /* @__PURE__ */ React41.createElement(React41.Fragment, null, PageTitle, CardContent))
  ));
}, "LoginPage");

// src/components/pages/auth/components/register/index.tsx
import React42 from "react";
import {
  useLink as useLink5
} from "@refinedev/core";
import {
  Row as Row2,
  Col as Col2,
  Layout as Layout3,
  Card as Card5,
  Typography as Typography10,
  Form as Form5,
  Input as Input2,
  Button as Button16,
  Divider as Divider2,
  theme as theme5
} from "antd";
import { useTranslate as useTranslate13, useRegister } from "@refinedev/core";
var RegisterPage = /* @__PURE__ */ __name(({
  providers,
  loginLink,
  wrapperProps,
  contentProps,
  renderContent,
  formProps,
  title,
  hideForm,
  mutationVariables
}) => {
  const { token } = theme5.useToken();
  const [form] = Form5.useForm();
  const translate = useTranslate13();
  const Link = useLink5();
  const { mutate: register, isPending } = useRegister();
  const PageTitle = title === false ? null : /* @__PURE__ */ React42.createElement(
    "div",
    {
      style: {
        display: "flex",
        justifyContent: "center",
        marginBottom: "32px",
        fontSize: "20px"
      }
    },
    title ?? /* @__PURE__ */ React42.createElement(ThemedTitle, { collapsed: false })
  );
  const CardTitle = /* @__PURE__ */ React42.createElement(
    Typography10.Title,
    {
      level: 3,
      style: {
        color: token.colorPrimaryTextHover,
        ...titleStyles
      }
    },
    translate("pages.register.title", "Sign up for your account")
  );
  const renderProviders = /* @__PURE__ */ __name(() => {
    if (providers && providers.length > 0) {
      return /* @__PURE__ */ React42.createElement(React42.Fragment, null, providers.map((provider) => {
        return /* @__PURE__ */ React42.createElement(
          Button16,
          {
            key: provider.name,
            type: "default",
            block: true,
            icon: provider.icon,
            style: {
              display: "flex",
              justifyContent: "center",
              alignItems: "center",
              width: "100%",
              marginBottom: "8px"
            },
            onClick: () => register({
              ...mutationVariables,
              providerName: provider.name
            })
          },
          provider.label
        );
      }), !hideForm && /* @__PURE__ */ React42.createElement(Divider2, null, /* @__PURE__ */ React42.createElement(
        Typography10.Text,
        {
          style: {
            color: token.colorTextLabel
          }
        },
        translate(
          "pages.register.divider",
          translate("pages.login.divider", "or")
        )
      )));
    }
    return null;
  }, "renderProviders");
  const CardContent = /* @__PURE__ */ React42.createElement(
    Card5,
    {
      title: CardTitle,
      styles: {
        header: headStyles,
        body: bodyStyles
      },
      style: {
        ...containerStyles,
        backgroundColor: token.colorBgElevated
      },
      ...contentProps ?? {}
    },
    renderProviders(),
    !hideForm && /* @__PURE__ */ React42.createElement(
      Form5,
      {
        layout: "vertical",
        form,
        onFinish: (values) => register({ ...mutationVariables, ...values }),
        requiredMark: false,
        ...formProps
      },
      /* @__PURE__ */ React42.createElement(
        Form5.Item,
        {
          name: "email",
          label: translate("pages.register.email", "Email"),
          rules: [
            {
              required: true,
              message: translate(
                "pages.register.errors.requiredEmail",
                "Email is required"
              )
            },
            {
              type: "email",
              message: translate(
                "pages.register.errors.validEmail",
                "Invalid email address"
              )
            }
          ]
        },
        /* @__PURE__ */ React42.createElement(
          Input2,
          {
            size: "large",
            placeholder: translate("pages.register.fields.email", "Email")
          }
        )
      ),
      /* @__PURE__ */ React42.createElement(
        Form5.Item,
        {
          name: "password",
          label: translate("pages.register.fields.password", "Password"),
          rules: [
            {
              required: true,
              message: translate(
                "pages.register.errors.requiredPassword",
                "Password is required"
              )
            }
          ]
        },
        /* @__PURE__ */ React42.createElement(Input2, { type: "password", placeholder: "\u25CF\u25CF\u25CF\u25CF\u25CF\u25CF\u25CF\u25CF", size: "large" })
      ),
      /* @__PURE__ */ React42.createElement(
        "div",
        {
          style: {
            display: "flex",
            justifyContent: "space-between",
            marginBottom: "24px"
          }
        },
        loginLink ?? /* @__PURE__ */ React42.createElement(
          Typography10.Text,
          {
            style: {
              fontSize: 12,
              marginLeft: "auto"
            }
          },
          translate(
            "pages.register.buttons.haveAccount",
            translate(
              "pages.login.buttons.haveAccount",
              "Have an account?"
            )
          ),
          " ",
          /* @__PURE__ */ React42.createElement(
            Link,
            {
              style: {
                fontWeight: "bold",
                color: token.colorPrimaryTextHover
              },
              to: "/login"
            },
            translate(
              "pages.register.signin",
              translate("pages.login.signin", "Sign in")
            )
          )
        )
      ),
      /* @__PURE__ */ React42.createElement(
        Form5.Item,
        {
          style: {
            marginBottom: 0
          }
        },
        /* @__PURE__ */ React42.createElement(
          Button16,
          {
            type: "primary",
            size: "large",
            htmlType: "submit",
            loading: isPending,
            block: true
          },
          translate("pages.register.buttons.submit", "Sign up")
        )
      )
    ),
    hideForm && loginLink !== false && /* @__PURE__ */ React42.createElement(
      "div",
      {
        style: {
          marginTop: hideForm ? 16 : 8
        }
      },
      /* @__PURE__ */ React42.createElement(
        Typography10.Text,
        {
          style: {
            fontSize: 12
          }
        },
        translate(
          "pages.register.buttons.haveAccount",
          translate("pages.login.buttons.haveAccount", "Have an account?")
        ),
        " ",
        /* @__PURE__ */ React42.createElement(
          Link,
          {
            style: {
              fontWeight: "bold",
              color: token.colorPrimaryTextHover
            },
            to: "/login"
          },
          translate(
            "pages.register.signin",
            translate("pages.login.signin", "Sign in")
          )
        )
      )
    )
  );
  return /* @__PURE__ */ React42.createElement(Layout3, { style: layoutStyles, ...wrapperProps ?? {} }, /* @__PURE__ */ React42.createElement(
    Row2,
    {
      justify: "center",
      align: hideForm ? "top" : "middle",
      style: {
        padding: "16px 0",
        minHeight: "100dvh",
        paddingTop: hideForm ? "15dvh" : "16px"
      }
    },
    /* @__PURE__ */ React42.createElement(Col2, { xs: 22 }, renderContent ? renderContent(CardContent, PageTitle) : /* @__PURE__ */ React42.createElement(React42.Fragment, null, PageTitle, CardContent))
  ));
}, "RegisterPage");

// src/components/pages/auth/components/forgotPassword/index.tsx
import React43 from "react";
import {
  useLink as useLink6
} from "@refinedev/core";
import {
  Row as Row3,
  Col as Col3,
  Layout as Layout4,
  Card as Card6,
  Typography as Typography11,
  Form as Form6,
  Input as Input3,
  Button as Button17,
  theme as theme6
} from "antd";
import { useTranslate as useTranslate14, useForgotPassword } from "@refinedev/core";
var ForgotPasswordPage = /* @__PURE__ */ __name(({
  loginLink,
  wrapperProps,
  contentProps,
  renderContent,
  formProps,
  title,
  mutationVariables
}) => {
  const { token } = theme6.useToken();
  const [form] = Form6.useForm();
  const translate = useTranslate14();
  const Link = useLink6();
  const { mutate: forgotPassword, isPending } = useForgotPassword();
  const PageTitle = title === false ? null : /* @__PURE__ */ React43.createElement(
    "div",
    {
      style: {
        display: "flex",
        justifyContent: "center",
        marginBottom: "32px",
        fontSize: "20px"
      }
    },
    title ?? /* @__PURE__ */ React43.createElement(ThemedTitle, { collapsed: false })
  );
  const CardTitle = /* @__PURE__ */ React43.createElement(
    Typography11.Title,
    {
      level: 3,
      style: {
        color: token.colorPrimaryTextHover,
        ...titleStyles
      }
    },
    translate("pages.forgotPassword.title", "Forgot your password?")
  );
  const CardContent = /* @__PURE__ */ React43.createElement(
    Card6,
    {
      title: CardTitle,
      styles: {
        header: headStyles,
        body: bodyStyles
      },
      style: {
        ...containerStyles,
        backgroundColor: token.colorBgElevated
      },
      ...contentProps ?? {}
    },
    /* @__PURE__ */ React43.createElement(
      Form6,
      {
        layout: "vertical",
        form,
        onFinish: (values) => forgotPassword({ ...values, ...mutationVariables }),
        requiredMark: false,
        ...formProps
      },
      /* @__PURE__ */ React43.createElement(
        Form6.Item,
        {
          name: "email",
          label: translate("pages.forgotPassword.fields.email", "Email"),
          rules: [
            {
              required: true,
              message: translate(
                "pages.forgotPassword.errors.requiredEmail",
                "Email is required"
              )
            },
            {
              type: "email",
              message: translate(
                "pages.forgotPassword.errors.validEmail",
                "Invalid email address"
              )
            }
          ]
        },
        /* @__PURE__ */ React43.createElement(
          Input3,
          {
            type: "email",
            size: "large",
            placeholder: translate(
              "pages.forgotPassword.fields.email",
              "Email"
            )
          }
        )
      ),
      /* @__PURE__ */ React43.createElement(
        "div",
        {
          style: {
            display: "flex",
            justifyContent: "space-between"
          }
        },
        loginLink ?? /* @__PURE__ */ React43.createElement(
          Typography11.Text,
          {
            style: {
              fontSize: 12,
              marginLeft: "auto"
            }
          },
          translate(
            "pages.forgotPassword.buttons.haveAccount",
            translate(
              "pages.register.buttons.haveAccount",
              "Have an account? "
            )
          ),
          " ",
          /* @__PURE__ */ React43.createElement(
            Link,
            {
              style: {
                fontWeight: "bold",
                color: token.colorPrimaryTextHover
              },
              to: "/login"
            },
            translate(
              "pages.forgotPassword.signin",
              translate("pages.login.signin", "Sign in")
            )
          )
        )
      ),
      /* @__PURE__ */ React43.createElement(
        Form6.Item,
        {
          style: {
            marginTop: "24px",
            marginBottom: 0
          }
        },
        /* @__PURE__ */ React43.createElement(
          Button17,
          {
            type: "primary",
            size: "large",
            htmlType: "submit",
            loading: isPending,
            block: true
          },
          translate(
            "pages.forgotPassword.buttons.submit",
            "Send reset instructions"
          )
        )
      )
    )
  );
  return /* @__PURE__ */ React43.createElement(Layout4, { style: layoutStyles, ...wrapperProps ?? {} }, /* @__PURE__ */ React43.createElement(
    Row3,
    {
      justify: "center",
      align: "middle",
      style: {
        padding: "16px 0",
        minHeight: "100dvh"
      }
    },
    /* @__PURE__ */ React43.createElement(Col3, { xs: 22 }, renderContent ? renderContent(CardContent, PageTitle) : /* @__PURE__ */ React43.createElement(React43.Fragment, null, PageTitle, CardContent))
  ));
}, "ForgotPasswordPage");

// src/components/pages/auth/components/updatePassword/index.tsx
import React44 from "react";
import {
  Row as Row4,
  Col as Col4,
  Layout as Layout5,
  Card as Card7,
  Typography as Typography12,
  Form as Form7,
  Input as Input4,
  Button as Button18,
  theme as theme7
} from "antd";
import { useTranslate as useTranslate15, useUpdatePassword } from "@refinedev/core";
var UpdatePasswordPage = /* @__PURE__ */ __name(({
  wrapperProps,
  contentProps,
  renderContent,
  formProps,
  title,
  mutationVariables
}) => {
  const { token } = theme7.useToken();
  const [form] = Form7.useForm();
  const translate = useTranslate15();
  const { mutate: updatePassword, isPending } = useUpdatePassword();
  const PageTitle = title === false ? null : /* @__PURE__ */ React44.createElement(
    "div",
    {
      style: {
        display: "flex",
        justifyContent: "center",
        marginBottom: "32px",
        fontSize: "20px"
      }
    },
    title ?? /* @__PURE__ */ React44.createElement(ThemedTitle, { collapsed: false })
  );
  const CardTitle = /* @__PURE__ */ React44.createElement(
    Typography12.Title,
    {
      level: 3,
      style: {
        color: token.colorPrimaryTextHover,
        ...titleStyles
      }
    },
    translate("pages.updatePassword.title", "Set New Password")
  );
  const CardContent = /* @__PURE__ */ React44.createElement(
    Card7,
    {
      title: CardTitle,
      styles: {
        header: headStyles,
        body: bodyStyles
      },
      style: {
        ...containerStyles,
        backgroundColor: token.colorBgElevated
      },
      ...contentProps ?? {}
    },
    /* @__PURE__ */ React44.createElement(
      Form7,
      {
        layout: "vertical",
        form,
        onFinish: (values) => updatePassword({ ...values, ...mutationVariables }),
        requiredMark: false,
        ...formProps
      },
      /* @__PURE__ */ React44.createElement(
        Form7.Item,
        {
          name: "password",
          label: translate(
            "pages.updatePassword.fields.password",
            "New Password"
          ),
          rules: [
            {
              required: true,
              message: translate(
                "pages.updatePassword.errors.requiredPassword",
                "Password is required"
              )
            }
          ],
          style: { marginBottom: "12px" }
        },
        /* @__PURE__ */ React44.createElement(Input4, { type: "password", placeholder: "\u25CF\u25CF\u25CF\u25CF\u25CF\u25CF\u25CF\u25CF", size: "large" })
      ),
      /* @__PURE__ */ React44.createElement(
        Form7.Item,
        {
          name: "confirmPassword",
          label: translate(
            "pages.updatePassword.fields.confirmPassword",
            "Confirm New Password"
          ),
          hasFeedback: true,
          dependencies: ["password"],
          rules: [
            {
              required: true,
              message: translate(
                "pages.updatePassword.errors.requiredConfirmPassword",
                "Confirm password is required"
              )
            },
            ({ getFieldValue }) => ({
              validator(_, value) {
                if (!value || getFieldValue("password") === value) {
                  return Promise.resolve();
                }
                return Promise.reject(
                  new Error(
                    translate(
                      "pages.updatePassword.errors.confirmPasswordNotMatch",
                      "Passwords do not match"
                    )
                  )
                );
              }
            })
          ]
        },
        /* @__PURE__ */ React44.createElement(Input4, { type: "password", placeholder: "\u25CF\u25CF\u25CF\u25CF\u25CF\u25CF\u25CF\u25CF", size: "large" })
      ),
      /* @__PURE__ */ React44.createElement(
        Form7.Item,
        {
          style: {
            marginBottom: 0
          }
        },
        /* @__PURE__ */ React44.createElement(
          Button18,
          {
            type: "primary",
            size: "large",
            htmlType: "submit",
            loading: isPending,
            block: true
          },
          translate("pages.updatePassword.buttons.submit", "Update")
        )
      )
    )
  );
  return /* @__PURE__ */ React44.createElement(Layout5, { style: layoutStyles, ...wrapperProps ?? {} }, /* @__PURE__ */ React44.createElement(
    Row4,
    {
      justify: "center",
      align: "middle",
      style: {
        padding: "16px 0",
        minHeight: "100dvh"
      }
    },
    /* @__PURE__ */ React44.createElement(Col4, { xs: 22 }, renderContent ? renderContent(CardContent, PageTitle) : /* @__PURE__ */ React44.createElement(React44.Fragment, null, PageTitle, CardContent))
  ));
}, "UpdatePasswordPage");

// src/components/pages/auth/index.tsx
var AuthPage = /* @__PURE__ */ __name((props) => {
  const { type } = props;
  const renderView = /* @__PURE__ */ __name(() => {
    switch (type) {
      case "register":
        return /* @__PURE__ */ React45.createElement(RegisterPage, { ...props });
      case "forgotPassword":
        return /* @__PURE__ */ React45.createElement(ForgotPasswordPage, { ...props });
      case "updatePassword":
        return /* @__PURE__ */ React45.createElement(UpdatePasswordPage, { ...props });
      default:
        return /* @__PURE__ */ React45.createElement(LoginPage, { ...props });
    }
  }, "renderView");
  return /* @__PURE__ */ React45.createElement(React45.Fragment, null, renderView());
}, "AuthPage");

// src/components/breadcrumb/index.tsx
import React46 from "react";
import {
  useBreadcrumb,
  useLink as useLink7,
  useResourceParams as useResourceParams9,
  matchResourceFromRoute
} from "@refinedev/core";
import {
  Breadcrumb as AntdBreadcrumb
} from "antd";
import { HomeOutlined } from "@ant-design/icons";
var Breadcrumb = /* @__PURE__ */ __name(({
  breadcrumbProps,
  showHome = true,
  hideIcons = false,
  meta,
  minItems = 2
}) => {
  const { breadcrumbs } = useBreadcrumb({
    meta
  });
  const Link = useLink7();
  const { resources } = useResourceParams9();
  const rootRouteResource = matchResourceFromRoute("/", resources);
  if (breadcrumbs.length < minItems)
    return null;
  const breadCrumbItems = breadcrumbs.map(({ label, icon, href }) => ({
    key: `breadcrumb-item-${label}`,
    title: /* @__PURE__ */ React46.createElement(
      "div",
      {
        style: {
          display: "flex",
          alignItems: "center",
          justifyContent: "center",
          gap: 4
        }
      },
      !hideIcons && icon,
      href ? /* @__PURE__ */ React46.createElement(Link, { to: href }, label) : /* @__PURE__ */ React46.createElement("span", null, label)
    )
  }));
  const getBreadcrumbItems = /* @__PURE__ */ __name(() => {
    var _a, _b;
    if (showHome && rootRouteResource.found) {
      return [
        {
          key: "breadcrumb-item-home",
          title: /* @__PURE__ */ React46.createElement(Link, { to: "/" }, ((_b = (_a = rootRouteResource == null ? void 0 : rootRouteResource.resource) == null ? void 0 : _a.meta) == null ? void 0 : _b.icon) ?? /* @__PURE__ */ React46.createElement(HomeOutlined, null))
        },
        ...breadCrumbItems
      ];
    }
    return breadCrumbItems;
  }, "getBreadcrumbItems");
  return /* @__PURE__ */ React46.createElement(AntdBreadcrumb, { items: getBreadcrumbItems(), ...breadcrumbProps });
}, "Breadcrumb");

// src/components/pageHeader/index.tsx
import React47, { useContext as useContext3 } from "react";
import {
  PageHeader as AntdPageHeader
} from "@ant-design/pro-layout";
import { Button as Button19, ConfigProvider as ConfigProvider2, Typography as Typography13 } from "antd";
import { ArrowLeftOutlined, ArrowRightOutlined } from "@ant-design/icons";
import { RefinePageHeaderClassNames } from "@refinedev/ui-types";
var PageHeader = /* @__PURE__ */ __name(({ children, ...props }) => {
  var _a;
  const direction = (_a = useContext3(ConfigProvider2.ConfigContext)) == null ? void 0 : _a.direction;
  const renderBackButton = /* @__PURE__ */ __name(() => {
    const BackIcon = direction === "rtl" ? ArrowRightOutlined : ArrowLeftOutlined;
    return /* @__PURE__ */ React47.createElement(Button19, { type: "text", icon: /* @__PURE__ */ React47.createElement(BackIcon, null) });
  }, "renderBackButton");
  const backIcon = typeof props.backIcon === "undefined" ? renderBackButton() : props.backIcon;
  const title = typeof props.title === "string" ? /* @__PURE__ */ React47.createElement(
    Typography13.Title,
    {
      className: RefinePageHeaderClassNames.Title,
      level: 4,
      style: { marginBottom: 0 }
    },
    props.title
  ) : props.title;
  const subtitle = typeof props.title === "string" ? /* @__PURE__ */ React47.createElement(
    Typography13.Title,
    {
      className: RefinePageHeaderClassNames.SubTitle,
      level: 5,
      type: "secondary",
      style: { marginBottom: 0 }
    },
    props.subTitle
  ) : props.subTitle;
  return /* @__PURE__ */ React47.createElement(
    AntdPageHeader,
    {
      ...props,
      backIcon,
      title,
      subTitle: subtitle,
      style: { padding: 0, ...props.style }
    },
    children
  );
}, "PageHeader");

// src/components/autoSaveIndicator/index.tsx
import React48 from "react";
import {
  useTranslate as useTranslate16,
  AutoSaveIndicator as AutoSaveIndicatorCore
} from "@refinedev/core";
import { Typography as Typography14, theme as theme8 } from "antd";
import {
  EllipsisOutlined,
  SyncOutlined,
  CheckCircleOutlined,
  ExclamationCircleOutlined
} from "@ant-design/icons";
var AutoSaveIndicator = /* @__PURE__ */ __name(({
  status,
  elements: {
    success = /* @__PURE__ */ React48.createElement(
      Message,
      {
        translationKey: "autoSave.success",
        defaultMessage: "saved",
        icon: /* @__PURE__ */ React48.createElement(CheckCircleOutlined, null)
      }
    ),
    error = /* @__PURE__ */ React48.createElement(
      Message,
      {
        translationKey: "autoSave.error",
        defaultMessage: "auto save failure",
        icon: /* @__PURE__ */ React48.createElement(ExclamationCircleOutlined, null)
      }
    ),
    loading = /* @__PURE__ */ React48.createElement(
      Message,
      {
        translationKey: "autoSave.loading",
        defaultMessage: "saving...",
        icon: /* @__PURE__ */ React48.createElement(SyncOutlined, null)
      }
    ),
    idle = /* @__PURE__ */ React48.createElement(
      Message,
      {
        translationKey: "autoSave.idle",
        defaultMessage: "waiting for changes",
        icon: /* @__PURE__ */ React48.createElement(EllipsisOutlined, null)
      }
    )
  } = {}
}) => {
  return /* @__PURE__ */ React48.createElement(
    AutoSaveIndicatorCore,
    {
      status,
      elements: {
        success,
        error,
        loading,
        idle
      }
    }
  );
}, "AutoSaveIndicator");
var Message = /* @__PURE__ */ __name(({
  translationKey,
  defaultMessage,
  icon
}) => {
  const translate = useTranslate16();
  const { token } = theme8.useToken();
  return /* @__PURE__ */ React48.createElement(
    Typography14.Text,
    {
      style: {
        marginRight: 5,
        color: token.colorTextTertiary,
        fontSize: ".8rem"
      }
    },
    translate(translationKey, defaultMessage),
    /* @__PURE__ */ React48.createElement("span", { style: { marginLeft: ".2rem" } }, icon)
  );
}, "Message");

// src/definitions/filter-mappers/index.ts
import dayjs3 from "dayjs";
var rangePickerFilterMapper = /* @__PURE__ */ __name((selectedKeys, event) => {
  if (!selectedKeys) {
    return selectedKeys;
  }
  if (event === "value") {
    return selectedKeys.map((key) => {
      if (typeof key === "string") {
        return dayjs3(key);
      }
      return key;
    });
  }
  if (event === "onChange") {
    if (selectedKeys.every(dayjs3.isDayjs)) {
      return selectedKeys.map((date) => dayjs3(date).toISOString());
    }
  }
  return selectedKeys;
}, "rangePickerFilterMapper");

// src/definitions/upload/index.ts
var getValueFromEvent = /* @__PURE__ */ __name((event) => {
  const { fileList } = event;
  return [...fileList];
}, "getValueFromEvent");

// src/definitions/themes/index.ts
var RefineThemes = {
  Blue: {
    token: {
      colorPrimary: "#1677FF"
    }
  },
  Purple: {
    token: {
      colorPrimary: "#722ED1"
    }
  },
  Magenta: {
    token: {
      colorPrimary: "#EB2F96"
    }
  },
  Red: {
    token: {
      colorPrimary: "#F5222D"
    }
  },
  Orange: {
    token: {
      colorPrimary: "#FA541C"
    }
  },
  Yellow: {
    token: {
      colorPrimary: "#FAAD14"
    }
  },
  Green: {
    token: {
      colorPrimary: "#52C41A"
    }
  }
};
export {
  AuthPage,
  AutoSaveIndicator,
  BooleanField,
  Breadcrumb,
  CloneButton,
  Create,
  CreateButton,
  DateField,
  DeleteButton,
  Edit,
  EditButton,
  EmailField,
  ErrorComponent,
  ExportButton,
  FileField,
  FilterDropdown,
  ImageField,
  ImportButton,
  List,
  ListButton,
  MarkdownField,
  NumberField,
  PageHeader,
  RefineThemes,
  RefreshButton,
  SaveButton,
  Show,
  ShowButton,
  TagField,
  TextField,
  ThemedHeader,
  ThemedLayout,
  ThemedLayoutContext,
  ThemedLayoutContextProvider,
  ThemedSider,
  ThemedTitle,
  UrlField,
  WelcomePage,
  getDefaultFilter,
  getDefaultSortOrder,
  getValueFromEvent,
  mapAntdFilterToCrudFilter,
  mapAntdSorterToCrudSorting,
  rangePickerFilterMapper,
  useCheckboxGroup,
  useDrawer,
  useDrawerForm,
  useEditableTable,
  useFileUploadState,
  useForm,
  useImport,
  useModal,
  useModalForm,
  useNotificationProvider,
  useRadioGroup,
  useSelect,
  useSimpleList,
  useStepsForm,
  useTable,
  useThemedLayoutContext
};
