// @refinedev/antd/dist/index.cjs
"use client"
var __create = Object.create;
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __getProtoOf = Object.getPrototypeOf;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __name = (target, value) => __defProp(target, "name", { value, configurable: true });
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toESM = (mod, isNodeMode, target) => (target = mod != null ? __create(__getProtoOf(mod)) : {}, __copyProps(
  // If the importer is in node compatibility mode or this is not an ESM
  // file that has been converted to a CommonJS file using a Babel-
  // compatible transform (i.e. "__esModule" has not been set), then set
  // "default" to the CommonJS "module.exports" for node compatibility.
  isNodeMode || !mod || !mod.__esModule ? __defProp(target, "default", { value: mod, enumerable: true }) : target,
  mod
));
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// src/index.tsx
var src_exports = {};
__export(src_exports, {
  AuthPage: () => AuthPage,
  AutoSaveIndicator: () => AutoSaveIndicator,
  BooleanField: () => BooleanField,
  Breadcrumb: () => Breadcrumb,
  CloneButton: () => CloneButton,
  Create: () => Create,
  CreateButton: () => CreateButton,
  DateField: () => DateField,
  DeleteButton: () => DeleteButton,
  Edit: () => Edit,
  EditButton: () => EditButton,
  EmailField: () => EmailField,
  ErrorComponent: () => ErrorComponent,
  ExportButton: () => ExportButton,
  FileField: () => FileField,
  FilterDropdown: () => FilterDropdown,
  ImageField: () => ImageField,
  ImportButton: () => ImportButton,
  List: () => List,
  ListButton: () => ListButton,
  MarkdownField: () => MarkdownField,
  NumberField: () => NumberField,
  PageHeader: () => PageHeader,
  RefineThemes: () => RefineThemes,
  RefreshButton: () => RefreshButton,
  SaveButton: () => SaveButton,
  Show: () => Show,
  ShowButton: () => ShowButton,
  TagField: () => TagField,
  TextField: () => TextField,
  ThemedHeader: () => ThemedHeader,
  ThemedLayout: () => ThemedLayout,
  ThemedLayoutContext: () => ThemedLayoutContext,
  ThemedLayoutContextProvider: () => ThemedLayoutContextProvider,
  ThemedSider: () => ThemedSider,
  ThemedTitle: () => ThemedTitle,
  UrlField: () => UrlField,
  WelcomePage: () => WelcomePage,
  getDefaultFilter: () => getDefaultFilter,
  getDefaultSortOrder: () => getDefaultSortOrder,
  getValueFromEvent: () => getValueFromEvent,
  mapAntdFilterToCrudFilter: () => mapAntdFilterToCrudFilter,
  mapAntdSorterToCrudSorting: () => mapAntdSorterToCrudSorting,
  rangePickerFilterMapper: () => rangePickerFilterMapper,
  useCheckboxGroup: () => useCheckboxGroup,
  useDrawer: () => useDrawer,
  useDrawerForm: () => useDrawerForm,
  useEditableTable: () => useEditableTable,
  useFileUploadState: () => useFileUploadState,
  useForm: () => useForm,
  useImport: () => useImport,
  useModal: () => useModal,
  useModalForm: () => useModalForm,
  useNotificationProvider: () => useNotificationProvider,
  useRadioGroup: () => useRadioGroup,
  useSelect: () => useSelect,
  useSimpleList: () => useSimpleList,
  useStepsForm: () => useStepsForm,
  useTable: () => useTable,
  useThemedLayoutContext: () => useThemedLayoutContext
});
module.exports = __toCommonJS(src_exports);

// src/hooks/form/useForm.ts
var import_react = __toESM(require("react"));
var import_antd = require("antd");
var import_sunflower_antd = require("sunflower-antd");
var import_core = require("@refinedev/core");
var useForm = /* @__PURE__ */ __name(({
  action,
  resource,
  onMutationSuccess: onMutationSuccessProp,
  onMutationError: onMutationErrorProp,
  autoSave,
  submitOnEnter = false,
  warnWhenUnsavedChanges: warnWhenUnsavedChangesProp,
  redirect,
  successNotification,
  errorNotification,
  meta,
  queryMeta,
  mutationMeta,
  liveMode,
  liveParams,
  mutationMode,
  dataProviderName,
  onLiveEvent,
  invalidates,
  undoableTimeout,
  queryOptions,
  createMutationOptions,
  updateMutationOptions,
  id: idFromProps,
  overtimeOptions,
  optimisticUpdateMap,
  defaultFormValues,
  disableServerSideValidation: disableServerSideValidationProp = false
} = {}) => {
  var _a, _b;
  const { options } = (0, import_core.useRefineContext)();
  const disableServerSideValidation = (options == null ? void 0 : options.disableServerSideValidation) || disableServerSideValidationProp;
  const translate = (0, import_core.useTranslate)();
  const [formAnt] = import_antd.Form.useForm();
  const formSF = (0, import_sunflower_antd.useForm)({
    form: formAnt,
    defaultFormValues
  });
  const { form } = formSF;
  const useFormCoreResult = (0, import_core.useForm)({
    onMutationSuccess: onMutationSuccessProp ? onMutationSuccessProp : void 0,
    onMutationError: async (error, _variables, _context) => {
      if (disableServerSideValidation) {
        onMutationErrorProp == null ? void 0 : onMutationErrorProp(error, _variables, _context);
        return;
      }
      let parsedErrors = [];
      const fieldsValue = form.getFieldsValue();
      const fields = Object.keys((0, import_core.flattenObjectKeys)(fieldsValue));
      parsedErrors = fields.map((field) => {
        return {
          name: (0, import_core.propertyPathToArray)(field),
          errors: void 0
        };
      });
      form.setFields(parsedErrors);
      const errors = error == null ? void 0 : error.errors;
      for (const key in errors) {
        const fieldError = errors[key];
        let newError = [];
        if (Array.isArray(fieldError)) {
          newError = fieldError;
        }
        if (typeof fieldError === "string") {
          newError = [fieldError];
        }
        if (typeof fieldError === "boolean" && fieldError) {
          newError = ["Field is not valid."];
        }
        if (typeof fieldError === "object" && "key" in fieldError) {
          const translatedMessage = translate(
            fieldError.key,
            fieldError.message
          );
          newError = [translatedMessage];
        }
        parsedErrors.push({
          name: (0, import_core.propertyPathToArray)(key),
          errors: newError
        });
      }
      form.setFields([...parsedErrors]);
      onMutationErrorProp == null ? void 0 : onMutationErrorProp(error, _variables, _context);
    },
    redirect,
    action,
    resource,
    successNotification,
    errorNotification,
    meta,
    queryMeta,
    mutationMeta,
    liveMode,
    liveParams,
    mutationMode,
    dataProviderName,
    onLiveEvent,
    invalidates,
    undoableTimeout,
    queryOptions,
    createMutationOptions,
    updateMutationOptions,
    id: idFromProps,
    overtimeOptions,
    optimisticUpdateMap,
    autoSave
  });
  const { formLoading, onFinish, query, id, onFinishAutoSave } = useFormCoreResult;
  const { warnWhenUnsavedChanges: warnWhenUnsavedChangesRefine, setWarnWhen } = (0, import_core.useWarnAboutChange)();
  const warnWhenUnsavedChanges = warnWhenUnsavedChangesProp ?? warnWhenUnsavedChangesRefine;
  import_react.default.useEffect(() => {
    form.resetFields();
  }, [(_a = query == null ? void 0 : query.data) == null ? void 0 : _a.data, id]);
  const onKeyUp = /* @__PURE__ */ __name((event) => {
    if (submitOnEnter && event.key === "Enter") {
      form.submit();
    }
  }, "onKeyUp");
  const onValuesChange = /* @__PURE__ */ __name((changeValues, allValues) => {
    if (changeValues && warnWhenUnsavedChanges) {
      setWarnWhen(true);
    }
    if (autoSave == null ? void 0 : autoSave.enabled) {
      setWarnWhen(false);
      const onFinishFromProps = (autoSave == null ? void 0 : autoSave.onFinish) ?? ((values) => values);
      return onFinishAutoSave(onFinishFromProps(allValues)).catch(
        (error) => error
      );
    }
    return changeValues;
  }, "onValuesChange");
  const saveButtonProps = {
    disabled: formLoading,
    onClick: () => {
      form.submit();
    }
  };
  return {
    form: formSF.form,
    formProps: {
      ...formSF.formProps,
      onFinish: (values) => onFinish(values).catch((error) => error),
      onKeyUp,
      onValuesChange,
      initialValues: (_b = query == null ? void 0 : query.data) == null ? void 0 : _b.data
    },
    saveButtonProps,
    defaultFormValuesLoading: formSF.defaultFormValuesLoading,
    ...useFormCoreResult,
    onFinish: async (values) => {
      return await onFinish(values ?? formSF.form.getFieldsValue(true));
    }
  };
}, "useForm");

// src/hooks/form/useModalForm/useModalForm.ts
var import_react2 = __toESM(require("react"));
var import_core3 = require("@refinedev/core");

// src/hooks/modal/useModal/index.tsx
var import_core2 = require("@refinedev/core");
var useModal = /* @__PURE__ */ __name(({
  modalProps = {}
} = {}) => {
  const { show, close, visible } = (0, import_core2.useModal)({
    defaultVisible: modalProps.open
  });
  return {
    modalProps: {
      ...modalProps,
      onCancel: (e) => {
        var _a;
        (_a = modalProps.onCancel) == null ? void 0 : _a.call(modalProps, e);
        close();
      },
      open: visible,
      visible
    },
    show,
    close
  };
}, "useModal");

// src/hooks/form/useModalForm/useModalForm.ts
var useModalForm = /* @__PURE__ */ __name(({
  syncWithLocation,
  defaultVisible = false,
  autoSubmitClose = true,
  autoResetForm = true,
  autoResetFormWhenClose = true,
  autoSave,
  invalidates,
  ...rest
}) => {
  var _a;
  const [initiallySynced, setInitiallySynced] = import_react2.default.useState(false);
  const invalidate = (0, import_core3.useInvalidate)();
  const {
    resource,
    action: actionFromParams,
    identifier
  } = (0, import_core3.useResourceParams)({ resource: rest.resource });
  const parsed = (0, import_core3.useParsed)();
  const go = (0, import_core3.useGo)();
  const getUserFriendlyName = (0, import_core3.useUserFriendlyName)();
  const action = rest.action ?? actionFromParams ?? "";
  const syncingId = !(typeof syncWithLocation === "object" && (syncWithLocation == null ? void 0 : syncWithLocation.syncId) === false);
  const syncWithLocationKey = typeof syncWithLocation === "object" && "key" in syncWithLocation ? syncWithLocation.key : resource && action && syncWithLocation ? `modal-${identifier}-${action}` : void 0;
  const useFormProps = useForm({
    meta: {
      ...syncWithLocationKey ? { [syncWithLocationKey]: void 0 } : {},
      ...rest.meta
    },
    autoSave,
    invalidates,
    ...rest
  });
  const { form, formProps, id, setId, formLoading, onFinish, autoSaveProps } = useFormProps;
  const translate = (0, import_core3.useTranslate)();
  const { warnWhen, setWarnWhen } = (0, import_core3.useWarnAboutChange)();
  const { show, close, modalProps } = useModal({
    modalProps: {
      open: defaultVisible
    }
  });
  const visible = modalProps.open || false;
  const sunflowerUseModal = {
    modalProps,
    form,
    formLoading,
    formProps,
    formResult: void 0,
    formValues: form.getFieldsValue,
    defaultFormValuesLoading: false,
    initialValues: {},
    submit: onFinish,
    close,
    open: modalProps.open || false,
    show
  };
  import_react2.default.useEffect(() => {
    var _a2, _b, _c, _d;
    if (initiallySynced === false && syncWithLocationKey) {
      const openStatus = (_b = (_a2 = parsed == null ? void 0 : parsed.params) == null ? void 0 : _a2[syncWithLocationKey]) == null ? void 0 : _b.open;
      if (typeof openStatus === "boolean") {
        if (openStatus) {
          show();
        }
      } else if (typeof openStatus === "string") {
        if (openStatus === "true") {
          show();
        }
      }
      if (syncingId) {
        const idFromParams = (_d = (_c = parsed == null ? void 0 : parsed.params) == null ? void 0 : _c[syncWithLocationKey]) == null ? void 0 : _d.id;
        if (idFromParams) {
          setId == null ? void 0 : setId(idFromParams);
        }
      }
      setInitiallySynced(true);
    }
  }, [syncWithLocationKey, parsed, syncingId, setId]);
  import_react2.default.useEffect(() => {
    var _a2;
    if (initiallySynced === true) {
      if (visible && syncWithLocationKey) {
        go({
          query: {
            [syncWithLocationKey]: {
              ...(_a2 = parsed == null ? void 0 : parsed.params) == null ? void 0 : _a2[syncWithLocationKey],
              open: true,
              ...syncingId && id && { id }
            }
          },
          options: { keepQuery: true },
          type: "replace"
        });
      } else if (syncWithLocationKey && !visible) {
        go({
          query: {
            [syncWithLocationKey]: void 0
          },
          options: { keepQuery: true },
          type: "replace"
        });
      }
    }
  }, [id, visible, show, syncWithLocationKey, syncingId]);
  const saveButtonPropsSF = {
    disabled: formLoading,
    loading: formLoading,
    onClick: () => {
      form.submit();
    }
  };
  const handleClose = (0, import_react2.useCallback)(() => {
    if (autoSaveProps.status === "success" && (autoSave == null ? void 0 : autoSave.invalidateOnClose)) {
      invalidate({
        id,
        invalidates: invalidates || ["list", "many", "detail"],
        dataProviderName: rest.dataProviderName,
        resource: identifier
      });
    }
    if (warnWhen) {
      const warnWhenConfirm = window.confirm(
        translate(
          "warnWhenUnsavedChanges",
          "Are you sure you want to leave? You have unsaved changes."
        )
      );
      if (warnWhenConfirm) {
        setWarnWhen(false);
      } else {
        return;
      }
    }
    setId == null ? void 0 : setId(void 0);
    sunflowerUseModal.close();
    if (autoResetFormWhenClose) {
      form.resetFields();
    }
  }, [warnWhen, autoSaveProps.status]);
  const handleShow = (0, import_react2.useCallback)(
    (showId) => {
      if (typeof showId !== "undefined") {
        setId == null ? void 0 : setId(showId);
      }
      const needsIdToOpen = action === "edit" || action === "clone";
      const hasId = typeof showId !== "undefined" || typeof id !== "undefined";
      if (needsIdToOpen ? hasId : true) {
        sunflowerUseModal.show();
      }
    },
    [id]
  );
  const { visible: _visible, ...otherModalProps } = modalProps;
  const newModalProps = { open: _visible, ...otherModalProps };
  return {
    ...useFormProps,
    ...sunflowerUseModal,
    show: handleShow,
    close: handleClose,
    open: visible,
    formProps: {
      ...formProps,
      ...useFormProps.formProps,
      onValuesChange: formProps == null ? void 0 : formProps.onValuesChange,
      onKeyUp: formProps == null ? void 0 : formProps.onKeyUp,
      onFinish: async (values) => {
        await onFinish(values);
        if (autoSubmitClose) {
          close();
        }
        if (autoResetForm) {
          form.resetFields();
        }
      }
    },
    modalProps: {
      ...newModalProps,
      width: "1000px",
      okButtonProps: saveButtonPropsSF,
      title: translate(
        `${identifier}.titles.${rest.action}`,
        `${getUserFriendlyName(
          `${rest.action} ${((_a = resource == null ? void 0 : resource.meta) == null ? void 0 : _a.label) ?? identifier}`,
          "singular"
        )}`
      ),
      okText: translate("buttons.save", "Save"),
      cancelText: translate("buttons.cancel", "Cancel"),
      onCancel: handleClose,
      forceRender: true
    },
    formLoading
  };
}, "useModalForm");

// src/hooks/form/useDrawerForm/useDrawerForm.ts
var import_react3 = __toESM(require("react"));
var import_core5 = require("@refinedev/core");

// src/hooks/drawer/useDrawer/index.tsx
var import_core4 = require("@refinedev/core");
var useDrawer = /* @__PURE__ */ __name(({
  drawerProps = {}
} = {}) => {
  const { show, close, visible } = (0, import_core4.useModal)({
    defaultVisible: drawerProps.open
  });
  return {
    drawerProps: {
      ...drawerProps,
      onClose: (e) => {
        var _a;
        (_a = drawerProps.onClose) == null ? void 0 : _a.call(drawerProps, e);
        close();
      },
      open: visible
    },
    show,
    close
  };
}, "useDrawer");

// src/hooks/form/useDrawerForm/useDrawerForm.ts
var useDrawerForm = /* @__PURE__ */ __name(({
  syncWithLocation,
  defaultVisible = false,
  autoSubmitClose = true,
  autoResetForm = true,
  autoSave,
  invalidates,
  ...rest
}) => {
  const invalidate = (0, import_core5.useInvalidate)();
  const [initiallySynced, setInitiallySynced] = import_react3.default.useState(false);
  const { show, close, drawerProps } = useDrawer({
    drawerProps: {
      open: defaultVisible
    }
  });
  const visible = drawerProps.open || false;
  const {
    resource,
    action: actionFromParams,
    identifier
  } = (0, import_core5.useResourceParams)({
    resource: rest.resource
  });
  const parsed = (0, import_core5.useParsed)();
  const go = (0, import_core5.useGo)();
  const action = rest.action ?? actionFromParams ?? "";
  const syncingId = !(typeof syncWithLocation === "object" && (syncWithLocation == null ? void 0 : syncWithLocation.syncId) === false);
  const syncWithLocationKey = typeof syncWithLocation === "object" && "key" in syncWithLocation ? syncWithLocation.key : resource && action && syncWithLocation ? `drawer-${(resource == null ? void 0 : resource.identifier) ?? (resource == null ? void 0 : resource.name)}-${action}` : void 0;
  const useFormProps = useForm({
    meta: {
      ...syncWithLocationKey ? { [syncWithLocationKey]: void 0 } : {},
      ...rest.meta
    },
    autoSave,
    invalidates,
    ...rest
  });
  const { form, formProps, formLoading, id, setId, onFinish, autoSaveProps } = useFormProps;
  import_react3.default.useEffect(() => {
    var _a, _b, _c, _d;
    if (initiallySynced === false && syncWithLocationKey) {
      const openStatus = (_b = (_a = parsed == null ? void 0 : parsed.params) == null ? void 0 : _a[syncWithLocationKey]) == null ? void 0 : _b.open;
      if (typeof openStatus === "boolean") {
        openStatus ? show() : close();
      } else if (typeof openStatus === "string") {
        if (openStatus === "true") {
          show();
        }
      }
      if (syncingId) {
        const idFromParams = (_d = (_c = parsed == null ? void 0 : parsed.params) == null ? void 0 : _c[syncWithLocationKey]) == null ? void 0 : _d.id;
        if (idFromParams) {
          setId == null ? void 0 : setId(idFromParams);
        }
      }
      setInitiallySynced(true);
    }
  }, [syncWithLocationKey, parsed, syncingId, setId, initiallySynced]);
  import_react3.default.useEffect(() => {
    var _a;
    if (initiallySynced === true) {
      if (visible && syncWithLocationKey) {
        go({
          query: {
            [syncWithLocationKey]: {
              ...(_a = parsed == null ? void 0 : parsed.params) == null ? void 0 : _a[syncWithLocationKey],
              open: true,
              ...syncingId && id && { id }
            }
          },
          options: { keepQuery: true },
          type: "replace"
        });
      } else if (syncWithLocationKey && !visible) {
        go({
          query: {
            [syncWithLocationKey]: void 0
          },
          options: { keepQuery: true },
          type: "replace"
        });
      }
    }
  }, [
    id,
    visible,
    show,
    close,
    syncWithLocationKey,
    syncingId,
    initiallySynced
  ]);
  const translate = (0, import_core5.useTranslate)();
  const { warnWhen, setWarnWhen } = (0, import_core5.useWarnAboutChange)();
  const saveButtonProps = {
    disabled: formLoading,
    onClick: () => {
      form.submit();
    },
    loading: formLoading
  };
  const deleteButtonProps = {
    recordItemId: id,
    onSuccess: () => {
      setId == null ? void 0 : setId(void 0);
      close();
    }
  };
  const handleClose = (0, import_react3.useCallback)(() => {
    if (autoSaveProps.status === "success" && (autoSave == null ? void 0 : autoSave.invalidateOnClose)) {
      invalidate({
        id,
        invalidates: invalidates || ["list", "many", "detail"],
        dataProviderName: rest.dataProviderName,
        resource: identifier
      });
    }
    if (warnWhen) {
      const warnWhenConfirm = window.confirm(
        translate(
          "warnWhenUnsavedChanges",
          "Are you sure you want to leave? You have unsaved changes."
        )
      );
      if (warnWhenConfirm) {
        setWarnWhen(false);
      } else {
        return;
      }
    }
    close();
    setId == null ? void 0 : setId(void 0);
  }, [warnWhen]);
  const handleShow = (0, import_react3.useCallback)(
    (showId) => {
      if (typeof showId !== "undefined") {
        setId == null ? void 0 : setId(showId);
      }
      const needsIdToOpen = action === "edit" || action === "clone";
      const hasId = typeof showId !== "undefined" || typeof id !== "undefined";
      if (needsIdToOpen ? hasId : true) {
        show();
      }
    },
    [id]
  );
  return {
    ...useFormProps,
    show: handleShow,
    close: handleClose,
    formProps: {
      form,
      ...useFormProps.formProps,
      onValuesChange: formProps == null ? void 0 : formProps.onValuesChange,
      onKeyUp: formProps == null ? void 0 : formProps.onKeyUp,
      onFinish: async (values) => {
        await onFinish(values);
        if (autoSubmitClose) {
          close();
        }
        if (autoResetForm) {
          form.resetFields();
        }
      }
    },
    drawerProps: {
      ...drawerProps,
      width: "500px",
      onClose: handleClose,
      open: visible,
      forceRender: true
    },
    saveButtonProps,
    deleteButtonProps,
    formLoading
  };
}, "useDrawerForm");

// src/hooks/form/useStepsForm/useStepsForm.ts
var import_sunflower_antd2 = require("sunflower-antd");
var useStepsForm = /* @__PURE__ */ __name((props = {}) => {
  const useFormProps = useForm({
    ...props
  });
  const { form, formProps } = useFormProps;
  const stepsPropsSunflower = (0, import_sunflower_antd2.useStepsForm)({
    isBackValidate: false,
    form,
    submit: (values) => {
      var _a;
      (_a = formProps == null ? void 0 : formProps.onFinish) == null ? void 0 : _a.call(formProps, values);
    },
    ...props
  });
  return {
    ...useFormProps,
    ...stepsPropsSunflower,
    formLoading: useFormProps.formLoading,
    formProps: {
      ...stepsPropsSunflower.formProps,
      ...useFormProps.formProps,
      onValuesChange: formProps == null ? void 0 : formProps.onValuesChange,
      onKeyUp: formProps == null ? void 0 : formProps.onKeyUp
    },
    saveButtonProps: {
      ...useFormProps.saveButtonProps,
      onClick: () => stepsPropsSunflower.submit()
    }
  };
}, "useStepsForm");

// src/hooks/table/useTable/useTable.ts
var import_react5 = __toESM(require("react"));
var import_antd2 = require("antd");
var import_sunflower_antd3 = require("sunflower-antd");
var import_core8 = require("@refinedev/core");

// src/definitions/table/index.ts
var import_core6 = require("@refinedev/core");
var getDefaultSortOrder = /* @__PURE__ */ __name((columnName, sorter) => {
  const sort = (0, import_core6.getDefaultSortOrder)(columnName, sorter);
  if (sort) {
    return `${sort}end`;
  }
  return void 0;
}, "getDefaultSortOrder");
var getDefaultFilter = /* @__PURE__ */ __name((columnName, filters, operatorType = "eq") => {
  return (0, import_core6.getDefaultFilter)(columnName, filters, operatorType);
}, "getDefaultFilter");
var mapAntdSorterToCrudSorting = /* @__PURE__ */ __name((sorter) => {
  const crudSorting = [];
  if (Array.isArray(sorter)) {
    sorter.sort((a, b) => {
      var _a, _b;
      return (((_a = a.column) == null ? void 0 : _a.sorter).multiple ?? 0) < (((_b = b.column) == null ? void 0 : _b.sorter).multiple ?? 0) ? -1 : 0;
    }).map((item) => {
      if (item.field && item.order) {
        const field = Array.isArray(item.field) ? item.field.join(".") : `${item.field}`;
        crudSorting.push({
          field: `${item.columnKey ?? field}`,
          order: item.order.replace("end", "")
        });
      }
    });
  } else {
    if (sorter.field && sorter.order) {
      const field = Array.isArray(sorter.field) ? sorter.field.join(".") : `${sorter.field}`;
      crudSorting.push({
        field: `${sorter.columnKey ?? field}`,
        order: sorter.order.replace("end", "")
      });
    }
  }
  return crudSorting;
}, "mapAntdSorterToCrudSorting");
var mapAntdFilterToCrudFilter = /* @__PURE__ */ __name((tableFilters, prevFilters, initialFilters) => {
  const crudFilters = [];
  const mapInitialFilter = (initialFilters ?? []).reduce((acc, item) => {
    const field = item.key || item.field;
    return { ...acc, [field]: item };
  }, {});
  Object.keys(tableFilters).map((field) => {
    var _a, _b;
    const value = tableFilters[field];
    const operator = ((_a = prevFilters.filter((i) => i.operator !== "or").find((p) => p.field === field)) == null ? void 0 : _a.operator) || ((_b = mapInitialFilter[field]) == null ? void 0 : _b.operator);
    if (operator !== "or" && operator !== "and") {
      crudFilters.push({
        field,
        operator: operator ?? (Array.isArray(value) ? "in" : "eq"),
        value
      });
    }
  });
  return crudFilters;
}, "mapAntdFilterToCrudFilter");

// src/hooks/table/useTable/paginationLink.tsx
var import_core7 = require("@refinedev/core");
var import_react4 = __toESM(require("react"));
var PaginationLink = /* @__PURE__ */ __name(({ to, element }) => {
  const Link = (0, import_core7.useLink)();
  return /* @__PURE__ */ import_react4.default.createElement(
    Link,
    {
      to,
      replace: false,
      onClick: (e) => {
        e.preventDefault();
      }
    },
    element
  );
}, "PaginationLink");

// src/hooks/table/useTable/useTable.ts
var useTable = /* @__PURE__ */ __name(({
  onSearch,
  pagination: paginationFromProp,
  filters: filtersFromProp,
  sorters: sortersFromProp,
  syncWithLocation,
  resource,
  successNotification,
  errorNotification,
  queryOptions,
  liveMode: liveModeFromProp,
  onLiveEvent,
  liveParams,
  meta,
  dataProviderName
} = {}) => {
  const {
    tableQuery,
    currentPage,
    setCurrentPage,
    pageSize,
    setPageSize,
    filters,
    setFilters,
    sorters,
    setSorters,
    createLinkForSyncWithLocation,
    pageCount,
    overtime,
    result
  } = (0, import_core8.useTable)({
    pagination: paginationFromProp,
    filters: filtersFromProp,
    sorters: sortersFromProp,
    syncWithLocation,
    resource,
    successNotification,
    errorNotification,
    queryOptions,
    liveMode: liveModeFromProp,
    onLiveEvent,
    liveParams,
    meta,
    dataProviderName
  });
  const { syncWithLocation: defaultSyncWithLocation } = (0, import_core8.useSyncWithLocation)();
  const shouldSyncWithLocation = syncWithLocation ?? defaultSyncWithLocation;
  const breakpoint = import_antd2.Grid.useBreakpoint();
  const [form] = import_antd2.Form.useForm();
  const formSF = (0, import_sunflower_antd3.useForm)({
    form
  });
  const liveMode = (0, import_core8.useLiveMode)(liveModeFromProp);
  const isPaginationEnabled = (paginationFromProp == null ? void 0 : paginationFromProp.mode) !== "off";
  const preferredInitialFilters = filtersFromProp == null ? void 0 : filtersFromProp.initial;
  const { data, isFetched, isLoading } = tableQuery;
  import_react5.default.useEffect(() => {
    if (shouldSyncWithLocation) {
      const registeredFields = formSF.form.getFieldsValue();
      const filterFilterMap = Object.keys(registeredFields).reduce(
        (acc, curr) => {
          const filter = filters.find(
            (filter2) => "field" in filter2 && filter2.field === curr
          );
          if (filter) {
            acc[curr] = filter == null ? void 0 : filter.value;
          }
          return acc;
        },
        {}
      );
      formSF.form.setFieldsValue(filterFilterMap);
    }
  }, [shouldSyncWithLocation]);
  const onChange = /* @__PURE__ */ __name((paginationState, tableFilters, sorter) => {
    if (tableFilters && Object.keys(tableFilters).length > 0) {
      const crudFilters = mapAntdFilterToCrudFilter(
        tableFilters,
        filters,
        preferredInitialFilters
      );
      setFilters(crudFilters);
    }
    if (sorter && Object.keys(sorter).length > 0) {
      const crudSorting = mapAntdSorterToCrudSorting(sorter);
      setSorters(crudSorting);
    }
    if (isPaginationEnabled) {
      setCurrentPage == null ? void 0 : setCurrentPage(paginationState.current || 1);
      setPageSize == null ? void 0 : setPageSize(paginationState.pageSize || 10);
    }
  }, "onChange");
  const onFinish = /* @__PURE__ */ __name(async (value) => {
    if (onSearch) {
      const searchFilters = await onSearch(value);
      setFilters(searchFilters);
      if (isPaginationEnabled) {
        setCurrentPage == null ? void 0 : setCurrentPage(1);
      }
    }
  }, "onFinish");
  const antdPagination = /* @__PURE__ */ __name(() => {
    if (isPaginationEnabled) {
      return {
        itemRender: (page, type, element) => {
          var _a;
          const link = createLinkForSyncWithLocation({
            pagination: {
              pageSize,
              currentPage: page
            },
            sorters,
            filters
          });
          if (type === "page") {
            return (0, import_react5.createElement)(PaginationLink, {
              to: link,
              element: `${page}`
            });
          }
          if (type === "next" || type === "prev") {
            return (0, import_react5.createElement)(PaginationLink, {
              to: link,
              element
            });
          }
          if (type === "jump-next" || type === "jump-prev") {
            const elementChildren = (_a = element == null ? void 0 : element.props) == null ? void 0 : _a.children;
            return (0, import_react5.createElement)(PaginationLink, {
              to: link,
              element: import_react5.Children.count(elementChildren) > 1 ? (0, import_react5.createElement)(import_react5.Fragment, {}, elementChildren) : elementChildren
            });
          }
          return element;
        },
        pageSize,
        current: currentPage,
        simple: !breakpoint.sm,
        position: !breakpoint.sm ? ["bottomCenter"] : ["bottomRight"],
        total: data == null ? void 0 : data.total
      };
    }
    return false;
  }, "antdPagination");
  return {
    searchFormProps: {
      ...formSF.formProps,
      onFinish
    },
    tableProps: {
      dataSource: data == null ? void 0 : data.data,
      loading: liveMode === "auto" ? isLoading : !isFetched,
      onChange,
      pagination: antdPagination(),
      scroll: { x: true }
    },
    tableQuery,
    sorters,
    filters,
    setSorters,
    setFilters,
    currentPage,
    setCurrentPage,
    pageSize,
    setPageSize,
    pageCount,
    createLinkForSyncWithLocation,
    overtime,
    result
  };
}, "useTable");

// src/hooks/table/useEditableTable/useEditableTable.ts
var useEditableTable = /* @__PURE__ */ __name(({
  autoSubmitClose = true,
  ...props
} = {}) => {
  const table = useTable({
    ...props,
    successNotification: void 0,
    errorNotification: void 0
  });
  const edit = useForm({
    ...props,
    action: "edit",
    redirect: false
  });
  const { id: editId, setId, saveButtonProps } = edit;
  const cancelButtonProps = {
    onClick: () => {
      setId(void 0);
    }
  };
  const editButtonProps = /* @__PURE__ */ __name((id) => {
    return {
      onClick: () => setId(id)
    };
  }, "editButtonProps");
  const isEditing = /* @__PURE__ */ __name((id) => id === editId, "isEditing");
  return {
    ...table,
    ...edit,
    formProps: {
      ...edit.formProps,
      onFinish: async (values) => {
        try {
          const result = await edit.onFinish(values);
          if (autoSubmitClose) {
            setId(void 0);
          }
          return result;
        } catch (error) {
          return error;
        }
      }
    },
    saveButtonProps,
    cancelButtonProps,
    editButtonProps,
    isEditing
  };
}, "useEditableTable");

// src/hooks/fields/useSelect/index.ts
var import_core9 = require("@refinedev/core");
var useSelect = /* @__PURE__ */ __name((props) => {
  const { query, defaultValueQuery, onSearch, options } = (0, import_core9.useSelect)(props);
  return {
    selectProps: {
      options,
      onSearch,
      loading: defaultValueQuery.query.isFetching,
      showSearch: true,
      filterOption: false
    },
    query,
    defaultValueQuery: defaultValueQuery.query
  };
}, "useSelect");

// src/hooks/fields/useCheckboxGroup/index.ts
var import_core10 = require("@refinedev/core");
var useCheckboxGroup = /* @__PURE__ */ __name(({
  resource,
  sorters,
  filters,
  optionLabel,
  optionValue,
  queryOptions,
  pagination,
  liveMode,
  defaultValue,
  selectedOptionsOrder,
  onLiveEvent,
  liveParams,
  meta,
  dataProviderName,
  ...rest
}) => {
  const { query, options } = (0, import_core10.useSelect)({
    resource,
    sorters,
    filters,
    optionLabel,
    optionValue,
    queryOptions,
    pagination,
    liveMode,
    defaultValue,
    selectedOptionsOrder,
    onLiveEvent,
    liveParams,
    meta,
    dataProviderName,
    ...rest
  });
  return {
    checkboxGroupProps: {
      options,
      defaultValue
    },
    query
  };
}, "useCheckboxGroup");

// src/hooks/fields/useRadioGroup/index.ts
var import_core11 = require("@refinedev/core");
var useRadioGroup = /* @__PURE__ */ __name(({
  resource,
  sorters,
  filters,
  optionLabel,
  optionValue,
  queryOptions,
  pagination,
  liveMode,
  defaultValue,
  selectedOptionsOrder,
  onLiveEvent,
  liveParams,
  meta,
  dataProviderName,
  ...rest
}) => {
  const { query, options } = (0, import_core11.useSelect)({
    resource,
    sorters,
    filters,
    optionLabel,
    optionValue,
    queryOptions,
    pagination,
    liveMode,
    defaultValue,
    selectedOptionsOrder,
    onLiveEvent,
    liveParams,
    meta,
    dataProviderName,
    ...rest
  });
  return {
    radioGroupProps: {
      options,
      defaultValue
    },
    query
  };
}, "useRadioGroup");

// src/hooks/import/index.tsx
var import_react6 = __toESM(require("react"));
var import_antd3 = require("antd");
var import_core12 = require("@refinedev/core");
var useImport = /* @__PURE__ */ __name(({
  resource: resourceFromProp,
  mapData = /* @__PURE__ */ __name((item) => item, "mapData"),
  paparseOptions,
  batchSize = Number.MAX_SAFE_INTEGER,
  onFinish,
  meta,
  dataProviderName,
  onProgress: onProgressFromProp
} = {}) => {
  const t = (0, import_core12.useTranslate)();
  const { resource } = (0, import_core12.useResourceParams)({ resource: resourceFromProp });
  const { mutationResult, isLoading, handleChange } = (0, import_core12.useImport)({
    resource: (resource == null ? void 0 : resource.identifier) ?? (resource == null ? void 0 : resource.name),
    mapData,
    paparseOptions,
    batchSize,
    meta,
    dataProviderName,
    onFinish,
    onProgress: onProgressFromProp ?? (({ totalAmount, processedAmount }) => {
      if (totalAmount > 0 && processedAmount > 0) {
        const description = /* @__PURE__ */ import_react6.default.createElement(
          "div",
          {
            style: {
              display: "flex",
              alignItems: "center",
              justifyContent: "space-between",
              marginTop: "-7px"
            }
          },
          /* @__PURE__ */ import_react6.default.createElement(
            import_antd3.Progress,
            {
              type: "circle",
              percent: Math.floor(processedAmount / totalAmount * 100),
              size: 50,
              strokeColor: "#1890ff",
              status: "normal"
            }
          ),
          /* @__PURE__ */ import_react6.default.createElement("span", { style: { marginLeft: 8, width: "100%" } }, t(
            "notifications.importProgress",
            {
              processed: processedAmount,
              total: totalAmount
            },
            `Importing: ${processedAmount}/${totalAmount}`
          ))
        );
        import_antd3.notification.open({
          description,
          message: null,
          key: `${resource}-import`,
          duration: 0
        });
        if (processedAmount >= totalAmount) {
        }
        if (processedAmount === totalAmount) {
          setTimeout(() => {
            import_antd3.notification.destroy(`${resource}-import`);
          }, 4500);
        }
      }
    })
  });
  return {
    uploadProps: {
      onChange: handleChange,
      beforeUpload: () => false,
      showUploadList: false,
      accept: ".csv"
    },
    buttonProps: {
      type: "default",
      loading: isLoading
    },
    mutationResult,
    isLoading
  };
}, "useImport");

// src/hooks/useSimpleList/useSimpleList.ts
var import_react7 = require("react");
var import_antd4 = require("antd");
var import_core13 = require("@refinedev/core");
var import_core14 = require("@refinedev/core");
var useSimpleList = /* @__PURE__ */ __name(({
  resource,
  pagination: paginationFromProp,
  filters: filtersFromProp,
  sorters: sortersFromProp,
  onSearch,
  queryOptions,
  syncWithLocation,
  successNotification,
  errorNotification,
  liveMode: liveModeFromProp,
  onLiveEvent,
  liveParams,
  meta,
  dataProviderName
} = {}) => {
  const {
    sorters,
    filters,
    currentPage,
    pageSize,
    pageCount,
    setFilters,
    setCurrentPage,
    setPageSize,
    setSorters,
    createLinkForSyncWithLocation,
    tableQuery,
    overtime,
    result
  } = (0, import_core13.useTable)({
    resource,
    pagination: paginationFromProp,
    filters: filtersFromProp,
    sorters: sortersFromProp,
    queryOptions,
    successNotification,
    errorNotification,
    liveMode: liveModeFromProp,
    onLiveEvent,
    liveParams,
    meta,
    syncWithLocation,
    dataProviderName
  });
  const isPaginationEnabled = (paginationFromProp == null ? void 0 : paginationFromProp.mode) !== "off";
  const breakpoint = import_antd4.Grid.useBreakpoint();
  const liveMode = (0, import_core14.useLiveMode)(liveModeFromProp);
  const [form] = import_antd4.Form.useForm();
  const { data, isFetched, isLoading } = tableQuery;
  const onChange = /* @__PURE__ */ __name((page, pageSize2) => {
    if (isPaginationEnabled) {
      setCurrentPage(page);
      setPageSize(pageSize2 || 10);
    }
  }, "onChange");
  const onFinish = /* @__PURE__ */ __name(async (values) => {
    if (onSearch) {
      const searchFilters = await onSearch(values);
      if (isPaginationEnabled) {
        setCurrentPage == null ? void 0 : setCurrentPage(1);
      }
      return setFilters(searchFilters);
    }
  }, "onFinish");
  const antdPagination = /* @__PURE__ */ __name(() => {
    if (isPaginationEnabled) {
      return {
        itemRender: (page, type, element) => {
          var _a;
          const link = createLinkForSyncWithLocation({
            pagination: {
              pageSize,
              currentPage: page
            },
            sorters,
            filters
          });
          if (type === "page") {
            return (0, import_react7.createElement)(PaginationLink, {
              to: link,
              element: `${page}`
            });
          }
          if (type === "next" || type === "prev") {
            return (0, import_react7.createElement)(PaginationLink, {
              to: link,
              element
            });
          }
          if (type === "jump-next" || type === "jump-prev") {
            const elementChildren = (_a = element == null ? void 0 : element.props) == null ? void 0 : _a.children;
            return (0, import_react7.createElement)(PaginationLink, {
              to: link,
              element: import_react7.Children.count(elementChildren) > 1 ? (0, import_react7.createElement)(import_react7.Fragment, {}, elementChildren) : elementChildren
            });
          }
          return element;
        },
        pageSize,
        current: currentPage,
        simple: !breakpoint.sm,
        total: data == null ? void 0 : data.total,
        onChange
      };
    }
    return false;
  }, "antdPagination");
  return {
    searchFormProps: {
      form,
      onFinish
    },
    listProps: {
      dataSource: data == null ? void 0 : data.data,
      loading: liveMode === "auto" ? isLoading : !isFetched,
      pagination: antdPagination()
    },
    query: tableQuery,
    filters,
    setFilters,
    sorters,
    setSorters,
    currentPage,
    setCurrentPage,
    pageSize,
    setPageSize,
    pageCount,
    createLinkForSyncWithLocation,
    overtime,
    result
  };
}, "useSimpleList");

// src/hooks/useFileUploadState/index.ts
var import_react8 = require("react");
var useFileUploadState = /* @__PURE__ */ __name(() => {
  const [isLoading, setIsloading] = (0, import_react8.useState)(false);
  const onChange = (0, import_react8.useCallback)((info) => {
    const fileListLoadings = mapStatusToLoading(info.fileList);
    if (fileListLoadings.includes(true)) {
      setIsloading(true);
    } else {
      setIsloading(false);
    }
  }, []);
  return (0, import_react8.useMemo)(() => ({ isLoading, onChange }), [isLoading]);
}, "useFileUploadState");
var mapStatusToLoading = /* @__PURE__ */ __name((files) => {
  return files.map((file) => {
    switch (file.status) {
      case "uploading":
        return true;
      default:
        return false;
    }
  });
}, "mapStatusToLoading");

// src/hooks/useThemedLayoutContext/index.ts
var import_react10 = require("react");

// src/contexts/themedLayoutContext/index.tsx
var import_react9 = __toESM(require("react"));
var ThemedLayoutContext = import_react9.default.createContext({
  siderCollapsed: false,
  mobileSiderOpen: false,
  setSiderCollapsed: () => void 0,
  setMobileSiderOpen: () => void 0
});
var ThemedLayoutContextProvider = /* @__PURE__ */ __name(({ children, initialSiderCollapsed, onSiderCollapsed }) => {
  const [siderCollapsed, setSiderCollapsedState] = (0, import_react9.useState)(
    initialSiderCollapsed ?? false
  );
  const [mobileSiderOpen, setMobileSiderOpen] = (0, import_react9.useState)(false);
  const setSiderCollapsed = /* @__PURE__ */ __name((collapsed) => {
    setSiderCollapsedState(collapsed);
    if (onSiderCollapsed) {
      onSiderCollapsed(collapsed);
    }
  }, "setSiderCollapsed");
  return /* @__PURE__ */ import_react9.default.createElement(
    ThemedLayoutContext.Provider,
    {
      value: {
        siderCollapsed,
        mobileSiderOpen,
        setSiderCollapsed,
        setMobileSiderOpen
      }
    },
    children
  );
}, "ThemedLayoutContextProvider");

// src/hooks/useThemedLayoutContext/index.ts
var useThemedLayoutContext = /* @__PURE__ */ __name(() => {
  const {
    mobileSiderOpen,
    siderCollapsed,
    setMobileSiderOpen,
    setSiderCollapsed
  } = (0, import_react10.useContext)(ThemedLayoutContext);
  return {
    mobileSiderOpen,
    siderCollapsed,
    setMobileSiderOpen,
    setSiderCollapsed
  };
}, "useThemedLayoutContext");

// src/providers/notificationProvider/index.tsx
var import_antd6 = require("antd");
var import_react12 = __toESM(require("react"));

// src/components/undoableNotification/index.tsx
var import_react11 = __toESM(require("react"));
var import_antd5 = require("antd");
var import_icons = require("@ant-design/icons");
var UndoableNotification = /* @__PURE__ */ __name(({
  message,
  cancelMutation,
  undoableTimeout
}) => /* @__PURE__ */ import_react11.default.createElement(
  "div",
  {
    style: {
      display: "flex",
      alignItems: "center",
      justifyContent: "space-between",
      marginTop: "-7px"
    }
  },
  /* @__PURE__ */ import_react11.default.createElement(
    import_antd5.Progress,
    {
      type: "circle",
      percent: (undoableTimeout ?? 0) * 20,
      format: (time) => time && time / 20,
      size: 50,
      strokeColor: "#1890ff",
      status: "normal"
    }
  ),
  /* @__PURE__ */ import_react11.default.createElement("span", { style: { marginLeft: 8, width: "100%" } }, message),
  /* @__PURE__ */ import_react11.default.createElement(
    import_antd5.Button,
    {
      style: { flexShrink: 0 },
      onClick: cancelMutation,
      disabled: undoableTimeout === 0,
      icon: /* @__PURE__ */ import_react11.default.createElement(import_icons.UndoOutlined, null)
    }
  )
), "UndoableNotification");

// src/providers/notificationProvider/index.tsx
var useNotificationProvider = /* @__PURE__ */ __name(() => {
  const { notification: notificationFromContext } = import_antd6.App.useApp();
  const notification2 = "open" in notificationFromContext ? notificationFromContext : import_antd6.notification;
  const notificationProvider = {
    open: ({
      key,
      message,
      description,
      type,
      cancelMutation,
      undoableTimeout
    }) => {
      if (type === "progress") {
        notification2.open({
          key,
          description: /* @__PURE__ */ import_react12.default.createElement(
            UndoableNotification,
            {
              notificationKey: key,
              message,
              cancelMutation: () => {
                cancelMutation == null ? void 0 : cancelMutation();
                notification2.destroy(key ?? "");
              },
              undoableTimeout
            }
          ),
          message: null,
          duration: 0,
          closeIcon: /* @__PURE__ */ import_react12.default.createElement(import_react12.default.Fragment, null)
        });
      } else {
        notification2.open({
          key,
          description: message,
          message: description ?? null,
          type
        });
      }
    },
    close: (key) => notification2.destroy(key)
  };
  return notificationProvider;
}, "useNotificationProvider");

// src/components/themedLayout/index.tsx
var import_react15 = __toESM(require("react"));
var import_antd9 = require("antd");

// src/components/themedLayout/sider/index.tsx
var import_react13 = __toESM(require("react"));
var import_antd7 = require("antd");
var import_icons2 = require("@ant-design/icons");
var import_core15 = require("@refinedev/core");

// src/components/themedLayout/sider/styles.ts
var drawerButtonStyles = {
  borderStartStartRadius: 0,
  borderEndStartRadius: 0,
  position: "fixed",
  top: 64,
  zIndex: 999
};

// src/components/themedLayout/sider/index.tsx
var ThemedSider = /* @__PURE__ */ __name(({
  Title: TitleFromProps,
  render,
  meta,
  fixed,
  activeItemDisabled = false,
  siderItemsAreCollapsed = true
}) => {
  var _a;
  const { token } = import_antd7.theme.useToken();
  const {
    siderCollapsed,
    setSiderCollapsed,
    mobileSiderOpen,
    setMobileSiderOpen
  } = useThemedLayoutContext();
  const isExistAuthentication = (0, import_core15.useIsExistAuthentication)();
  const direction = (_a = (0, import_react13.useContext)(import_antd7.ConfigProvider.ConfigContext)) == null ? void 0 : _a.direction;
  const Link = (0, import_core15.useLink)();
  const { warnWhen, setWarnWhen } = (0, import_core15.useWarnAboutChange)();
  const translate = (0, import_core15.useTranslate)();
  const { menuItems, selectedKey, defaultOpenKeys } = (0, import_core15.useMenu)({ meta });
  const breakpoint = import_antd7.Grid.useBreakpoint();
  const { mutate: mutateLogout } = (0, import_core15.useLogout)();
  const isMobile = typeof breakpoint.lg === "undefined" ? false : !breakpoint.lg;
  const RenderToTitle = TitleFromProps ?? ThemedTitle;
  const renderTreeView = /* @__PURE__ */ __name((tree, selectedKey2) => {
    return tree.map((item) => {
      const { key, name, children, meta: meta2, list } = item;
      const parentName = meta2 == null ? void 0 : meta2.parent;
      const label = (item == null ? void 0 : item.label) ?? (meta2 == null ? void 0 : meta2.label) ?? name;
      const icon = meta2 == null ? void 0 : meta2.icon;
      const route = list;
      if (children.length > 0) {
        return /* @__PURE__ */ import_react13.default.createElement(
          import_core15.CanAccess,
          {
            key: item.key,
            resource: name,
            action: "list",
            params: {
              resource: item
            }
          },
          /* @__PURE__ */ import_react13.default.createElement(
            import_antd7.Menu.SubMenu,
            {
              key: item.key,
              icon: icon ?? /* @__PURE__ */ import_react13.default.createElement(import_icons2.UnorderedListOutlined, null),
              title: label
            },
            renderTreeView(children, selectedKey2)
          )
        );
      }
      const isSelected = key === selectedKey2;
      const isRoute = !(parentName !== void 0 && children.length === 0);
      const linkStyle = activeItemDisabled && isSelected ? { pointerEvents: "none" } : {};
      return /* @__PURE__ */ import_react13.default.createElement(
        import_core15.CanAccess,
        {
          key: item.key,
          resource: name,
          action: "list",
          params: {
            resource: item
          }
        },
        /* @__PURE__ */ import_react13.default.createElement(
          import_antd7.Menu.Item,
          {
            key: item.key,
            icon: icon ?? (isRoute && /* @__PURE__ */ import_react13.default.createElement(import_icons2.UnorderedListOutlined, null)),
            style: linkStyle
          },
          /* @__PURE__ */ import_react13.default.createElement(Link, { to: route ?? "", style: linkStyle }, label),
          !siderCollapsed && isSelected && /* @__PURE__ */ import_react13.default.createElement("div", { className: "ant-menu-tree-arrow" })
        )
      );
    });
  }, "renderTreeView");
  const handleLogout = /* @__PURE__ */ __name(() => {
    if (warnWhen) {
      const confirm = window.confirm(
        translate(
          "warnWhenUnsavedChanges",
          "Are you sure you want to leave? You have unsaved changes."
        )
      );
      if (confirm) {
        setWarnWhen(false);
        mutateLogout();
      }
    } else {
      mutateLogout();
    }
  }, "handleLogout");
  const logout = isExistAuthentication && /* @__PURE__ */ import_react13.default.createElement(
    import_antd7.Menu.Item,
    {
      key: "logout",
      onClick: () => handleLogout(),
      icon: /* @__PURE__ */ import_react13.default.createElement(import_icons2.LogoutOutlined, null)
    },
    translate("buttons.logout", "Logout")
  );
  const defaultExpandMenuItems = (() => {
    if (siderItemsAreCollapsed)
      return [];
    return menuItems.map(({ key }) => key);
  })();
  const items = renderTreeView(menuItems, selectedKey);
  const renderSider = /* @__PURE__ */ __name(() => {
    if (render) {
      return render({
        items,
        logout,
        collapsed: siderCollapsed
      });
    }
    return [...items, logout].filter(Boolean);
  }, "renderSider");
  const renderMenu = /* @__PURE__ */ __name(() => {
    return /* @__PURE__ */ import_react13.default.createElement(
      import_antd7.Menu,
      {
        selectedKeys: selectedKey ? [selectedKey] : [],
        defaultOpenKeys: [...defaultOpenKeys, ...defaultExpandMenuItems],
        mode: "inline",
        style: {
          paddingTop: "8px",
          border: "none",
          overflow: "auto",
          height: "calc(100% - 72px)"
        },
        onClick: () => {
          setMobileSiderOpen(false);
        }
      },
      renderSider()
    );
  }, "renderMenu");
  const renderDrawerSider = /* @__PURE__ */ __name(() => {
    return /* @__PURE__ */ import_react13.default.createElement(import_react13.default.Fragment, null, /* @__PURE__ */ import_react13.default.createElement(
      import_antd7.Drawer,
      {
        open: mobileSiderOpen,
        onClose: () => setMobileSiderOpen(false),
        placement: direction === "rtl" ? "right" : "left",
        closable: false,
        width: 200,
        styles: {
          body: {
            padding: 0
          }
        },
        maskClosable: true
      },
      /* @__PURE__ */ import_react13.default.createElement(import_antd7.Layout, null, /* @__PURE__ */ import_react13.default.createElement(
        import_antd7.Layout.Sider,
        {
          style: {
            height: "100vh",
            backgroundColor: token.colorBgContainer,
            borderRight: `1px solid ${token.colorBgElevated}`
          }
        },
        /* @__PURE__ */ import_react13.default.createElement(
          "div",
          {
            style: {
              width: "200px",
              padding: "0 16px",
              display: "flex",
              justifyContent: "flex-start",
              alignItems: "center",
              height: "64px",
              backgroundColor: token.colorBgElevated
            }
          },
          /* @__PURE__ */ import_react13.default.createElement(RenderToTitle, { collapsed: false })
        ),
        renderMenu()
      ))
    ), /* @__PURE__ */ import_react13.default.createElement(
      import_antd7.Button,
      {
        style: drawerButtonStyles,
        size: "large",
        onClick: () => setMobileSiderOpen(true),
        icon: /* @__PURE__ */ import_react13.default.createElement(import_icons2.BarsOutlined, null)
      }
    ));
  }, "renderDrawerSider");
  if (isMobile) {
    return renderDrawerSider();
  }
  const siderStyles = {
    backgroundColor: token.colorBgContainer,
    borderRight: `1px solid ${token.colorBgElevated}`
  };
  if (fixed) {
    siderStyles.position = "fixed";
    siderStyles.top = 0;
    siderStyles.height = "100vh";
    siderStyles.zIndex = 999;
  }
  const renderClosingIcons = /* @__PURE__ */ __name(() => {
    const iconProps = { style: { color: token.colorPrimary } };
    const OpenIcon = direction === "rtl" ? import_icons2.RightOutlined : import_icons2.LeftOutlined;
    const CollapsedIcon = direction === "rtl" ? import_icons2.LeftOutlined : import_icons2.RightOutlined;
    const IconComponent = siderCollapsed ? CollapsedIcon : OpenIcon;
    return /* @__PURE__ */ import_react13.default.createElement(IconComponent, { ...iconProps });
  }, "renderClosingIcons");
  return /* @__PURE__ */ import_react13.default.createElement(import_react13.default.Fragment, null, fixed && /* @__PURE__ */ import_react13.default.createElement(
    "div",
    {
      style: {
        width: siderCollapsed ? "80px" : "200px",
        transition: "all 0.2s"
      }
    }
  ), /* @__PURE__ */ import_react13.default.createElement(
    import_antd7.Layout.Sider,
    {
      style: siderStyles,
      collapsible: true,
      collapsed: siderCollapsed,
      onCollapse: (collapsed, type) => {
        if (type === "clickTrigger") {
          setSiderCollapsed(collapsed);
        }
      },
      collapsedWidth: 80,
      breakpoint: "lg",
      trigger: /* @__PURE__ */ import_react13.default.createElement(
        import_antd7.Button,
        {
          type: "text",
          style: {
            borderRadius: 0,
            height: "100%",
            width: "100%",
            backgroundColor: token.colorBgElevated
          }
        },
        renderClosingIcons()
      )
    },
    /* @__PURE__ */ import_react13.default.createElement(
      "div",
      {
        style: {
          width: siderCollapsed ? "80px" : "200px",
          padding: siderCollapsed ? "0" : "0 16px",
          display: "flex",
          justifyContent: siderCollapsed ? "center" : "flex-start",
          alignItems: "center",
          height: "64px",
          backgroundColor: token.colorBgElevated,
          fontSize: "14px"
        }
      },
      /* @__PURE__ */ import_react13.default.createElement(RenderToTitle, { collapsed: siderCollapsed })
    ),
    renderMenu()
  ));
}, "ThemedSider");

// src/components/themedLayout/header/index.tsx
var import_react14 = __toESM(require("react"));
var import_antd8 = require("antd");
var import_core16 = require("@refinedev/core");
var ThemedHeader = /* @__PURE__ */ __name(({
  sticky
}) => {
  const { token } = import_antd8.theme.useToken();
  const { data: user } = (0, import_core16.useGetIdentity)();
  const shouldRenderHeader = user && (user.name || user.avatar);
  if (!shouldRenderHeader) {
    return null;
  }
  const headerStyles = {
    backgroundColor: token.colorBgElevated,
    display: "flex",
    justifyContent: "flex-end",
    alignItems: "center",
    padding: "0px 24px",
    height: "64px"
  };
  if (sticky) {
    headerStyles.position = "sticky";
    headerStyles.top = 0;
    headerStyles.zIndex = 1;
  }
  return /* @__PURE__ */ import_react14.default.createElement(import_antd8.Layout.Header, { style: headerStyles }, /* @__PURE__ */ import_react14.default.createElement(import_antd8.Space, null, /* @__PURE__ */ import_react14.default.createElement(import_antd8.Space, { size: "middle" }, (user == null ? void 0 : user.name) && /* @__PURE__ */ import_react14.default.createElement(import_antd8.Typography.Text, { strong: true }, user.name), (user == null ? void 0 : user.avatar) && /* @__PURE__ */ import_react14.default.createElement(import_antd8.Avatar, { src: user == null ? void 0 : user.avatar, alt: user == null ? void 0 : user.name }))));
}, "ThemedHeader");

// src/components/themedLayout/index.tsx
var ThemedLayout = /* @__PURE__ */ __name(({
  children,
  Header,
  Sider,
  Title,
  Footer,
  OffLayoutArea,
  initialSiderCollapsed,
  onSiderCollapsed
}) => {
  const breakpoint = import_antd9.Grid.useBreakpoint();
  const SiderToRender = Sider ?? ThemedSider;
  const HeaderToRender = Header ?? ThemedHeader;
  const isSmall = typeof breakpoint.sm === "undefined" ? true : breakpoint.sm;
  const hasSider = !!SiderToRender({ Title });
  return /* @__PURE__ */ import_react15.default.createElement(
    ThemedLayoutContextProvider,
    {
      initialSiderCollapsed,
      onSiderCollapsed
    },
    /* @__PURE__ */ import_react15.default.createElement(import_antd9.Layout, { style: { minHeight: "100vh" }, hasSider }, /* @__PURE__ */ import_react15.default.createElement(SiderToRender, { Title }), /* @__PURE__ */ import_react15.default.createElement(import_antd9.Layout, null, /* @__PURE__ */ import_react15.default.createElement(HeaderToRender, null), /* @__PURE__ */ import_react15.default.createElement(import_antd9.Layout.Content, null, /* @__PURE__ */ import_react15.default.createElement(
      "div",
      {
        style: {
          minHeight: 360,
          padding: isSmall ? 24 : 12
        }
      },
      children
    ), OffLayoutArea && /* @__PURE__ */ import_react15.default.createElement(OffLayoutArea, null)), Footer && /* @__PURE__ */ import_react15.default.createElement(Footer, null)))
  );
}, "ThemedLayout");

// src/components/themedLayout/title/index.tsx
var import_react16 = __toESM(require("react"));
var import_core17 = require("@refinedev/core");
var import_antd10 = require("antd");
var ThemedTitle = /* @__PURE__ */ __name(({
  collapsed,
  icon: iconFromProps,
  text: textFromProps,
  wrapperStyles
}) => {
  const {
    title: { icon: defaultIcon, text: defaultText } = {}
  } = (0, import_core17.useRefineOptions)();
  const icon = typeof iconFromProps === "undefined" ? defaultIcon : iconFromProps;
  const text = typeof textFromProps === "undefined" ? defaultText : textFromProps;
  const { token } = import_antd10.theme.useToken();
  const Link = (0, import_core17.useLink)();
  return /* @__PURE__ */ import_react16.default.createElement(
    Link,
    {
      to: "/",
      style: {
        display: "inline-block",
        textDecoration: "none"
      }
    },
    /* @__PURE__ */ import_react16.default.createElement(
      import_antd10.Space,
      {
        style: {
          display: "flex",
          alignItems: "center",
          fontSize: "inherit",
          ...wrapperStyles
        }
      },
      /* @__PURE__ */ import_react16.default.createElement(
        "div",
        {
          style: {
            height: "24px",
            width: "24px",
            color: token.colorPrimary
          }
        },
        icon
      ),
      !collapsed && /* @__PURE__ */ import_react16.default.createElement(
        import_antd10.Typography.Title,
        {
          style: {
            fontSize: "inherit",
            marginBottom: 0,
            fontWeight: 700
          }
        },
        text
      )
    )
  );
}, "ThemedTitle");

// src/components/buttons/create/index.tsx
var import_react17 = __toESM(require("react"));
var import_antd11 = require("antd");
var import_icons3 = require("@ant-design/icons");
var import_core18 = require("@refinedev/core");
var import_ui_types = require("@refinedev/ui-types");
var CreateButton = /* @__PURE__ */ __name(({
  resource: resourceNameFromProps,
  hideText = false,
  accessControl,
  meta,
  children,
  onClick,
  ...rest
}) => {
  const { hidden, disabled, label, title, LinkComponent, to } = (0, import_core18.useCreateButton)(
    {
      resource: resourceNameFromProps,
      meta,
      accessControl
    }
  );
  const isDisabled = disabled || rest.disabled;
  const isHidden = hidden || rest.hidden;
  if (isHidden)
    return null;
  return /* @__PURE__ */ import_react17.default.createElement(
    LinkComponent,
    {
      to,
      replace: false,
      onClick: (e) => {
        if (isDisabled) {
          e.preventDefault();
          return;
        }
        if (onClick) {
          e.preventDefault();
          onClick(e);
        }
      }
    },
    /* @__PURE__ */ import_react17.default.createElement(
      import_antd11.Button,
      {
        icon: /* @__PURE__ */ import_react17.default.createElement(import_icons3.PlusSquareOutlined, null),
        disabled: isDisabled,
        title,
        
        className: import_ui_types.RefineButtonClassNames.CreateButton,
        type: "primary",
        ...rest
      },
      !hideText && (children ?? label)
    )
  );
}, "CreateButton");

// src/components/buttons/edit/index.tsx
var import_react18 = __toESM(require("react"));
var import_antd12 = require("antd");
var import_icons4 = require("@ant-design/icons");
var import_core19 = require("@refinedev/core");
var import_ui_types2 = require("@refinedev/ui-types");
var EditButton = /* @__PURE__ */ __name(({
  resource: resourceNameFromProps,
  recordItemId,
  hideText = false,
  accessControl,
  meta,
  children,
  onClick,
  ...rest
}) => {
  const { to, label, title, hidden, disabled, LinkComponent } = (0, import_core19.useEditButton)({
    resource: resourceNameFromProps,
    id: recordItemId,
    meta,
    accessControl
  });
  const isDisabled = disabled || rest.disabled;
  const isHidden = hidden || rest.hidden;
  if (isHidden)
    return null;
  return /* @__PURE__ */ import_react18.default.createElement(
    LinkComponent,
    {
      to,
      replace: false,
      onClick: (e) => {
        if (isDisabled) {
          e.preventDefault();
          return;
        }
        if (onClick) {
          e.preventDefault();
          onClick(e);
        }
      }
    },
    /* @__PURE__ */ import_react18.default.createElement(
      import_antd12.Button,
      {
        icon: /* @__PURE__ */ import_react18.default.createElement(import_icons4.EditOutlined, null),
        disabled: isDisabled,
        title,
        
        className: import_ui_types2.RefineButtonClassNames.EditButton,
        ...rest
      },
      !hideText && (children ?? label)
    )
  );
}, "EditButton");

// src/components/buttons/delete/index.tsx
var import_react19 = __toESM(require("react"));
var import_antd13 = require("antd");
var import_icons5 = require("@ant-design/icons");
var import_core20 = require("@refinedev/core");
var import_ui_types3 = require("@refinedev/ui-types");
var DeleteButton = /* @__PURE__ */ __name(({
  resource: resourceNameFromProps,
  recordItemId,
  onSuccess,
  mutationMode: mutationModeProp,
  children,
  successNotification,
  errorNotification,
  hideText = false,
  accessControl,
  meta,
  dataProviderName,
  confirmTitle,
  confirmOkText,
  confirmCancelText,
  invalidates,
  ...rest
}) => {
  const {
    title,
    label,
    hidden,
    disabled,
    loading,
    confirmTitle: defaultConfirmTitle,
    confirmOkLabel: defaultConfirmOkLabel,
    cancelLabel: defaultCancelLabel,
    onConfirm
  } = (0, import_core20.useDeleteButton)({
    resource: resourceNameFromProps,
    id: recordItemId,
    dataProviderName,
    invalidates,
    meta,
    onSuccess,
    mutationMode: mutationModeProp,
    errorNotification,
    successNotification,
    accessControl
  });
  const isDisabled = disabled || rest.disabled;
  const isHidden = hidden || rest.hidden;
  if (isHidden)
    return null;
  return /* @__PURE__ */ import_react19.default.createElement(
    import_antd13.Popconfirm,
    {
      key: "delete",
      okText: confirmOkText ?? defaultConfirmOkLabel,
      cancelText: confirmCancelText ?? defaultCancelLabel,
      okType: "danger",
      title: confirmTitle ?? defaultConfirmTitle,
      okButtonProps: { disabled: loading },
      onConfirm,
      disabled: isDisabled
    },
    /* @__PURE__ */ import_react19.default.createElement(
      import_antd13.Button,
      {
        danger: true,
        loading,
        icon: /* @__PURE__ */ import_react19.default.createElement(import_icons5.DeleteOutlined, null),
        title,
        disabled: isDisabled,
        
        className: import_ui_types3.RefineButtonClassNames.DeleteButton,
        ...rest
      },
      !hideText && (children ?? label)
    )
  );
}, "DeleteButton");

// src/components/buttons/refresh/index.tsx
var import_react20 = __toESM(require("react"));
var import_antd14 = require("antd");
var import_icons6 = require("@ant-design/icons");
var import_core21 = require("@refinedev/core");
var import_ui_types4 = require("@refinedev/ui-types");
var RefreshButton = /* @__PURE__ */ __name(({
  resource: resourceNameFromProps,
  recordItemId,
  hideText = false,
  dataProviderName,
  children,
  ...rest
}) => {
  const { onClick, label, loading } = (0, import_core21.useRefreshButton)({
    resource: resourceNameFromProps,
    id: recordItemId,
    dataProviderName
  });
  return /* @__PURE__ */ import_react20.default.createElement(
    import_antd14.Button,
    {
      onClick,
      icon: /* @__PURE__ */ import_react20.default.createElement(import_icons6.RedoOutlined, { spin: loading }),
      
      className: import_ui_types4.RefineButtonClassNames.RefreshButton,
      ...rest
    },
    !hideText && (children ?? label)
  );
}, "RefreshButton");

// src/components/buttons/show/index.tsx
var import_react21 = __toESM(require("react"));
var import_antd15 = require("antd");
var import_icons7 = require("@ant-design/icons");
var import_core22 = require("@refinedev/core");
var import_ui_types5 = require("@refinedev/ui-types");
var ShowButton = /* @__PURE__ */ __name(({
  resource: resourceNameFromProps,
  recordItemId,
  hideText = false,
  accessControl,
  meta,
  children,
  onClick,
  ...rest
}) => {
  const { to, label, title, hidden, disabled, LinkComponent } = (0, import_core22.useShowButton)({
    resource: resourceNameFromProps,
    id: recordItemId,
    accessControl,
    meta
  });
  const isDisabled = disabled || rest.disabled;
  const isHidden = hidden || rest.hidden;
  if (isHidden)
    return null;
  return /* @__PURE__ */ import_react21.default.createElement(
    LinkComponent,
    {
      to,
      replace: false,
      onClick: (e) => {
        if (isDisabled) {
          e.preventDefault();
          return;
        }
        if (onClick) {
          e.preventDefault();
          onClick(e);
        }
      }
    },
    /* @__PURE__ */ import_react21.default.createElement(
      import_antd15.Button,
      {
        disabled: isDisabled,
        icon: /* @__PURE__ */ import_react21.default.createElement(import_icons7.EyeOutlined, null),
        title,
        
        className: import_ui_types5.RefineButtonClassNames.ShowButton,
        ...rest
      },
      !hideText && (children ?? label)
    )
  );
}, "ShowButton");

// src/components/buttons/list/index.tsx
var import_react22 = __toESM(require("react"));
var import_antd16 = require("antd");
var import_icons8 = require("@ant-design/icons");
var import_core23 = require("@refinedev/core");
var import_ui_types6 = require("@refinedev/ui-types");
var ListButton = /* @__PURE__ */ __name(({
  resource: resourceNameFromProps,
  hideText = false,
  accessControl,
  meta,
  children,
  onClick,
  ...rest
}) => {
  const { to, label, title, hidden, disabled, LinkComponent } = (0, import_core23.useListButton)({
    resource: resourceNameFromProps,
    meta,
    accessControl
  });
  const isDisabled = disabled || rest.disabled;
  const isHidden = hidden || rest.hidden;
  if (isHidden)
    return null;
  return /* @__PURE__ */ import_react22.default.createElement(
    LinkComponent,
    {
      to,
      replace: false,
      onClick: (e) => {
        if (isDisabled) {
          e.preventDefault();
          return;
        }
        if (onClick) {
          e.preventDefault();
          onClick(e);
        }
      }
    },
    /* @__PURE__ */ import_react22.default.createElement(
      import_antd16.Button,
      {
        icon: /* @__PURE__ */ import_react22.default.createElement(import_icons8.BarsOutlined, null),
        disabled: isDisabled,
        title,
        
        className: import_ui_types6.RefineButtonClassNames.ListButton,
        ...rest
      },
      !hideText && (children ?? label)
    )
  );
}, "ListButton");

// src/components/buttons/export/index.tsx
var import_react23 = __toESM(require("react"));
var import_antd17 = require("antd");
var import_icons9 = require("@ant-design/icons");
var import_core24 = require("@refinedev/core");
var import_ui_types7 = require("@refinedev/ui-types");
var ExportButton = /* @__PURE__ */ __name(({
  hideText = false,
  children,
  ...rest
}) => {
  const { label } = (0, import_core24.useExportButton)();
  return /* @__PURE__ */ import_react23.default.createElement(
    import_antd17.Button,
    {
      type: "default",
      icon: /* @__PURE__ */ import_react23.default.createElement(import_icons9.ExportOutlined, null),
      
      className: import_ui_types7.RefineButtonClassNames.ExportButton,
      ...rest
    },
    !hideText && (children ?? label)
  );
}, "ExportButton");

// src/components/buttons/save/index.tsx
var import_react24 = __toESM(require("react"));
var import_antd18 = require("antd");
var import_icons10 = require("@ant-design/icons");
var import_core25 = require("@refinedev/core");
var import_ui_types8 = require("@refinedev/ui-types");
var SaveButton = /* @__PURE__ */ __name(({
  hideText = false,
  children,
  ...rest
}) => {
  const { label } = (0, import_core25.useSaveButton)();
  return /* @__PURE__ */ import_react24.default.createElement(
    import_antd18.Button,
    {
      type: "primary",
      icon: /* @__PURE__ */ import_react24.default.createElement(import_icons10.SaveOutlined, null),
      
      className: import_ui_types8.RefineButtonClassNames.SaveButton,
      ...rest
    },
    !hideText && (children ?? label)
  );
}, "SaveButton");

// src/components/buttons/clone/index.tsx
var import_react25 = __toESM(require("react"));
var import_antd19 = require("antd");
var import_icons11 = require("@ant-design/icons");
var import_core26 = require("@refinedev/core");
var import_ui_types9 = require("@refinedev/ui-types");
var CloneButton = /* @__PURE__ */ __name(({
  resource: resourceNameFromProps,
  recordItemId,
  hideText = false,
  accessControl,
  meta,
  children,
  onClick,
  ...rest
}) => {
  const { to, LinkComponent, label, disabled, hidden, title } = (0, import_core26.useCloneButton)({
    id: recordItemId,
    resource: resourceNameFromProps,
    accessControl,
    meta
  });
  const isDisabled = disabled || rest.disabled;
  const isHidden = hidden || rest.hidden;
  if (isHidden)
    return null;
  return /* @__PURE__ */ import_react25.default.createElement(
    LinkComponent,
    {
      to,
      replace: false,
      onClick: (e) => {
        if (isDisabled) {
          e.preventDefault();
          return;
        }
        if (onClick) {
          e.preventDefault();
          onClick(e);
        }
      }
    },
    /* @__PURE__ */ import_react25.default.createElement(
      import_antd19.Button,
      {
        icon: /* @__PURE__ */ import_react25.default.createElement(import_icons11.PlusSquareOutlined, null),
        disabled: isDisabled,
        title,
        
        className: import_ui_types9.RefineButtonClassNames.CloneButton,
        ...rest
      },
      !hideText && (children ?? label)
    )
  );
}, "CloneButton");

// src/components/buttons/import/index.tsx
var import_react26 = __toESM(require("react"));
var import_antd20 = require("antd");
var import_icons12 = require("@ant-design/icons");
var import_core27 = require("@refinedev/core");
var import_ui_types10 = require("@refinedev/ui-types");
var ImportButton = /* @__PURE__ */ __name(({
  uploadProps,
  buttonProps,
  hideText = false,
  children,
  loading,
  ...rest
}) => {
  const { label } = (0, import_core27.useImportButton)();
  return /* @__PURE__ */ import_react26.default.createElement(import_antd20.Upload, { ...uploadProps }, /* @__PURE__ */ import_react26.default.createElement(
    import_antd20.Button,
    {
      icon: /* @__PURE__ */ import_react26.default.createElement(import_icons12.ImportOutlined, null),
      
      className: import_ui_types10.RefineButtonClassNames.ImportButton,
      loading,
      ...buttonProps,
      ...rest
    },
    !hideText && (children ?? label)
  ));
}, "ImportButton");

// src/components/crud/list/index.tsx
var import_react27 = __toESM(require("react"));
var import_antd21 = require("antd");
var import_core28 = require("@refinedev/core");
var List = /* @__PURE__ */ __name(({
  canCreate,
  title,
  children,
  createButtonProps: createButtonPropsFromProps,
  resource: resourceFromProps,
  wrapperProps,
  contentProps,
  headerProps,
  breadcrumb: breadcrumbFromProps,
  headerButtonProps,
  headerButtons
}) => {
  var _a;
  const translate = (0, import_core28.useTranslate)();
  const {
    options: { breadcrumb: globalBreadcrumb } = {}
  } = (0, import_core28.useRefineContext)();
  const getUserFriendlyName = (0, import_core28.useUserFriendlyName)();
  const { resource, identifier } = (0, import_core28.useResourceParams)({
    resource: resourceFromProps
  });
  const isCreateButtonVisible = canCreate ?? (!!(resource == null ? void 0 : resource.create) || !!createButtonPropsFromProps);
  const breadcrumb = typeof breadcrumbFromProps === "undefined" ? globalBreadcrumb : breadcrumbFromProps;
  const createButtonProps = isCreateButtonVisible ? {
    size: "middle",
    resource: identifier,
    ...createButtonPropsFromProps
  } : void 0;
  const defaultExtra = isCreateButtonVisible ? /* @__PURE__ */ import_react27.default.createElement(CreateButton, { ...createButtonProps }) : null;
  return /* @__PURE__ */ import_react27.default.createElement("div", { ...wrapperProps ?? {} }, /* @__PURE__ */ import_react27.default.createElement(
    PageHeader,
    {
      title: title ?? translate(
        `${identifier}.titles.list`,
        getUserFriendlyName(((_a = resource == null ? void 0 : resource.meta) == null ? void 0 : _a.label) ?? identifier, "plural")
      ),
      extra: headerButtons ? /* @__PURE__ */ import_react27.default.createElement(import_antd21.Space, { wrap: true, ...headerButtonProps }, typeof headerButtons === "function" ? headerButtons({
        defaultButtons: defaultExtra,
        createButtonProps
      }) : headerButtons) : defaultExtra,
      breadcrumb: typeof breadcrumb !== "undefined" ? /* @__PURE__ */ import_react27.default.createElement(import_react27.default.Fragment, null, breadcrumb) : /* @__PURE__ */ import_react27.default.createElement(Breadcrumb, null),
      ...headerProps ?? {}
    },
    /* @__PURE__ */ import_react27.default.createElement("div", { ...contentProps ?? {} }, children)
  ));
}, "List");

// src/components/crud/create/index.tsx
var import_react28 = __toESM(require("react"));
var import_antd22 = require("antd");
var import_core29 = require("@refinedev/core");
var Create = /* @__PURE__ */ __name(({
  title,
  saveButtonProps: saveButtonPropsFromProps,
  children,
  resource: resourceFromProps,
  isLoading = false,
  breadcrumb: breadcrumbFromProps,
  wrapperProps,
  headerProps,
  contentProps,
  headerButtonProps,
  headerButtons,
  footerButtonProps,
  footerButtons,
  goBack: goBackFromProps
}) => {
  var _a;
  const translate = (0, import_core29.useTranslate)();
  const {
    options: { breadcrumb: globalBreadcrumb } = {}
  } = (0, import_core29.useRefineContext)();
  const back = (0, import_core29.useBack)();
  const getUserFriendlyName = (0, import_core29.useUserFriendlyName)();
  const { resource, identifier } = (0, import_core29.useResourceParams)({
    resource: resourceFromProps
  });
  const breadcrumb = typeof breadcrumbFromProps === "undefined" ? globalBreadcrumb : breadcrumbFromProps;
  const saveButtonProps = {
    ...isLoading ? { disabled: true } : {},
    ...saveButtonPropsFromProps,
    htmlType: "submit"
  };
  const defaultFooterButtons = /* @__PURE__ */ import_react28.default.createElement(import_react28.default.Fragment, null, /* @__PURE__ */ import_react28.default.createElement(SaveButton, { ...saveButtonProps }));
  return /* @__PURE__ */ import_react28.default.createElement("div", { ...wrapperProps ?? {} }, /* @__PURE__ */ import_react28.default.createElement(
    PageHeader,
    {
      backIcon: goBackFromProps,
      onBack: back,
      title: title ?? translate(
        `${identifier}.titles.create`,
        `Create ${getUserFriendlyName(
          ((_a = resource == null ? void 0 : resource.meta) == null ? void 0 : _a.label) ?? identifier,
          "singular"
        )}`
      ),
      breadcrumb: typeof breadcrumb !== "undefined" ? /* @__PURE__ */ import_react28.default.createElement(import_react28.default.Fragment, null, breadcrumb) : /* @__PURE__ */ import_react28.default.createElement(Breadcrumb, null),
      extra: /* @__PURE__ */ import_react28.default.createElement(import_antd22.Space, { wrap: true, ...headerButtonProps ?? {} }, headerButtons ? typeof headerButtons === "function" ? headerButtons({
        defaultButtons: null
      }) : headerButtons : null),
      ...headerProps ?? {}
    },
    /* @__PURE__ */ import_react28.default.createElement(import_antd22.Spin, { spinning: isLoading }, /* @__PURE__ */ import_react28.default.createElement(
      import_antd22.Card,
      {
        variant: "borderless",
        actions: [
          /* @__PURE__ */ import_react28.default.createElement(
            import_antd22.Space,
            {
              key: "action-buttons",
              style: { float: "right", marginRight: 24 },
              ...footerButtonProps ?? {}
            },
            footerButtons ? typeof footerButtons === "function" ? footerButtons({
              defaultButtons: defaultFooterButtons,
              saveButtonProps
            }) : footerButtons : defaultFooterButtons
          )
        ],
        ...contentProps ?? {}
      },
      children
    ))
  ));
}, "Create");

// src/components/crud/edit/index.tsx
var import_react29 = __toESM(require("react"));
var import_antd23 = require("antd");
var import_core30 = require("@refinedev/core");
var Edit = /* @__PURE__ */ __name(({
  title,
  saveButtonProps: saveButtonPropsFromProps,
  mutationMode: mutationModeProp,
  recordItemId,
  children,
  deleteButtonProps: deleteButtonPropsFromProps,
  canDelete,
  resource: resourceFromProps,
  isLoading = false,
  dataProviderName,
  breadcrumb: breadcrumbFromProps,
  wrapperProps,
  headerProps,
  contentProps,
  headerButtonProps,
  headerButtons,
  footerButtonProps,
  footerButtons,
  goBack: goBackFromProps,
  autoSaveProps
}) => {
  var _a, _b;
  const translate = (0, import_core30.useTranslate)();
  const {
    options: { breadcrumb: globalBreadcrumb } = {}
  } = (0, import_core30.useRefineContext)();
  const { mutationMode: mutationModeContext } = (0, import_core30.useMutationMode)();
  const mutationMode = mutationModeProp ?? mutationModeContext;
  const back = (0, import_core30.useBack)();
  const go = (0, import_core30.useGo)();
  const getUserFriendlyName = (0, import_core30.useUserFriendlyName)();
  const {
    resource,
    action,
    id: idFromParams,
    identifier
  } = (0, import_core30.useResourceParams)({
    resource: resourceFromProps
  });
  const goListPath = (0, import_core30.useToPath)({
    resource,
    action: "list"
  });
  const id = recordItemId ?? idFromParams;
  const breadcrumb = typeof breadcrumbFromProps === "undefined" ? globalBreadcrumb : breadcrumbFromProps;
  const hasList = (resource == null ? void 0 : resource.list) && !recordItemId;
  const isDeleteButtonVisible = canDelete ?? (((_a = resource == null ? void 0 : resource.meta) == null ? void 0 : _a.canDelete) || deleteButtonPropsFromProps);
  const listButtonProps = hasList ? {
    ...isLoading ? { disabled: true } : {},
    resource: identifier
  } : void 0;
  const refreshButtonProps = {
    ...isLoading ? { disabled: true } : {},
    resource: identifier,
    recordItemId: id,
    dataProviderName
  };
  const deleteButtonProps = isDeleteButtonVisible ? {
    ...isLoading ? { disabled: true } : {},
    resource: identifier,
    mutationMode,
    onSuccess: () => {
      go({ to: goListPath });
    },
    recordItemId: id,
    dataProviderName,
    ...deleteButtonPropsFromProps
  } : void 0;
  const saveButtonProps = {
    ...isLoading ? { disabled: true } : {},
    ...saveButtonPropsFromProps
  };
  const defaultHeaderButtons = /* @__PURE__ */ import_react29.default.createElement(import_react29.default.Fragment, null, autoSaveProps && /* @__PURE__ */ import_react29.default.createElement(AutoSaveIndicator, { ...autoSaveProps }), hasList && /* @__PURE__ */ import_react29.default.createElement(ListButton, { ...listButtonProps }), /* @__PURE__ */ import_react29.default.createElement(RefreshButton, { ...refreshButtonProps }));
  const defaultFooterButtons = /* @__PURE__ */ import_react29.default.createElement(import_react29.default.Fragment, null, isDeleteButtonVisible && /* @__PURE__ */ import_react29.default.createElement(DeleteButton, { ...deleteButtonProps }), /* @__PURE__ */ import_react29.default.createElement(SaveButton, { ...saveButtonProps }));
  return /* @__PURE__ */ import_react29.default.createElement("div", { ...wrapperProps ?? {} }, /* @__PURE__ */ import_react29.default.createElement(
    PageHeader,
    {
      backIcon: goBackFromProps,
      onBack: action !== "list" && typeof action !== "undefined" ? back : void 0,
      title: title ?? translate(
        `${identifier}.titles.edit`,
        `Edit ${getUserFriendlyName(
          ((_b = resource == null ? void 0 : resource.meta) == null ? void 0 : _b.label) ?? identifier,
          "singular"
        )}`
      ),
      extra: /* @__PURE__ */ import_react29.default.createElement(import_antd23.Space, { wrap: true, ...headerButtonProps ?? {} }, headerButtons ? typeof headerButtons === "function" ? headerButtons({
        defaultButtons: defaultHeaderButtons,
        listButtonProps,
        refreshButtonProps
      }) : headerButtons : defaultHeaderButtons),
      breadcrumb: typeof breadcrumb !== "undefined" ? /* @__PURE__ */ import_react29.default.createElement(import_react29.default.Fragment, null, breadcrumb) : /* @__PURE__ */ import_react29.default.createElement(Breadcrumb, null),
      ...headerProps ?? {}
    },
    /* @__PURE__ */ import_react29.default.createElement(import_antd23.Spin, { spinning: isLoading }, /* @__PURE__ */ import_react29.default.createElement(
      import_antd23.Card,
      {
        variant: "borderless",
        actions: [
          /* @__PURE__ */ import_react29.default.createElement(
            import_antd23.Space,
            {
              key: "footer-buttons",
              wrap: true,
              style: {
                float: "right",
                marginRight: 24
              },
              ...footerButtonProps ?? {}
            },
            footerButtons ? typeof footerButtons === "function" ? footerButtons({
              defaultButtons: defaultFooterButtons,
              deleteButtonProps,
              saveButtonProps
            }) : footerButtons : defaultFooterButtons
          )
        ],
        ...contentProps ?? {}
      },
      children
    ))
  ));
}, "Edit");

// src/components/crud/show/index.tsx
var import_react30 = __toESM(require("react"));
var import_antd24 = require("antd");
var import_core31 = require("@refinedev/core");
var Show = /* @__PURE__ */ __name(({
  title,
  canEdit,
  canDelete,
  deleteButtonProps: deleteButtonPropsFromProps,
  isLoading = false,
  children,
  resource: resourceFromProps,
  recordItemId,
  dataProviderName,
  breadcrumb: breadcrumbFromProps,
  contentProps,
  headerProps,
  wrapperProps,
  headerButtons,
  footerButtons,
  footerButtonProps,
  headerButtonProps,
  goBack: goBackFromProps
}) => {
  var _a, _b;
  const translate = (0, import_core31.useTranslate)();
  const {
    options: { breadcrumb: globalBreadcrumb } = {}
  } = (0, import_core31.useRefineContext)();
  const back = (0, import_core31.useBack)();
  const go = (0, import_core31.useGo)();
  const getUserFriendlyName = (0, import_core31.useUserFriendlyName)();
  const {
    resource,
    action,
    id: idFromParams,
    identifier
  } = (0, import_core31.useResourceParams)({
    resource: resourceFromProps
  });
  const goListPath = (0, import_core31.useToPath)({
    resource,
    action: "list"
  });
  const id = recordItemId ?? idFromParams;
  const breadcrumb = typeof breadcrumbFromProps === "undefined" ? globalBreadcrumb : breadcrumbFromProps;
  const hasList = (resource == null ? void 0 : resource.list) && !recordItemId;
  const isDeleteButtonVisible = canDelete ?? (((_a = resource == null ? void 0 : resource.meta) == null ? void 0 : _a.canDelete) || deleteButtonPropsFromProps);
  const isEditButtonVisible = canEdit ?? !!(resource == null ? void 0 : resource.edit);
  const listButtonProps = hasList ? {
    resource: identifier
  } : void 0;
  const editButtonProps = isEditButtonVisible ? {
    ...isLoading ? { disabled: true } : {},
    type: "primary",
    resource: identifier,
    recordItemId: id
  } : void 0;
  const deleteButtonProps = isDeleteButtonVisible ? {
    ...isLoading ? { disabled: true } : {},
    resource: identifier,
    recordItemId: id,
    onSuccess: () => {
      go({ to: goListPath });
    },
    dataProviderName,
    ...deleteButtonPropsFromProps
  } : void 0;
  const refreshButtonProps = {
    ...isLoading ? { disabled: true } : {},
    resource: identifier,
    recordItemId: id,
    dataProviderName
  };
  const defaultHeaderButtons = /* @__PURE__ */ import_react30.default.createElement(import_react30.default.Fragment, null, hasList && /* @__PURE__ */ import_react30.default.createElement(ListButton, { ...listButtonProps }), isEditButtonVisible && /* @__PURE__ */ import_react30.default.createElement(EditButton, { ...editButtonProps }), isDeleteButtonVisible && /* @__PURE__ */ import_react30.default.createElement(DeleteButton, { ...deleteButtonProps }), /* @__PURE__ */ import_react30.default.createElement(RefreshButton, { ...refreshButtonProps }));
  return /* @__PURE__ */ import_react30.default.createElement("div", { ...wrapperProps ?? {} }, /* @__PURE__ */ import_react30.default.createElement(
    PageHeader,
    {
      backIcon: goBackFromProps,
      onBack: action !== "list" && typeof action !== "undefined" ? back : void 0,
      title: title ?? translate(
        `${identifier}.titles.show`,
        `Show ${getUserFriendlyName(
          ((_b = resource == null ? void 0 : resource.meta) == null ? void 0 : _b.label) ?? identifier,
          "singular"
        )}`
      ),
      extra: /* @__PURE__ */ import_react30.default.createElement(import_antd24.Space, { key: "extra-buttons", wrap: true, ...headerButtonProps ?? {} }, headerButtons ? typeof headerButtons === "function" ? headerButtons({
        defaultButtons: defaultHeaderButtons,
        deleteButtonProps,
        editButtonProps,
        listButtonProps,
        refreshButtonProps
      }) : headerButtons : defaultHeaderButtons),
      breadcrumb: typeof breadcrumb !== "undefined" ? /* @__PURE__ */ import_react30.default.createElement(import_react30.default.Fragment, null, breadcrumb) : /* @__PURE__ */ import_react30.default.createElement(Breadcrumb, null),
      ...headerProps ?? {}
    },
    /* @__PURE__ */ import_react30.default.createElement(import_antd24.Spin, { spinning: isLoading }, /* @__PURE__ */ import_react30.default.createElement(
      import_antd24.Card,
      {
        variant: "borderless",
        actions: footerButtons ? [
          /* @__PURE__ */ import_react30.default.createElement(import_antd24.Space, { key: "footer-buttons", wrap: true, ...footerButtonProps }, typeof footerButtons === "function" ? footerButtons({
            defaultButtons: null
          }) : footerButtons)
        ] : void 0,
        ...contentProps ?? {}
      },
      children
    ))
  ));
}, "Show");

// src/components/fields/text/index.tsx
var import_react31 = __toESM(require("react"));
var import_antd25 = require("antd");
var TextField = /* @__PURE__ */ __name(({ value, ...rest }) => {
  return /* @__PURE__ */ import_react31.default.createElement(import_antd25.Typography.Text, { ...rest }, value);
}, "TextField");

// src/components/fields/tag/index.tsx
var import_react32 = __toESM(require("react"));
var import_antd26 = require("antd");
var TagField = /* @__PURE__ */ __name(({ value, ...rest }) => {
  return /* @__PURE__ */ import_react32.default.createElement(import_antd26.Tag, { ...rest }, value == null ? void 0 : value.toString());
}, "TagField");

// src/components/fields/email/index.tsx
var import_react33 = __toESM(require("react"));
var import_antd27 = require("antd");
var EmailField = /* @__PURE__ */ __name(({ value, ...rest }) => {
  return /* @__PURE__ */ import_react33.default.createElement(import_antd27.Typography.Link, { href: `mailto:${value}`, ...rest }, value);
}, "EmailField");

// src/components/fields/image/index.tsx
var import_react34 = __toESM(require("react"));
var import_antd28 = require("antd");
var ImageField = /* @__PURE__ */ __name(({
  value,
  imageTitle,
  ...rest
}) => {
  return /* @__PURE__ */ import_react34.default.createElement(import_antd28.Image, { ...rest, src: value, title: imageTitle });
}, "ImageField");

// src/components/fields/boolean/index.tsx
var import_react35 = __toESM(require("react"));
var import_antd29 = require("antd");
var import_icons13 = require("@ant-design/icons");
var BooleanField = /* @__PURE__ */ __name(({
  value,
  valueLabelTrue = "true",
  valueLabelFalse = "false",
  trueIcon = /* @__PURE__ */ import_react35.default.createElement(import_icons13.CheckOutlined, null),
  falseIcon = /* @__PURE__ */ import_react35.default.createElement(import_icons13.CloseOutlined, null),
  ...rest
}) => {
  return /* @__PURE__ */ import_react35.default.createElement(import_antd29.Tooltip, { title: value ? valueLabelTrue : valueLabelFalse, ...rest }, value ? /* @__PURE__ */ import_react35.default.createElement("span", null, trueIcon) : /* @__PURE__ */ import_react35.default.createElement("span", null, falseIcon));
}, "BooleanField");

// src/components/fields/date/index.tsx
var import_react36 = __toESM(require("react"));
var import_dayjs = __toESM(require("dayjs"));
var import_antd30 = require("antd");
var import_localizedFormat = __toESM(require("dayjs/plugin/localizedFormat"));
var DateField = /* @__PURE__ */ __name(({
  value,
  locales,
  format: dateFormat = "L",
  ...rest
}) => {
  import_dayjs.default.extend(import_localizedFormat.default);
  const defaultLocale = import_dayjs.default.locale();
  return /* @__PURE__ */ import_react36.default.createElement(import_antd30.Typography.Text, { ...rest }, value ? (0, import_dayjs.default)(value).locale(locales || defaultLocale).format(dateFormat) : "");
}, "DateField");

// src/components/fields/file/index.tsx
var import_react37 = __toESM(require("react"));
var FileField = /* @__PURE__ */ __name(({
  title,
  src,
  ...rest
}) => {
  return /* @__PURE__ */ import_react37.default.createElement(UrlField, { value: src, title, ...rest }, title ?? src);
}, "FileField");

// src/components/fields/url/index.tsx
var import_react38 = __toESM(require("react"));
var import_antd31 = require("antd");
var UrlField = /* @__PURE__ */ __name(({
  children,
  value,
  ...rest
}) => {
  return /* @__PURE__ */ import_react38.default.createElement(import_antd31.Typography.Link, { href: value, ...rest }, children ?? value);
}, "UrlField");

// src/components/fields/number/index.tsx
var import_react39 = __toESM(require("react"));
var import_antd32 = require("antd");
function toLocaleStringSupportsOptions() {
  return !!(typeof Intl === "object" && Intl && typeof Intl.NumberFormat === "function");
}
__name(toLocaleStringSupportsOptions, "toLocaleStringSupportsOptions");
var NumberField = /* @__PURE__ */ __name(({
  value,
  locale,
  options,
  ...rest
}) => {
  const number = Number(value);
  return /* @__PURE__ */ import_react39.default.createElement(import_antd32.Typography.Text, { ...rest }, toLocaleStringSupportsOptions() ? number.toLocaleString(locale, options) : number);
}, "NumberField");

// src/components/fields/markdown/index.tsx
var import_react40 = __toESM(require("react"));
var import_react_markdown = __toESM(require("react-markdown"));
var import_remark_gfm = __toESM(require("remark-gfm"));
var MarkdownField = /* @__PURE__ */ __name(({
  value = ""
}) => {
  return /* @__PURE__ */ import_react40.default.createElement(
    import_react_markdown.default,
    {
      remarkPlugins: [import_remark_gfm.default]
    },
    value
  );
}, "MarkdownField");

// src/components/table/components/filterDropdown/index.tsx
var import_react41 = __toESM(require("react"));
var import_antd33 = require("antd");
var import_dayjs2 = __toESM(require("dayjs"));
var import_icons14 = require("@ant-design/icons");
var import_core32 = require("@refinedev/core");
var FilterDropdown = /* @__PURE__ */ __name((props) => {
  const {
    setSelectedKeys,
    confirm,
    clearFilters,
    mapValue = /* @__PURE__ */ __name((value) => value, "mapValue"),
    selectedKeys,
    children
  } = props;
  const translate = (0, import_core32.useTranslate)();
  const clearFilter = /* @__PURE__ */ __name(() => {
    if (clearFilters) {
      clearFilters();
    }
  }, "clearFilter");
  const onFilter = /* @__PURE__ */ __name(() => {
    let keys;
    if (typeof selectedKeys === "number") {
      keys = `${selectedKeys}`;
    } else if (import_dayjs2.default.isDayjs(selectedKeys)) {
      keys = [selectedKeys.toISOString()];
    } else {
      keys = selectedKeys;
    }
    setSelectedKeys(keys);
    confirm == null ? void 0 : confirm();
  }, "onFilter");
  const onChange = /* @__PURE__ */ __name((e) => {
    if (typeof e === "object") {
      if (Array.isArray(e)) {
        const mappedValue3 = mapValue(e, "onChange");
        return setSelectedKeys(mappedValue3);
      }
      const changeEvent = !e || !e.target || import_dayjs2.default.isDayjs(e) ? { target: { value: e } } : e;
      const { target } = changeEvent;
      const mappedValue2 = mapValue(target.value, "onChange");
      setSelectedKeys(mappedValue2);
      return;
    }
    const mappedValue = mapValue(e, "onChange");
    setSelectedKeys(mappedValue);
  }, "onChange");
  const childrenWithProps = import_react41.default.Children.map(children, (child) => {
    if (import_react41.default.isValidElement(child)) {
      return import_react41.default.cloneElement(child, {
        onChange,
        value: mapValue(selectedKeys, "value")
      });
    }
    return child;
  });
  return /* @__PURE__ */ import_react41.default.createElement(
    "div",
    {
      style: {
        padding: 10,
        display: "flex",
        flexDirection: "column",
        alignItems: "flex-end"
      }
    },
    /* @__PURE__ */ import_react41.default.createElement("div", { style: { marginBottom: 15 } }, childrenWithProps),
    /* @__PURE__ */ import_react41.default.createElement(import_antd33.Space, null, /* @__PURE__ */ import_react41.default.createElement(import_antd33.Button, { type: "primary", size: "small", onClick: () => onFilter() }, /* @__PURE__ */ import_react41.default.createElement(import_icons14.FilterOutlined, null), " ", translate("buttons.filter", "Filter")), /* @__PURE__ */ import_react41.default.createElement(import_antd33.Button, { danger: true, size: "small", onClick: () => clearFilter() }, translate("buttons.clear", "Clear")))
  );
}, "FilterDropdown");

// src/components/pages/error/index.tsx
var import_react42 = __toESM(require("react"));
var import_core33 = require("@refinedev/core");
var import_antd34 = require("antd");
var import_icons15 = require("@ant-design/icons");
var import_core34 = require("@refinedev/core");
var ErrorComponent = /* @__PURE__ */ __name(() => {
  const [errorMessage, setErrorMessage] = (0, import_react42.useState)();
  const translate = (0, import_core34.useTranslate)();
  const go = (0, import_core33.useGo)();
  const { resource, action } = (0, import_core33.useResourceParams)();
  (0, import_react42.useEffect)(() => {
    if (resource) {
      if (action) {
        setErrorMessage(
          translate(
            "pages.error.info",
            {
              action,
              resource: resource == null ? void 0 : resource.name
            },
            `You may have forgotten to add the "${action}" component to "${resource == null ? void 0 : resource.name}" resource.`
          )
        );
      }
    }
  }, [resource, action]);
  return /* @__PURE__ */ import_react42.default.createElement(
    import_antd34.Result,
    {
      status: "404",
      title: "404",
      extra: /* @__PURE__ */ import_react42.default.createElement(import_antd34.Space, { direction: "vertical", size: "large" }, /* @__PURE__ */ import_react42.default.createElement(import_antd34.Space, null, /* @__PURE__ */ import_react42.default.createElement(import_antd34.Typography.Text, null, translate(
        "pages.error.404",
        "Sorry, the page you visited does not exist."
      )), errorMessage && /* @__PURE__ */ import_react42.default.createElement(import_antd34.Tooltip, { title: errorMessage }, /* @__PURE__ */ import_react42.default.createElement(import_icons15.InfoCircleOutlined, { }))), /* @__PURE__ */ import_react42.default.createElement(import_antd34.Button, { type: "primary", onClick: () => go({ to: "/" }) }, translate("pages.error.backHome", "Back Home")))
    }
  );
}, "ErrorComponent");

// src/components/pages/welcome/index.tsx
var import_react43 = __toESM(require("react"));
var import_core35 = require("@refinedev/core");
var WelcomePage = /* @__PURE__ */ __name(() => {
  return /* @__PURE__ */ import_react43.default.createElement(import_core35.WelcomePage, null);
}, "WelcomePage");

// src/components/pages/auth/index.tsx
var import_react48 = __toESM(require("react"));

// src/components/pages/auth/components/login/index.tsx
var import_react44 = __toESM(require("react"));
var import_core36 = require("@refinedev/core");
var import_antd35 = require("antd");
var import_core37 = require("@refinedev/core");

// src/components/pages/auth/components/styles.ts
var layoutStyles = {};
var containerStyles = {
  maxWidth: "400px",
  margin: "auto",
  padding: "32px",
  boxShadow: "0px 2px 4px rgba(0, 0, 0, 0.02), 0px 1px 6px -1px rgba(0, 0, 0, 0.02), 0px 1px 2px rgba(0, 0, 0, 0.03)"
};
var headStyles = {
  borderBottom: 0,
  padding: 0
};
var bodyStyles = { padding: 0, marginTop: "32px" };
var titleStyles = {
  textAlign: "center",
  marginBottom: 0,
  fontSize: "24px",
  lineHeight: "32px",
  fontWeight: 700,
  overflowWrap: "break-word",
  hyphens: "manual",
  textOverflow: "unset",
  whiteSpace: "pre-wrap"
};

// src/components/pages/auth/components/login/index.tsx
var LoginPage = /* @__PURE__ */ __name(({
  providers,
  registerLink,
  forgotPasswordLink,
  rememberMe,
  contentProps,
  wrapperProps,
  renderContent,
  formProps,
  title,
  hideForm,
  mutationVariables
}) => {
  const { token } = import_antd35.theme.useToken();
  const [form] = import_antd35.Form.useForm();
  const translate = (0, import_core37.useTranslate)();
  const Link = (0, import_core36.useLink)();
  const { mutate: login, isPending } = (0, import_core37.useLogin)();
  const PageTitle = title === false ? null : /* @__PURE__ */ import_react44.default.createElement(
    "div",
    {
      style: {
        display: "flex",
        justifyContent: "center",
        marginBottom: "32px",
        fontSize: "20px"
      }
    },
    title ?? /* @__PURE__ */ import_react44.default.createElement(ThemedTitle, { collapsed: false })
  );
  const CardTitle = /* @__PURE__ */ import_react44.default.createElement(
    import_antd35.Typography.Title,
    {
      level: 3,
      style: {
        color: token.colorPrimaryTextHover,
        ...titleStyles
      }
    },
    translate("pages.login.title", "Sign in to your account")
  );
  const renderProviders = /* @__PURE__ */ __name(() => {
    if (providers && providers.length > 0) {
      return /* @__PURE__ */ import_react44.default.createElement(import_react44.default.Fragment, null, providers.map((provider) => {
        return /* @__PURE__ */ import_react44.default.createElement(
          import_antd35.Button,
          {
            key: provider.name,
            type: "default",
            block: true,
            icon: provider.icon,
            style: {
              display: "flex",
              justifyContent: "center",
              alignItems: "center",
              width: "100%",
              marginBottom: "8px"
            },
            onClick: () => login({
              ...mutationVariables,
              providerName: provider.name
            })
          },
          provider.label
        );
      }), !hideForm && /* @__PURE__ */ import_react44.default.createElement(import_antd35.Divider, null, /* @__PURE__ */ import_react44.default.createElement(
        import_antd35.Typography.Text,
        {
          style: {
            color: token.colorTextLabel
          }
        },
        translate("pages.login.divider", "or")
      )));
    }
    return null;
  }, "renderProviders");
  const CardContent = /* @__PURE__ */ import_react44.default.createElement(
    import_antd35.Card,
    {
      title: CardTitle,
      styles: {
        header: headStyles,
        body: bodyStyles
      },
      style: {
        ...containerStyles,
        backgroundColor: token.colorBgElevated
      },
      ...contentProps ?? {}
    },
    renderProviders(),
    !hideForm && /* @__PURE__ */ import_react44.default.createElement(
      import_antd35.Form,
      {
        layout: "vertical",
        form,
        onFinish: (values) => login({ ...values, ...mutationVariables }),
        requiredMark: false,
        initialValues: {
          remember: false
        },
        ...formProps
      },
      /* @__PURE__ */ import_react44.default.createElement(
        import_antd35.Form.Item,
        {
          name: "email",
          label: translate("pages.login.fields.email", "Email"),
          rules: [
            {
              required: true,
              message: translate(
                "pages.login.errors.requiredEmail",
                "Email is required"
              )
            },
            {
              type: "email",
              message: translate(
                "pages.login.errors.validEmail",
                "Invalid email address"
              )
            }
          ]
        },
        /* @__PURE__ */ import_react44.default.createElement(
          import_antd35.Input,
          {
            size: "large",
            placeholder: translate("pages.login.fields.email", "Email")
          }
        )
      ),
      /* @__PURE__ */ import_react44.default.createElement(
        import_antd35.Form.Item,
        {
          name: "password",
          label: translate("pages.login.fields.password", "Password"),
          rules: [
            {
              required: true,
              message: translate(
                "pages.login.errors.requiredPassword",
                "Password is required"
              )
            }
          ]
        },
        /* @__PURE__ */ import_react44.default.createElement(
          import_antd35.Input,
          {
            type: "password",
            autoComplete: "current-password",
            placeholder: "\u25CF\u25CF\u25CF\u25CF\u25CF\u25CF\u25CF\u25CF",
            size: "large"
          }
        )
      ),
      /* @__PURE__ */ import_react44.default.createElement(
        "div",
        {
          style: {
            display: "flex",
            justifyContent: "space-between",
            marginBottom: "24px"
          }
        },
        rememberMe ?? /* @__PURE__ */ import_react44.default.createElement(import_antd35.Form.Item, { name: "remember", valuePropName: "checked", noStyle: true }, /* @__PURE__ */ import_react44.default.createElement(
          import_antd35.Checkbox,
          {
            style: {
              fontSize: "12px"
            }
          },
          translate("pages.login.buttons.rememberMe", "Remember me")
        )),
        forgotPasswordLink ?? /* @__PURE__ */ import_react44.default.createElement(
          Link,
          {
            style: {
              color: token.colorPrimaryTextHover,
              fontSize: "12px",
              marginLeft: "auto"
            },
            to: "/forgot-password"
          },
          translate(
            "pages.login.buttons.forgotPassword",
            "Forgot password?"
          )
        )
      ),
      !hideForm && /* @__PURE__ */ import_react44.default.createElement(import_antd35.Form.Item, null, /* @__PURE__ */ import_react44.default.createElement(
        import_antd35.Button,
        {
          type: "primary",
          size: "large",
          htmlType: "submit",
          loading: isPending,
          block: true
        },
        translate("pages.login.signin", "Sign in")
      ))
    ),
    registerLink ?? /* @__PURE__ */ import_react44.default.createElement(
      "div",
      {
        style: {
          marginTop: hideForm ? 16 : 8
        }
      },
      /* @__PURE__ */ import_react44.default.createElement(import_antd35.Typography.Text, { style: { fontSize: 12 } }, translate(
        "pages.login.buttons.noAccount",
        "Don\u2019t have an account?"
      ), " ", /* @__PURE__ */ import_react44.default.createElement(
        Link,
        {
          to: "/register",
          style: {
            fontWeight: "bold",
            color: token.colorPrimaryTextHover
          }
        },
        translate("pages.login.signup", "Sign up")
      ))
    )
  );
  return /* @__PURE__ */ import_react44.default.createElement(import_antd35.Layout, { style: layoutStyles, ...wrapperProps ?? {} }, /* @__PURE__ */ import_react44.default.createElement(
    import_antd35.Row,
    {
      justify: "center",
      align: hideForm ? "top" : "middle",
      style: {
        padding: "16px 0",
        minHeight: "100dvh",
        paddingTop: hideForm ? "15dvh" : "16px"
      }
    },
    /* @__PURE__ */ import_react44.default.createElement(import_antd35.Col, { xs: 22 }, renderContent ? renderContent(CardContent, PageTitle) : /* @__PURE__ */ import_react44.default.createElement(import_react44.default.Fragment, null, PageTitle, CardContent))
  ));
}, "LoginPage");

// src/components/pages/auth/components/register/index.tsx
var import_react45 = __toESM(require("react"));
var import_core38 = require("@refinedev/core");
var import_antd36 = require("antd");
var import_core39 = require("@refinedev/core");
var RegisterPage = /* @__PURE__ */ __name(({
  providers,
  loginLink,
  wrapperProps,
  contentProps,
  renderContent,
  formProps,
  title,
  hideForm,
  mutationVariables
}) => {
  const { token } = import_antd36.theme.useToken();
  const [form] = import_antd36.Form.useForm();
  const translate = (0, import_core39.useTranslate)();
  const Link = (0, import_core38.useLink)();
  const { mutate: register, isPending } = (0, import_core39.useRegister)();
  const PageTitle = title === false ? null : /* @__PURE__ */ import_react45.default.createElement(
    "div",
    {
      style: {
        display: "flex",
        justifyContent: "center",
        marginBottom: "32px",
        fontSize: "20px"
      }
    },
    title ?? /* @__PURE__ */ import_react45.default.createElement(ThemedTitle, { collapsed: false })
  );
  const CardTitle = /* @__PURE__ */ import_react45.default.createElement(
    import_antd36.Typography.Title,
    {
      level: 3,
      style: {
        color: token.colorPrimaryTextHover,
        ...titleStyles
      }
    },
    translate("pages.register.title", "Sign up for your account")
  );
  const renderProviders = /* @__PURE__ */ __name(() => {
    if (providers && providers.length > 0) {
      return /* @__PURE__ */ import_react45.default.createElement(import_react45.default.Fragment, null, providers.map((provider) => {
        return /* @__PURE__ */ import_react45.default.createElement(
          import_antd36.Button,
          {
            key: provider.name,
            type: "default",
            block: true,
            icon: provider.icon,
            style: {
              display: "flex",
              justifyContent: "center",
              alignItems: "center",
              width: "100%",
              marginBottom: "8px"
            },
            onClick: () => register({
              ...mutationVariables,
              providerName: provider.name
            })
          },
          provider.label
        );
      }), !hideForm && /* @__PURE__ */ import_react45.default.createElement(import_antd36.Divider, null, /* @__PURE__ */ import_react45.default.createElement(
        import_antd36.Typography.Text,
        {
          style: {
            color: token.colorTextLabel
          }
        },
        translate(
          "pages.register.divider",
          translate("pages.login.divider", "or")
        )
      )));
    }
    return null;
  }, "renderProviders");
  const CardContent = /* @__PURE__ */ import_react45.default.createElement(
    import_antd36.Card,
    {
      title: CardTitle,
      styles: {
        header: headStyles,
        body: bodyStyles
      },
      style: {
        ...containerStyles,
        backgroundColor: token.colorBgElevated
      },
      ...contentProps ?? {}
    },
    renderProviders(),
    !hideForm && /* @__PURE__ */ import_react45.default.createElement(
      import_antd36.Form,
      {
        layout: "vertical",
        form,
        onFinish: (values) => register({ ...mutationVariables, ...values }),
        requiredMark: false,
        ...formProps
      },
      /* @__PURE__ */ import_react45.default.createElement(
        import_antd36.Form.Item,
        {
          name: "email",
          label: translate("pages.register.email", "Email"),
          rules: [
            {
              required: true,
              message: translate(
                "pages.register.errors.requiredEmail",
                "Email is required"
              )
            },
            {
              type: "email",
              message: translate(
                "pages.register.errors.validEmail",
                "Invalid email address"
              )
            }
          ]
        },
        /* @__PURE__ */ import_react45.default.createElement(
          import_antd36.Input,
          {
            size: "large",
            placeholder: translate("pages.register.fields.email", "Email")
          }
        )
      ),
      /* @__PURE__ */ import_react45.default.createElement(
        import_antd36.Form.Item,
        {
          name: "password",
          label: translate("pages.register.fields.password", "Password"),
          rules: [
            {
              required: true,
              message: translate(
                "pages.register.errors.requiredPassword",
                "Password is required"
              )
            }
          ]
        },
        /* @__PURE__ */ import_react45.default.createElement(import_antd36.Input, { type: "password", placeholder: "\u25CF\u25CF\u25CF\u25CF\u25CF\u25CF\u25CF\u25CF", size: "large" })
      ),
      /* @__PURE__ */ import_react45.default.createElement(
        "div",
        {
          style: {
            display: "flex",
            justifyContent: "space-between",
            marginBottom: "24px"
          }
        },
        loginLink ?? /* @__PURE__ */ import_react45.default.createElement(
          import_antd36.Typography.Text,
          {
            style: {
              fontSize: 12,
              marginLeft: "auto"
            }
          },
          translate(
            "pages.register.buttons.haveAccount",
            translate(
              "pages.login.buttons.haveAccount",
              "Have an account?"
            )
          ),
          " ",
          /* @__PURE__ */ import_react45.default.createElement(
            Link,
            {
              style: {
                fontWeight: "bold",
                color: token.colorPrimaryTextHover
              },
              to: "/login"
            },
            translate(
              "pages.register.signin",
              translate("pages.login.signin", "Sign in")
            )
          )
        )
      ),
      /* @__PURE__ */ import_react45.default.createElement(
        import_antd36.Form.Item,
        {
          style: {
            marginBottom: 0
          }
        },
        /* @__PURE__ */ import_react45.default.createElement(
          import_antd36.Button,
          {
            type: "primary",
            size: "large",
            htmlType: "submit",
            loading: isPending,
            block: true
          },
          translate("pages.register.buttons.submit", "Sign up")
        )
      )
    ),
    hideForm && loginLink !== false && /* @__PURE__ */ import_react45.default.createElement(
      "div",
      {
        style: {
          marginTop: hideForm ? 16 : 8
        }
      },
      /* @__PURE__ */ import_react45.default.createElement(
        import_antd36.Typography.Text,
        {
          style: {
            fontSize: 12
          }
        },
        translate(
          "pages.register.buttons.haveAccount",
          translate("pages.login.buttons.haveAccount", "Have an account?")
        ),
        " ",
        /* @__PURE__ */ import_react45.default.createElement(
          Link,
          {
            style: {
              fontWeight: "bold",
              color: token.colorPrimaryTextHover
            },
            to: "/login"
          },
          translate(
            "pages.register.signin",
            translate("pages.login.signin", "Sign in")
          )
        )
      )
    )
  );
  return /* @__PURE__ */ import_react45.default.createElement(import_antd36.Layout, { style: layoutStyles, ...wrapperProps ?? {} }, /* @__PURE__ */ import_react45.default.createElement(
    import_antd36.Row,
    {
      justify: "center",
      align: hideForm ? "top" : "middle",
      style: {
        padding: "16px 0",
        minHeight: "100dvh",
        paddingTop: hideForm ? "15dvh" : "16px"
      }
    },
    /* @__PURE__ */ import_react45.default.createElement(import_antd36.Col, { xs: 22 }, renderContent ? renderContent(CardContent, PageTitle) : /* @__PURE__ */ import_react45.default.createElement(import_react45.default.Fragment, null, PageTitle, CardContent))
  ));
}, "RegisterPage");

// src/components/pages/auth/components/forgotPassword/index.tsx
var import_react46 = __toESM(require("react"));
var import_core40 = require("@refinedev/core");
var import_antd37 = require("antd");
var import_core41 = require("@refinedev/core");
var ForgotPasswordPage = /* @__PURE__ */ __name(({
  loginLink,
  wrapperProps,
  contentProps,
  renderContent,
  formProps,
  title,
  mutationVariables
}) => {
  const { token } = import_antd37.theme.useToken();
  const [form] = import_antd37.Form.useForm();
  const translate = (0, import_core41.useTranslate)();
  const Link = (0, import_core40.useLink)();
  const { mutate: forgotPassword, isPending } = (0, import_core41.useForgotPassword)();
  const PageTitle = title === false ? null : /* @__PURE__ */ import_react46.default.createElement(
    "div",
    {
      style: {
        display: "flex",
        justifyContent: "center",
        marginBottom: "32px",
        fontSize: "20px"
      }
    },
    title ?? /* @__PURE__ */ import_react46.default.createElement(ThemedTitle, { collapsed: false })
  );
  const CardTitle = /* @__PURE__ */ import_react46.default.createElement(
    import_antd37.Typography.Title,
    {
      level: 3,
      style: {
        color: token.colorPrimaryTextHover,
        ...titleStyles
      }
    },
    translate("pages.forgotPassword.title", "Forgot your password?")
  );
  const CardContent = /* @__PURE__ */ import_react46.default.createElement(
    import_antd37.Card,
    {
      title: CardTitle,
      styles: {
        header: headStyles,
        body: bodyStyles
      },
      style: {
        ...containerStyles,
        backgroundColor: token.colorBgElevated
      },
      ...contentProps ?? {}
    },
    /* @__PURE__ */ import_react46.default.createElement(
      import_antd37.Form,
      {
        layout: "vertical",
        form,
        onFinish: (values) => forgotPassword({ ...values, ...mutationVariables }),
        requiredMark: false,
        ...formProps
      },
      /* @__PURE__ */ import_react46.default.createElement(
        import_antd37.Form.Item,
        {
          name: "email",
          label: translate("pages.forgotPassword.fields.email", "Email"),
          rules: [
            {
              required: true,
              message: translate(
                "pages.forgotPassword.errors.requiredEmail",
                "Email is required"
              )
            },
            {
              type: "email",
              message: translate(
                "pages.forgotPassword.errors.validEmail",
                "Invalid email address"
              )
            }
          ]
        },
        /* @__PURE__ */ import_react46.default.createElement(
          import_antd37.Input,
          {
            type: "email",
            size: "large",
            placeholder: translate(
              "pages.forgotPassword.fields.email",
              "Email"
            )
          }
        )
      ),
      /* @__PURE__ */ import_react46.default.createElement(
        "div",
        {
          style: {
            display: "flex",
            justifyContent: "space-between"
          }
        },
        loginLink ?? /* @__PURE__ */ import_react46.default.createElement(
          import_antd37.Typography.Text,
          {
            style: {
              fontSize: 12,
              marginLeft: "auto"
            }
          },
          translate(
            "pages.forgotPassword.buttons.haveAccount",
            translate(
              "pages.register.buttons.haveAccount",
              "Have an account? "
            )
          ),
          " ",
          /* @__PURE__ */ import_react46.default.createElement(
            Link,
            {
              style: {
                fontWeight: "bold",
                color: token.colorPrimaryTextHover
              },
              to: "/login"
            },
            translate(
              "pages.forgotPassword.signin",
              translate("pages.login.signin", "Sign in")
            )
          )
        )
      ),
      /* @__PURE__ */ import_react46.default.createElement(
        import_antd37.Form.Item,
        {
          style: {
            marginTop: "24px",
            marginBottom: 0
          }
        },
        /* @__PURE__ */ import_react46.default.createElement(
          import_antd37.Button,
          {
            type: "primary",
            size: "large",
            htmlType: "submit",
            loading: isPending,
            block: true
          },
          translate(
            "pages.forgotPassword.buttons.submit",
            "Send reset instructions"
          )
        )
      )
    )
  );
  return /* @__PURE__ */ import_react46.default.createElement(import_antd37.Layout, { style: layoutStyles, ...wrapperProps ?? {} }, /* @__PURE__ */ import_react46.default.createElement(
    import_antd37.Row,
    {
      justify: "center",
      align: "middle",
      style: {
        padding: "16px 0",
        minHeight: "100dvh"
      }
    },
    /* @__PURE__ */ import_react46.default.createElement(import_antd37.Col, { xs: 22 }, renderContent ? renderContent(CardContent, PageTitle) : /* @__PURE__ */ import_react46.default.createElement(import_react46.default.Fragment, null, PageTitle, CardContent))
  ));
}, "ForgotPasswordPage");

// src/components/pages/auth/components/updatePassword/index.tsx
var import_react47 = __toESM(require("react"));
var import_antd38 = require("antd");
var import_core42 = require("@refinedev/core");
var UpdatePasswordPage = /* @__PURE__ */ __name(({
  wrapperProps,
  contentProps,
  renderContent,
  formProps,
  title,
  mutationVariables
}) => {
  const { token } = import_antd38.theme.useToken();
  const [form] = import_antd38.Form.useForm();
  const translate = (0, import_core42.useTranslate)();
  const { mutate: updatePassword, isPending } = (0, import_core42.useUpdatePassword)();
  const PageTitle = title === false ? null : /* @__PURE__ */ import_react47.default.createElement(
    "div",
    {
      style: {
        display: "flex",
        justifyContent: "center",
        marginBottom: "32px",
        fontSize: "20px"
      }
    },
    title ?? /* @__PURE__ */ import_react47.default.createElement(ThemedTitle, { collapsed: false })
  );
  const CardTitle = /* @__PURE__ */ import_react47.default.createElement(
    import_antd38.Typography.Title,
    {
      level: 3,
      style: {
        color: token.colorPrimaryTextHover,
        ...titleStyles
      }
    },
    translate("pages.updatePassword.title", "Set New Password")
  );
  const CardContent = /* @__PURE__ */ import_react47.default.createElement(
    import_antd38.Card,
    {
      title: CardTitle,
      styles: {
        header: headStyles,
        body: bodyStyles
      },
      style: {
        ...containerStyles,
        backgroundColor: token.colorBgElevated
      },
      ...contentProps ?? {}
    },
    /* @__PURE__ */ import_react47.default.createElement(
      import_antd38.Form,
      {
        layout: "vertical",
        form,
        onFinish: (values) => updatePassword({ ...values, ...mutationVariables }),
        requiredMark: false,
        ...formProps
      },
      /* @__PURE__ */ import_react47.default.createElement(
        import_antd38.Form.Item,
        {
          name: "password",
          label: translate(
            "pages.updatePassword.fields.password",
            "New Password"
          ),
          rules: [
            {
              required: true,
              message: translate(
                "pages.updatePassword.errors.requiredPassword",
                "Password is required"
              )
            }
          ],
          style: { marginBottom: "12px" }
        },
        /* @__PURE__ */ import_react47.default.createElement(import_antd38.Input, { type: "password", placeholder: "\u25CF\u25CF\u25CF\u25CF\u25CF\u25CF\u25CF\u25CF", size: "large" })
      ),
      /* @__PURE__ */ import_react47.default.createElement(
        import_antd38.Form.Item,
        {
          name: "confirmPassword",
          label: translate(
            "pages.updatePassword.fields.confirmPassword",
            "Confirm New Password"
          ),
          hasFeedback: true,
          dependencies: ["password"],
          rules: [
            {
              required: true,
              message: translate(
                "pages.updatePassword.errors.requiredConfirmPassword",
                "Confirm password is required"
              )
            },
            ({ getFieldValue }) => ({
              validator(_, value) {
                if (!value || getFieldValue("password") === value) {
                  return Promise.resolve();
                }
                return Promise.reject(
                  new Error(
                    translate(
                      "pages.updatePassword.errors.confirmPasswordNotMatch",
                      "Passwords do not match"
                    )
                  )
                );
              }
            })
          ]
        },
        /* @__PURE__ */ import_react47.default.createElement(import_antd38.Input, { type: "password", placeholder: "\u25CF\u25CF\u25CF\u25CF\u25CF\u25CF\u25CF\u25CF", size: "large" })
      ),
      /* @__PURE__ */ import_react47.default.createElement(
        import_antd38.Form.Item,
        {
          style: {
            marginBottom: 0
          }
        },
        /* @__PURE__ */ import_react47.default.createElement(
          import_antd38.Button,
          {
            type: "primary",
            size: "large",
            htmlType: "submit",
            loading: isPending,
            block: true
          },
          translate("pages.updatePassword.buttons.submit", "Update")
        )
      )
    )
  );
  return /* @__PURE__ */ import_react47.default.createElement(import_antd38.Layout, { style: layoutStyles, ...wrapperProps ?? {} }, /* @__PURE__ */ import_react47.default.createElement(
    import_antd38.Row,
    {
      justify: "center",
      align: "middle",
      style: {
        padding: "16px 0",
        minHeight: "100dvh"
      }
    },
    /* @__PURE__ */ import_react47.default.createElement(import_antd38.Col, { xs: 22 }, renderContent ? renderContent(CardContent, PageTitle) : /* @__PURE__ */ import_react47.default.createElement(import_react47.default.Fragment, null, PageTitle, CardContent))
  ));
}, "UpdatePasswordPage");

// src/components/pages/auth/index.tsx
var AuthPage = /* @__PURE__ */ __name((props) => {
  const { type } = props;
  const renderView = /* @__PURE__ */ __name(() => {
    switch (type) {
      case "register":
        return /* @__PURE__ */ import_react48.default.createElement(RegisterPage, { ...props });
      case "forgotPassword":
        return /* @__PURE__ */ import_react48.default.createElement(ForgotPasswordPage, { ...props });
      case "updatePassword":
        return /* @__PURE__ */ import_react48.default.createElement(UpdatePasswordPage, { ...props });
      default:
        return /* @__PURE__ */ import_react48.default.createElement(LoginPage, { ...props });
    }
  }, "renderView");
  return /* @__PURE__ */ import_react48.default.createElement(import_react48.default.Fragment, null, renderView());
}, "AuthPage");

// src/components/breadcrumb/index.tsx
var import_react49 = __toESM(require("react"));
var import_core43 = require("@refinedev/core");
var import_antd39 = require("antd");
var import_icons16 = require("@ant-design/icons");
var Breadcrumb = /* @__PURE__ */ __name(({
  breadcrumbProps,
  showHome = true,
  hideIcons = false,
  meta,
  minItems = 2
}) => {
  const { breadcrumbs } = (0, import_core43.useBreadcrumb)({
    meta
  });
  const Link = (0, import_core43.useLink)();
  const { resources } = (0, import_core43.useResourceParams)();
  const rootRouteResource = (0, import_core43.matchResourceFromRoute)("/", resources);
  if (breadcrumbs.length < minItems)
    return null;
  const breadCrumbItems = breadcrumbs.map(({ label, icon, href }) => ({
    key: `breadcrumb-item-${label}`,
    title: /* @__PURE__ */ import_react49.default.createElement(
      "div",
      {
        style: {
          display: "flex",
          alignItems: "center",
          justifyContent: "center",
          gap: 4
        }
      },
      !hideIcons && icon,
      href ? /* @__PURE__ */ import_react49.default.createElement(Link, { to: href }, label) : /* @__PURE__ */ import_react49.default.createElement("span", null, label)
    )
  }));
  const getBreadcrumbItems = /* @__PURE__ */ __name(() => {
    var _a, _b;
    if (showHome && rootRouteResource.found) {
      return [
        {
          key: "breadcrumb-item-home",
          title: /* @__PURE__ */ import_react49.default.createElement(Link, { to: "/" }, ((_b = (_a = rootRouteResource == null ? void 0 : rootRouteResource.resource) == null ? void 0 : _a.meta) == null ? void 0 : _b.icon) ?? /* @__PURE__ */ import_react49.default.createElement(import_icons16.HomeOutlined, null))
        },
        ...breadCrumbItems
      ];
    }
    return breadCrumbItems;
  }, "getBreadcrumbItems");
  return /* @__PURE__ */ import_react49.default.createElement(import_antd39.Breadcrumb, { items: getBreadcrumbItems(), ...breadcrumbProps });
}, "Breadcrumb");

// src/components/pageHeader/index.tsx
var import_react50 = __toESM(require("react"));
var import_pro_layout = require("@ant-design/pro-layout");
var import_antd40 = require("antd");
var import_icons17 = require("@ant-design/icons");
var import_ui_types11 = require("@refinedev/ui-types");
var PageHeader = /* @__PURE__ */ __name(({ children, ...props }) => {
  var _a;
  const direction = (_a = (0, import_react50.useContext)(import_antd40.ConfigProvider.ConfigContext)) == null ? void 0 : _a.direction;
  const renderBackButton = /* @__PURE__ */ __name(() => {
    const BackIcon = direction === "rtl" ? import_icons17.ArrowRightOutlined : import_icons17.ArrowLeftOutlined;
    return /* @__PURE__ */ import_react50.default.createElement(import_antd40.Button, { type: "text", icon: /* @__PURE__ */ import_react50.default.createElement(BackIcon, null) });
  }, "renderBackButton");
  const backIcon = typeof props.backIcon === "undefined" ? renderBackButton() : props.backIcon;
  const title = typeof props.title === "string" ? /* @__PURE__ */ import_react50.default.createElement(
    import_antd40.Typography.Title,
    {
      className: import_ui_types11.RefinePageHeaderClassNames.Title,
      level: 4,
      style: { marginBottom: 0 }
    },
    props.title
  ) : props.title;
  const subtitle = typeof props.title === "string" ? /* @__PURE__ */ import_react50.default.createElement(
    import_antd40.Typography.Title,
    {
      className: import_ui_types11.RefinePageHeaderClassNames.SubTitle,
      level: 5,
      type: "secondary",
      style: { marginBottom: 0 }
    },
    props.subTitle
  ) : props.subTitle;
  return /* @__PURE__ */ import_react50.default.createElement(
    import_pro_layout.PageHeader,
    {
      ...props,
      backIcon,
      title,
      subTitle: subtitle,
      style: { padding: 0, ...props.style }
    },
    children
  );
}, "PageHeader");

// src/components/autoSaveIndicator/index.tsx
var import_react51 = __toESM(require("react"));
var import_core44 = require("@refinedev/core");
var import_antd41 = require("antd");
var import_icons18 = require("@ant-design/icons");
var AutoSaveIndicator = /* @__PURE__ */ __name(({
  status,
  elements: {
    success = /* @__PURE__ */ import_react51.default.createElement(
      Message,
      {
        translationKey: "autoSave.success",
        defaultMessage: "saved",
        icon: /* @__PURE__ */ import_react51.default.createElement(import_icons18.CheckCircleOutlined, null)
      }
    ),
    error = /* @__PURE__ */ import_react51.default.createElement(
      Message,
      {
        translationKey: "autoSave.error",
        defaultMessage: "auto save failure",
        icon: /* @__PURE__ */ import_react51.default.createElement(import_icons18.ExclamationCircleOutlined, null)
      }
    ),
    loading = /* @__PURE__ */ import_react51.default.createElement(
      Message,
      {
        translationKey: "autoSave.loading",
        defaultMessage: "saving...",
        icon: /* @__PURE__ */ import_react51.default.createElement(import_icons18.SyncOutlined, null)
      }
    ),
    idle = /* @__PURE__ */ import_react51.default.createElement(
      Message,
      {
        translationKey: "autoSave.idle",
        defaultMessage: "waiting for changes",
        icon: /* @__PURE__ */ import_react51.default.createElement(import_icons18.EllipsisOutlined, null)
      }
    )
  } = {}
}) => {
  return /* @__PURE__ */ import_react51.default.createElement(
    import_core44.AutoSaveIndicator,
    {
      status,
      elements: {
        success,
        error,
        loading,
        idle
      }
    }
  );
}, "AutoSaveIndicator");
var Message = /* @__PURE__ */ __name(({
  translationKey,
  defaultMessage,
  icon
}) => {
  const translate = (0, import_core44.useTranslate)();
  const { token } = import_antd41.theme.useToken();
  return /* @__PURE__ */ import_react51.default.createElement(
    import_antd41.Typography.Text,
    {
      style: {
        marginRight: 5,
        color: token.colorTextTertiary,
        fontSize: ".8rem"
      }
    },
    translate(translationKey, defaultMessage),
    /* @__PURE__ */ import_react51.default.createElement("span", { style: { marginLeft: ".2rem" } }, icon)
  );
}, "Message");

// src/definitions/filter-mappers/index.ts
var import_dayjs3 = __toESM(require("dayjs"));
var rangePickerFilterMapper = /* @__PURE__ */ __name((selectedKeys, event) => {
  if (!selectedKeys) {
    return selectedKeys;
  }
  if (event === "value") {
    return selectedKeys.map((key) => {
      if (typeof key === "string") {
        return (0, import_dayjs3.default)(key);
      }
      return key;
    });
  }
  if (event === "onChange") {
    if (selectedKeys.every(import_dayjs3.default.isDayjs)) {
      return selectedKeys.map((date) => (0, import_dayjs3.default)(date).toISOString());
    }
  }
  return selectedKeys;
}, "rangePickerFilterMapper");

// src/definitions/upload/index.ts
var getValueFromEvent = /* @__PURE__ */ __name((event) => {
  const { fileList } = event;
  return [...fileList];
}, "getValueFromEvent");

// src/definitions/themes/index.ts
var RefineThemes = {
  Blue: {
    token: {
      colorPrimary: "#1677FF"
    }
  },
  Purple: {
    token: {
      colorPrimary: "#722ED1"
    }
  },
  Magenta: {
    token: {
      colorPrimary: "#EB2F96"
    }
  },
  Red: {
    token: {
      colorPrimary: "#F5222D"
    }
  },
  Orange: {
    token: {
      colorPrimary: "#FA541C"
    }
  },
  Yellow: {
    token: {
      colorPrimary: "#FAAD14"
    }
  },
  Green: {
    token: {
      colorPrimary: "#52C41A"
    }
  }
};
