// @refinedev/react-router/dist/index.cjs
var __create = Object.create;
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __getProtoOf = Object.getPrototypeOf;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toESM = (mod, isNodeMode, target) => (target = mod != null ? __create(__getProtoOf(mod)) : {}, __copyProps(
  // If the importer is in node compatibility mode or this is not an ESM
  // file that has been converted to a CommonJS file using a Babel-
  // compatible transform (i.e. "__esModule" has not been set), then set
  // "default" to the CommonJS "module.exports" for node compatibility.
  isNodeMode || !mod || !mod.__esModule ? __defProp(target, "default", { value: mod, enumerable: true }) : target,
  mod
));
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// src/index.ts
var src_exports = {};
__export(src_exports, {
  CatchAllNavigate: () => CatchAllNavigate,
  DocumentTitleHandler: () => DocumentTitleHandler,
  NavigateToResource: () => NavigateToResource,
  UnsavedChangesNotifier: () => UnsavedChangesNotifier,
  default: () => routerProvider,
  stringifyConfig: () => stringifyConfig,
  useDocumentTitle: () => useDocumentTitle
});
module.exports = __toCommonJS(src_exports);

// src/bindings.tsx
var import_react = __toESM(require("react"));
var import_core = require("@refinedev/core");
var import_react2 = require("react");
var import_qs = __toESM(require("qs"));
var import_react_router = require("react-router");

// src/convert-to-number-if-possible.ts
var convertToNumberIfPossible = (value) => {
  if (typeof value === "undefined") {
    return value;
  }
  const num = Number(value);
  if (`${num}` === value) {
    return num;
  }
  return value;
};

// src/bindings.tsx
var stringifyConfig = {
  addQueryPrefix: true,
  skipNulls: true,
  arrayFormat: "indices",
  encode: false,
  encodeValuesOnly: true
};
var routerProvider = {
  go: () => {
    const { search: existingSearch, hash: existingHash } = (0, import_react_router.useLocation)();
    const navigate = (0, import_react_router.useNavigate)();
    const fn = (0, import_react2.useCallback)(
      ({
        to,
        type,
        query,
        hash,
        options: { keepQuery, keepHash } = {}
      }) => {
        const urlQuery = {
          ...keepQuery && existingSearch && import_qs.default.parse(existingSearch, { ignoreQueryPrefix: true }),
          ...query
        };
        if (urlQuery.to) {
          urlQuery.to = encodeURIComponent(`${urlQuery.to}`);
        }
        const hasUrlQuery = Object.keys(urlQuery).length > 0;
        const urlHash = `#${(hash || keepHash && existingHash || "").replace(
          /^#/,
          ""
        )}`;
        const hasUrlHash = urlHash.length > 1;
        const urlTo = to || "";
        const fullPath = `${urlTo}${hasUrlQuery ? import_qs.default.stringify(urlQuery, stringifyConfig) : ""}${hasUrlHash ? urlHash : ""}`;
        if (type === "path") {
          return fullPath;
        }
        navigate(fullPath, {
          replace: type === "replace"
        });
        return;
      },
      [existingHash, existingSearch, navigate]
    );
    return fn;
  },
  back: () => {
    const navigate = (0, import_react_router.useNavigate)();
    const fn = (0, import_react2.useCallback)(() => {
      navigate(-1);
    }, [navigate]);
    return fn;
  },
  parse: () => {
    var _a;
    let params = (0, import_react_router.useParams)();
    const { pathname, search } = (0, import_react_router.useLocation)();
    const { resources } = (0, import_react2.useContext)(import_core.ResourceContext);
    const { resource, action, matchedRoute } = import_react.default.useMemo(() => {
      return (0, import_core.matchResourceFromRoute)(pathname, resources);
    }, [resources, pathname]);
    if (Object.entries(params).length === 0 && matchedRoute) {
      params = ((_a = (0, import_react_router.matchPath)(matchedRoute, pathname)) == null ? void 0 : _a.params) || {};
    }
    const fn = (0, import_react2.useCallback)(() => {
      const parsedSearch = import_qs.default.parse(search, { ignoreQueryPrefix: true });
      const combinedParams = {
        ...params,
        ...parsedSearch
      };
      const response = {
        ...resource && { resource },
        ...action && { action },
        ...(params == null ? void 0 : params.id) && { id: decodeURIComponent(params.id) },
        // ...(params?.action && { action: params.action }), // lets see if there is a need for this
        pathname,
        params: {
          ...combinedParams,
          currentPage: convertToNumberIfPossible(
            combinedParams.currentPage
          ),
          pageSize: convertToNumberIfPossible(
            combinedParams.pageSize
          ),
          to: combinedParams.to ? decodeURIComponent(combinedParams.to) : void 0
        }
      };
      return response;
    }, [pathname, search, params, resource, action]);
    return fn;
  },
  Link: import_react.default.forwardRef(function RefineLink(props, ref) {
    return /* @__PURE__ */ import_react.default.createElement(import_react_router.Link, { to: props.to, ...props, ref });
  })
};

// src/navigate-to-resource.tsx
var import_core2 = require("@refinedev/core");
var import_react3 = __toESM(require("react"));
var import_react_router2 = require("react-router");
var NavigateToResource = ({
  resource: resourceProp,
  fallbackTo,
  meta
}) => {
  const getToPath = (0, import_core2.useGetToPath)();
  const { resource, resources } = (0, import_core2.useResourceParams)({
    resource: resourceProp
  });
  const toResource = resource || resources.find((r) => r.list);
  if (toResource) {
    const path = getToPath({
      resource: toResource,
      action: "list",
      meta
    });
    if (path) {
      return /* @__PURE__ */ import_react3.default.createElement(import_react_router2.Navigate, { to: path });
    }
    return null;
  }
  if (fallbackTo) {
    console.warn(`No resource is found. navigation to ${fallbackTo}.`);
    return /* @__PURE__ */ import_react3.default.createElement(import_react_router2.Navigate, { to: fallbackTo });
  }
  console.warn(
    'No resource and "fallbackTo" is found. No navigation will be made.'
  );
  return null;
};

// src/unsaved-changes-notifier.tsx
var import_react5 = __toESM(require("react"));
var import_core3 = require("@refinedev/core");

// src/use-prompt-workaround.ts
var import_react4 = __toESM(require("react"));
var import_react_router3 = require("react-router");
function useConfirmExit(confirmExit, when = true) {
  const { navigator } = import_react4.default.useContext(import_react_router3.UNSAFE_NavigationContext);
  import_react4.default.useEffect(() => {
    if (!when) {
      return;
    }
    const go = navigator.go;
    const push = navigator.push;
    navigator.push = (...args) => {
      const result = confirmExit();
      if (result !== false) {
        push(...args);
      }
    };
    navigator.go = (...args) => {
      const result = confirmExit();
      if (result !== false) {
        go(...args);
      }
    };
    return () => {
      navigator.push = push;
      navigator.go = go;
    };
  }, [navigator, confirmExit, when]);
}
function usePrompt(message, when = true, onConfirm) {
  const warnWhenListener = import_react4.default.useCallback(
    (e) => {
      e.preventDefault();
      e.returnValue = message;
      return e.returnValue;
    },
    [message]
  );
  import_react4.default.useEffect(() => {
    if (when) {
      window.addEventListener("beforeunload", warnWhenListener);
    }
    return () => {
      window.removeEventListener("beforeunload", warnWhenListener);
    };
  }, [warnWhenListener, when]);
  const confirmExit = import_react4.default.useCallback(() => {
    const confirm = window.confirm(message);
    if (confirm && onConfirm) {
      onConfirm();
    }
    return confirm;
  }, [message]);
  useConfirmExit(confirmExit, when);
}

// src/unsaved-changes-notifier.tsx
var import_react_router4 = require("react-router");
var UnsavedChangesNotifier = ({
  translationKey = "warnWhenUnsavedChanges",
  message = "Are you sure you want to leave? You have unsaved changes."
}) => {
  const translate = (0, import_core3.useTranslate)();
  const { pathname } = (0, import_react_router4.useLocation)();
  const { warnWhen, setWarnWhen } = (0, import_core3.useWarnAboutChange)();
  import_react5.default.useEffect(() => {
    return () => setWarnWhen == null ? void 0 : setWarnWhen(false);
  }, [pathname]);
  const warnMessage = import_react5.default.useMemo(() => {
    return translate(translationKey, message);
  }, [translationKey, message, translate]);
  usePrompt(warnMessage, warnWhen, () => {
    setWarnWhen == null ? void 0 : setWarnWhen(false);
  });
  return null;
};

// src/catch-all-navigate.tsx
var import_react6 = __toESM(require("react"));
var import_react_router5 = require("react-router");
var CatchAllNavigate = ({ to }) => {
  const { pathname, search } = (0, import_react_router5.useLocation)();
  const queryValue = `${pathname}${search}`;
  const query = queryValue.length > 1 ? `?to=${encodeURIComponent(queryValue)}` : "";
  return /* @__PURE__ */ import_react6.default.createElement(import_react_router5.Navigate, { to: `${to}${query}` });
};

// src/document-title-handler.tsx
var import_core4 = require("@refinedev/core");
var import_react7 = __toESM(require("react"));
var import_react_router6 = require("react-router");
function DocumentTitleHandler({ handler }) {
  var _a;
  const location = (0, import_react_router6.useLocation)();
  const { action, id, params, pathname, resource } = (0, import_core4.useParsed)();
  const translate = (0, import_core4.useTranslate)();
  const getUserFriendlyName = (0, import_core4.useUserFriendlyName)();
  const identifier = (resource == null ? void 0 : resource.identifier) ?? (resource == null ? void 0 : resource.name);
  const preferredLabel = (_a = resource == null ? void 0 : resource.meta) == null ? void 0 : _a.label;
  const resourceName = preferredLabel ?? getUserFriendlyName(identifier, action === "list" ? "plural" : "singular");
  const populatedLabel = translate(
    `${resource == null ? void 0 : resource.name}.${resource == null ? void 0 : resource.name}`,
    resourceName
  );
  (0, import_react7.useLayoutEffect)(() => {
    const autoGeneratedTitle = (0, import_core4.generateDefaultDocumentTitle)(
      translate,
      resource,
      action,
      `${id}`,
      resourceName,
      getUserFriendlyName
    );
    if (handler) {
      document.title = handler({
        action,
        resource: {
          ...resource ?? {},
          meta: {
            ...resource == null ? void 0 : resource.meta,
            label: populatedLabel
          }
        },
        params,
        pathname,
        autoGeneratedTitle
      });
    } else {
      document.title = autoGeneratedTitle;
    }
  }, [location]);
  return /* @__PURE__ */ import_react7.default.createElement(import_react7.default.Fragment, null);
}

// src/use-document-title.ts
var import_core5 = require("@refinedev/core");
var import_react8 = require("react");
var useDocumentTitle = (title) => {
  const translate = (0, import_core5.useTranslate)();
  (0, import_react8.useEffect)(() => {
    if (!title)
      return;
    if (typeof title === "string") {
      document.title = translate(title);
    } else {
      document.title = translate(title.i18nKey);
    }
  }, [title]);
  return (title2) => {
    if (typeof title2 === "string") {
      document.title = translate(title2);
    } else {
      document.title = translate(title2.i18nKey);
    }
  };
};
