// @refinedev/ui-types/dist/index.mjs
// src/ids.tsx
var RefineButtonTestIds = {
  CloneButton: "refine-clone-button",
  DeleteButton: "refine-delete-button",
  EditButton: "refine-edit-button",
  SaveButton: "refine-save-button",
  CreateButton: "refine-create-button",
  ImportButton: "refine-import-button",
  ExportButton: "refine-export-button",
  ListButton: "refine-list-button",
  RefreshButton: "refine-refresh-button",
  ShowButton: "refine-show-button"
};

// src/classNames.ts
var RefinePageHeaderClassNames = {
  Title: "refine-pageHeader-title",
  SubTitle: "refine-pageHeader-subTitle"
};
var RefineButtonClassNames = {
  CloneButton: "refine-clone-button",
  DeleteButton: "refine-delete-button",
  EditButton: "refine-edit-button",
  SaveButton: "refine-save-button",
  CreateButton: "refine-create-button",
  ImportButton: "refine-import-button",
  ExportButton: "refine-export-button",
  ListButton: "refine-list-button",
  RefreshButton: "refine-refresh-button",
  ShowButton: "refine-show-button"
};
export {
  RefineButtonClassNames,
  RefineButtonTestIds,
  RefinePageHeaderClassNames
};
