// @refinedev/kbar/dist/index.cjs
"use client"
var __create = Object.create;
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __getProtoOf = Object.getPrototypeOf;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __name = (target, value) => __defProp(target, "name", { value, configurable: true });
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toESM = (mod, isNodeMode, target) => (target = mod != null ? __create(__getProtoOf(mod)) : {}, __copyProps(
  // If the importer is in node compatibility mode or this is not an ESM
  // file that has been converted to a CommonJS file using a Babel-
  // compatible transform (i.e. "__esModule" has not been set), then set
  // "default" to the CommonJS "module.exports" for node compatibility.
  isNodeMode || !mod || !mod.__esModule ? __defProp(target, "default", { value: mod, enumerable: true }) : target,
  mod
));
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// src/index.tsx
var src_exports = {};
__export(src_exports, {
  KBAR_LISTBOX: () => import_kbar5.KBAR_LISTBOX,
  KBarAnimator: () => import_kbar5.KBarAnimator,
  KBarContext: () => import_kbar5.KBarContext,
  KBarPortal: () => import_kbar5.KBarPortal,
  KBarPositioner: () => import_kbar5.KBarPositioner,
  KBarProvider: () => import_kbar5.KBarProvider,
  KBarResults: () => import_kbar5.KBarResults,
  KBarSearch: () => import_kbar5.KBarSearch,
  NO_GROUP: () => import_kbar5.NO_GROUP,
  Priority: () => import_kbar5.Priority,
  RefineKbar: () => RefineKbar,
  RefineKbarPropsContext: () => RefineKbarPropsContext,
  RefineKbarProvider: () => RefineKbarProvider,
  VisualState: () => import_kbar5.VisualState,
  createAction: () => import_kbar5.createAction,
  getListboxItemId: () => import_kbar5.getListboxItemId,
  useKBar: () => import_kbar5.useKBar,
  useMatches: () => import_kbar5.useMatches,
  useRefineKbar: () => useRefineKbar,
  useRegisterActions: () => import_kbar5.useRegisterActions
});
module.exports = __toCommonJS(src_exports);
var import_react6 = __toESM(require("react"));
var import_kbar4 = require("kbar");
var import_kbar5 = require("kbar");

// src/hooks/useRefineKbar/index.tsx
var import_react = require("react");
var import_core = require("@refinedev/core");
var import_kbar = require("kbar");

// src/definitions/capitalize/index.ts
var capitalize = /* @__PURE__ */ __name((str) => str.replace(
  /\w\S*/g,
  (txt) => txt.charAt(0).toUpperCase() + txt.slice(1).toLowerCase()
), "capitalize");

// src/hooks/useRefineKbar/index.tsx
var import_core2 = require("@refinedev/core");
var useRefineKbar = /* @__PURE__ */ __name(() => {
  const t = (0, import_core.useTranslate)();
  const {
    id: idFromParams,
    action: actionFromParams,
    resource: resourceFromParams,
    resources
  } = (0, import_core.useResourceParams)();
  const getToPath = (0, import_core2.useGetToPath)();
  const go = (0, import_core.useGo)();
  const {
    mutation: { mutate }
  } = (0, import_core.useDelete)();
  const getUserFriendlyName = (0, import_core.useUserFriendlyName)();
  const kbarContext = (0, import_react.useContext)(import_kbar.KBarContext);
  const { can } = (0, import_core.useCanWithoutCache)();
  const [actions, setActions] = (0, import_react.useState)([]);
  (0, import_react.useEffect)(() => {
    const preaparedActions = /* @__PURE__ */ __name(async () => {
      return await Promise.all(
        moveActionToFirst().flatMap((resource) => {
          return createActionWithResource(resource);
        })
      );
    }, "preaparedActions");
    preaparedActions().then((actions2) => {
      return setActions(actions2.flatMap((action) => action));
    });
  }, [resources, idFromParams, resourceFromParams, actionFromParams]);
  (0, import_react.useEffect)(() => {
    if (actions.length === 0) {
      kbarContext.query.setVisualState(import_kbar.VisualState.hidden);
    }
  }, [actions]);
  const moveActionToFirst = /* @__PURE__ */ __name(() => {
    const orderedResources = [...resources];
    const fromIndex = orderedResources == null ? void 0 : orderedResources.findIndex(
      (resource) => (resource.identifier ?? (resource == null ? void 0 : resource.name)) === ((resourceFromParams == null ? void 0 : resourceFromParams.identifier) ?? (resourceFromParams == null ? void 0 : resourceFromParams.name))
    );
    if (fromIndex > 0) {
      const element = orderedResources[fromIndex];
      orderedResources.splice(fromIndex, 1);
      orderedResources.splice(0, 0, element);
    }
    return orderedResources;
  }, "moveActionToFirst");
  const createActionWithResource = /* @__PURE__ */ __name(async (resource) => {
    var _a, _b, _c, _d, _e, _f;
    const { name, list, create, show, edit } = resource;
    const label = (_a = resource == null ? void 0 : resource.meta) == null ? void 0 : _a.label;
    const icon = (_b = resource == null ? void 0 : resource.meta) == null ? void 0 : _b.icon;
    const canDelete = (_c = resource == null ? void 0 : resource.meta) == null ? void 0 : _c.canDelete;
    const section = label ?? t(
      `${resource.name}.${resource.name}`,
      getUserFriendlyName(resource.name, "plural")
    );
    const tempActions = [];
    if (list && (resourceFromParams !== void 0 && (resourceFromParams == null ? void 0 : resourceFromParams.name) !== name || actionFromParams !== void 0 && (resourceFromParams == null ? void 0 : resourceFromParams.name) === name)) {
      const { can: canList } = await (can == null ? void 0 : can({
        resource: name,
        action: "list" /* List */,
        params: { id: idFromParams, resource }
      })) || { can: true };
      if (canList) {
        tempActions.push(
          (0, import_kbar.createAction)({
            name: t("actions.list", capitalize("list" /* List */)),
            section,
            icon,
            perform: () => {
              const p = getToPath({
                resource,
                action: "list"
              });
              if (p) {
                go({ to: p });
              }
            }
          })
        );
      }
    }
    if ((((_d = resource == null ? void 0 : resource.meta) == null ? void 0 : _d.canCreate) || !!create) && create && ("create" /* Create */ !== actionFromParams || (resourceFromParams == null ? void 0 : resourceFromParams.name) !== name)) {
      const { can: canAccessCreate } = await (can == null ? void 0 : can({
        resource: name,
        action: "create" /* Create */,
        params: { resource }
      })) || { can: true };
      if (canAccessCreate) {
        tempActions.push(
          (0, import_kbar.createAction)({
            name: t("actions.create", capitalize("create" /* Create */)),
            section,
            icon,
            keywords: "new",
            perform: () => {
              const p = getToPath({
                resource,
                action: "create"
              });
              if (p) {
                go({ to: p });
              }
            }
          })
        );
      }
    }
    if ((resourceFromParams == null ? void 0 : resourceFromParams.name) === name && idFromParams) {
      if ((((_e = resource == null ? void 0 : resource.meta) == null ? void 0 : _e.canShow) || !!show) && show && "show" /* Show */ !== actionFromParams) {
        const { can: canAccessShow } = await (can == null ? void 0 : can({
          resource: name,
          action: "show" /* Show */,
          params: { id: idFromParams, resource }
        })) || { can: true };
        if (canAccessShow) {
          tempActions.push(
            (0, import_kbar.createAction)({
              name: t("actions.show", capitalize("show" /* Show */)),
              section,
              icon,
              perform: () => {
                const p = getToPath({
                  resource,
                  action: "show",
                  meta: {
                    id: idFromParams
                  }
                });
                if (p) {
                  go({ to: p });
                }
              }
            })
          );
        }
      }
      if ((((_f = resource == null ? void 0 : resource.meta) == null ? void 0 : _f.canEdit) || !!edit) && edit && "edit" /* Edit */ !== actionFromParams) {
        const { can: canAccessEdit } = await (can == null ? void 0 : can({
          resource: name,
          action: "edit" /* Edit */,
          params: { id: idFromParams, resource }
        })) || { can: true };
        if (canAccessEdit) {
          tempActions.push(
            (0, import_kbar.createAction)({
              name: t("actions.edit", capitalize("edit" /* Edit */)),
              section,
              icon,
              perform: () => {
                const p = getToPath({
                  resource,
                  action: "edit",
                  meta: {
                    id: idFromParams
                  }
                });
                if (p) {
                  go({ to: p });
                }
              }
            })
          );
        }
      }
      if (canDelete) {
        const { can: canAccessDelete } = await (can == null ? void 0 : can({
          resource: name,
          action: "delete" /* Delete */,
          params: { id: idFromParams, resource }
        })) || { can: true };
        if (canAccessDelete) {
          tempActions.push(
            {
              id: "delete",
              name: t(
                "actions.delete",
                capitalize("delete" /* Delete */)
              ),
              section,
              icon
            },
            (0, import_kbar.createAction)({
              name: t(
                "buttons.delete",
                capitalize("delete" /* Delete */)
              ),
              section: t("buttons.confirm", "Are you sure?"),
              parent: "delete",
              perform: () => {
                mutate(
                  {
                    resource: resource.name,
                    id: idFromParams
                  },
                  {
                    onSuccess: () => {
                      const p = getToPath({
                        resource,
                        action: "list"
                      });
                      if (p) {
                        go({ to: p });
                      }
                    }
                  }
                );
              }
            }),
            (0, import_kbar.createAction)({
              name: t("buttons.cancel", "Cancel"),
              parent: "delete",
              perform: () => null
            })
          );
        }
      }
    }
    return tempActions;
  }, "createActionWithResource");
  (0, import_kbar.useRegisterActions)(actions, [actions]);
}, "useRefineKbar");

// src/components/commandBar/index.tsx
var import_react2 = __toESM(require("react"));
var import_kbar2 = require("kbar");
var CommandBar = /* @__PURE__ */ __name(() => {
  const searchStyle = {
    padding: "12px 16px",
    fontSize: "16px",
    width: "100%",
    boxSizing: "border-box",
    outline: "none",
    border: "none",
    background: "rgb(252 252 252)",
    color: "black"
  };
  const animatorStyle = {
    maxWidth: "600px",
    width: "100%",
    background: "white",
    color: "black",
    borderRadius: "8px",
    overflow: "hidden",
    boxShadow: "0px 4px 4px rgba(0, 0, 0, 0.25)"
  };
  const positionerStyle = {
    opacity: 1,
    transition: "background 0.35s cubic-bezier(0.4, 0, 0.2, 1) 0s",
    backdropFilter: "saturate(180%) blur(1px)",
    background: "rgba(0, 0, 0, 0.1)",
    zIndex: "9999"
  };
  return /* @__PURE__ */ import_react2.default.createElement(import_kbar2.KBarPortal, null, /* @__PURE__ */ import_react2.default.createElement(import_kbar2.KBarPositioner, { style: positionerStyle }, /* @__PURE__ */ import_react2.default.createElement(import_kbar2.KBarAnimator, { style: animatorStyle }, /* @__PURE__ */ import_react2.default.createElement(import_kbar2.KBarSearch, { style: searchStyle }), /* @__PURE__ */ import_react2.default.createElement(RenderResults, null))));
}, "CommandBar");

// src/components/renderResults/index.tsx
var import_react3 = __toESM(require("react"));
var import_kbar3 = require("kbar");
var groupNameStyle = {
  padding: "8px 16px",
  fontSize: "14px",
  textTransform: "uppercase",
  fontWeight: "bold",
  opacity: 0.5
};
var RenderResults = /* @__PURE__ */ __name(() => {
  const { results, rootActionId } = (0, import_kbar3.useMatches)();
  return /* @__PURE__ */ import_react3.default.createElement(
    import_kbar3.KBarResults,
    {
      items: results,
      onRender: ({ item, active }) => {
        return typeof item === "string" ? /* @__PURE__ */ import_react3.default.createElement("div", { style: groupNameStyle }, item) : /* @__PURE__ */ import_react3.default.createElement(
          ResultItem,
          {
            action: item,
            active,
            currentRootActionId: rootActionId
          }
        );
      }
    }
  );
}, "RenderResults");

// src/components/resultItem/index.tsx
var import_react4 = __toESM(require("react"));
var ResultItem = import_react4.default.forwardRef(
  ({ action, active, currentRootActionId }, ref) => {
    var _a;
    const ancestors = import_react4.default.useMemo(() => {
      if (!currentRootActionId)
        return action.ancestors;
      const index = action.ancestors.findIndex(
        (ancestor) => ancestor.id === currentRootActionId
      );
      return action.ancestors.slice(index + 1);
    }, [action.ancestors, currentRootActionId]);
    return /* @__PURE__ */ import_react4.default.createElement(
      "div",
      {
        ref,
        style: {
          padding: "12px 16px",
          background: active ? "rgba(0 0 0 / 0.05)" : "transparent",
          borderLeft: `2px solid ${active ? "rgb(28 28 29)" : "transparent"}`,
          display: "flex",
          alignItems: "center",
          justifyContent: "space-between",
          cursor: "pointer"
        }
      },
      /* @__PURE__ */ import_react4.default.createElement(
        "div",
        {
          style: {
            display: "flex",
            gap: "8px",
            alignItems: "center",
            fontSize: 14
          }
        },
        action.icon && action.icon,
        /* @__PURE__ */ import_react4.default.createElement("div", { style: { display: "flex", flexDirection: "column" } }, /* @__PURE__ */ import_react4.default.createElement("div", null, ancestors.length > 0 && ancestors.map((ancestor) => /* @__PURE__ */ import_react4.default.createElement(import_react4.default.Fragment, { key: ancestor.id }, /* @__PURE__ */ import_react4.default.createElement(
          "span",
          {
            style: {
              opacity: 0.5,
              marginRight: 8
            }
          },
          ancestor.name
        ), /* @__PURE__ */ import_react4.default.createElement(
          "span",
          {
            style: {
              marginRight: 8
            }
          },
          "\u203A"
        ))), /* @__PURE__ */ import_react4.default.createElement(
          "span",
          {
            style: {
              color: action.name.toLocaleUpperCase() === "DELETE" ? "red" : "black"
            }
          },
          action.name
        )), action.subtitle && /* @__PURE__ */ import_react4.default.createElement("span", { style: { fontSize: 12 } }, action.subtitle))
      ),
      ((_a = action.shortcut) == null ? void 0 : _a.length) ? /* @__PURE__ */ import_react4.default.createElement(
        "div",
        {
          "aria-hidden": true,
          style: {
            display: "grid",
            gridAutoFlow: "column",
            gap: "4px"
          }
        },
        action.shortcut.map((sc) => /* @__PURE__ */ import_react4.default.createElement(
          "kbd",
          {
            key: sc,
            style: {
              padding: "4px 6px",
              background: "rgba(0 0 0 / .1)",
              borderRadius: "4px",
              fontSize: 14
            }
          },
          sc
        ))
      ) : null
    );
  }
);
ResultItem.displayName = "ResultItem";

// src/components/refineKbar/index.tsx
var import_react5 = __toESM(require("react"));
var RefineKbar = /* @__PURE__ */ __name(({ commandBarProps }) => {
  const context = (0, import_react5.useContext)(RefineKbarPropsContext);
  useRefineKbar();
  const props = { ...context, ...commandBarProps };
  return /* @__PURE__ */ import_react5.default.createElement(CommandBar, { ...props });
}, "RefineKbar");

// src/index.tsx
var RefineKbarPropsContext = (0, import_react6.createContext)({});
var RefineKbarProvider = /* @__PURE__ */ __name(({ children, commandBarProps }) => {
  return /* @__PURE__ */ import_react6.default.createElement(RefineKbarPropsContext.Provider, { value: commandBarProps ?? {} }, /* @__PURE__ */ import_react6.default.createElement(import_kbar4.KBarProvider, null, children));
}, "RefineKbarProvider");
