// @refinedev/kbar/dist/index.mjs
"use client"
var __defProp = Object.defineProperty;
var __name = (target, value) => __defProp(target, "name", { value, configurable: true });

// src/index.tsx
import React5, { createContext } from "react";
import { KBarProvider } from "kbar";
import {
  KBAR_LISTBOX,
  KBarAnimator as KBarAnimator2,
  KBarContext as KBarContext2,
  KBarPortal as KBarPortal2,
  KBarPositioner as KBarPositioner2,
  KBarProvider as KBarProvider2,
  KBarResults as KBarResults2,
  KBarSearch as KBarSearch2,
  NO_GROUP,
  Priority,
  VisualState as VisualState2,
  createAction as createAction2,
  getListboxItemId,
  useKBar,
  useMatches as useMatches2,
  useRegisterActions as useRegisterActions2
} from "kbar";

// src/hooks/useRefineKbar/index.tsx
import { useEffect, useState, useContext } from "react";
import {
  useDelete,
  useTranslate,
  useCanWithoutCache,
  useUserFriendlyName,
  useGo,
  useResourceParams
} from "@refinedev/core";
import {
  useRegisterActions,
  createAction,
  KBarContext,
  VisualState
} from "kbar";

// src/definitions/capitalize/index.ts
var capitalize = /* @__PURE__ */ __name((str) => str.replace(
  /\w\S*/g,
  (txt) => txt.charAt(0).toUpperCase() + txt.slice(1).toLowerCase()
), "capitalize");

// src/hooks/useRefineKbar/index.tsx
import { useGetToPath } from "@refinedev/core";
var useRefineKbar = /* @__PURE__ */ __name(() => {
  const t = useTranslate();
  const {
    id: idFromParams,
    action: actionFromParams,
    resource: resourceFromParams,
    resources
  } = useResourceParams();
  const getToPath = useGetToPath();
  const go = useGo();
  const {
    mutation: { mutate }
  } = useDelete();
  const getUserFriendlyName = useUserFriendlyName();
  const kbarContext = useContext(KBarContext);
  const { can } = useCanWithoutCache();
  const [actions, setActions] = useState([]);
  useEffect(() => {
    const preaparedActions = /* @__PURE__ */ __name(async () => {
      return await Promise.all(
        moveActionToFirst().flatMap((resource) => {
          return createActionWithResource(resource);
        })
      );
    }, "preaparedActions");
    preaparedActions().then((actions2) => {
      return setActions(actions2.flatMap((action) => action));
    });
  }, [resources, idFromParams, resourceFromParams, actionFromParams]);
  useEffect(() => {
    if (actions.length === 0) {
      kbarContext.query.setVisualState(VisualState.hidden);
    }
  }, [actions]);
  const moveActionToFirst = /* @__PURE__ */ __name(() => {
    const orderedResources = [...resources];
    const fromIndex = orderedResources == null ? void 0 : orderedResources.findIndex(
      (resource) => (resource.identifier ?? (resource == null ? void 0 : resource.name)) === ((resourceFromParams == null ? void 0 : resourceFromParams.identifier) ?? (resourceFromParams == null ? void 0 : resourceFromParams.name))
    );
    if (fromIndex > 0) {
      const element = orderedResources[fromIndex];
      orderedResources.splice(fromIndex, 1);
      orderedResources.splice(0, 0, element);
    }
    return orderedResources;
  }, "moveActionToFirst");
  const createActionWithResource = /* @__PURE__ */ __name(async (resource) => {
    var _a, _b, _c, _d, _e, _f;
    const { name, list, create, show, edit } = resource;
    const label = (_a = resource == null ? void 0 : resource.meta) == null ? void 0 : _a.label;
    const icon = (_b = resource == null ? void 0 : resource.meta) == null ? void 0 : _b.icon;
    const canDelete = (_c = resource == null ? void 0 : resource.meta) == null ? void 0 : _c.canDelete;
    const section = label ?? t(
      `${resource.name}.${resource.name}`,
      getUserFriendlyName(resource.name, "plural")
    );
    const tempActions = [];
    if (list && (resourceFromParams !== void 0 && (resourceFromParams == null ? void 0 : resourceFromParams.name) !== name || actionFromParams !== void 0 && (resourceFromParams == null ? void 0 : resourceFromParams.name) === name)) {
      const { can: canList } = await (can == null ? void 0 : can({
        resource: name,
        action: "list" /* List */,
        params: { id: idFromParams, resource }
      })) || { can: true };
      if (canList) {
        tempActions.push(
          createAction({
            name: t("actions.list", capitalize("list" /* List */)),
            section,
            icon,
            perform: () => {
              const p = getToPath({
                resource,
                action: "list"
              });
              if (p) {
                go({ to: p });
              }
            }
          })
        );
      }
    }
    if ((((_d = resource == null ? void 0 : resource.meta) == null ? void 0 : _d.canCreate) || !!create) && create && ("create" /* Create */ !== actionFromParams || (resourceFromParams == null ? void 0 : resourceFromParams.name) !== name)) {
      const { can: canAccessCreate } = await (can == null ? void 0 : can({
        resource: name,
        action: "create" /* Create */,
        params: { resource }
      })) || { can: true };
      if (canAccessCreate) {
        tempActions.push(
          createAction({
            name: t("actions.create", capitalize("create" /* Create */)),
            section,
            icon,
            keywords: "new",
            perform: () => {
              const p = getToPath({
                resource,
                action: "create"
              });
              if (p) {
                go({ to: p });
              }
            }
          })
        );
      }
    }
    if ((resourceFromParams == null ? void 0 : resourceFromParams.name) === name && idFromParams) {
      if ((((_e = resource == null ? void 0 : resource.meta) == null ? void 0 : _e.canShow) || !!show) && show && "show" /* Show */ !== actionFromParams) {
        const { can: canAccessShow } = await (can == null ? void 0 : can({
          resource: name,
          action: "show" /* Show */,
          params: { id: idFromParams, resource }
        })) || { can: true };
        if (canAccessShow) {
          tempActions.push(
            createAction({
              name: t("actions.show", capitalize("show" /* Show */)),
              section,
              icon,
              perform: () => {
                const p = getToPath({
                  resource,
                  action: "show",
                  meta: {
                    id: idFromParams
                  }
                });
                if (p) {
                  go({ to: p });
                }
              }
            })
          );
        }
      }
      if ((((_f = resource == null ? void 0 : resource.meta) == null ? void 0 : _f.canEdit) || !!edit) && edit && "edit" /* Edit */ !== actionFromParams) {
        const { can: canAccessEdit } = await (can == null ? void 0 : can({
          resource: name,
          action: "edit" /* Edit */,
          params: { id: idFromParams, resource }
        })) || { can: true };
        if (canAccessEdit) {
          tempActions.push(
            createAction({
              name: t("actions.edit", capitalize("edit" /* Edit */)),
              section,
              icon,
              perform: () => {
                const p = getToPath({
                  resource,
                  action: "edit",
                  meta: {
                    id: idFromParams
                  }
                });
                if (p) {
                  go({ to: p });
                }
              }
            })
          );
        }
      }
      if (canDelete) {
        const { can: canAccessDelete } = await (can == null ? void 0 : can({
          resource: name,
          action: "delete" /* Delete */,
          params: { id: idFromParams, resource }
        })) || { can: true };
        if (canAccessDelete) {
          tempActions.push(
            {
              id: "delete",
              name: t(
                "actions.delete",
                capitalize("delete" /* Delete */)
              ),
              section,
              icon
            },
            createAction({
              name: t(
                "buttons.delete",
                capitalize("delete" /* Delete */)
              ),
              section: t("buttons.confirm", "Are you sure?"),
              parent: "delete",
              perform: () => {
                mutate(
                  {
                    resource: resource.name,
                    id: idFromParams
                  },
                  {
                    onSuccess: () => {
                      const p = getToPath({
                        resource,
                        action: "list"
                      });
                      if (p) {
                        go({ to: p });
                      }
                    }
                  }
                );
              }
            }),
            createAction({
              name: t("buttons.cancel", "Cancel"),
              parent: "delete",
              perform: () => null
            })
          );
        }
      }
    }
    return tempActions;
  }, "createActionWithResource");
  useRegisterActions(actions, [actions]);
}, "useRefineKbar");

// src/components/commandBar/index.tsx
import React from "react";
import { KBarPortal, KBarPositioner, KBarAnimator, KBarSearch } from "kbar";
var CommandBar = /* @__PURE__ */ __name(() => {
  const searchStyle = {
    padding: "12px 16px",
    fontSize: "16px",
    width: "100%",
    boxSizing: "border-box",
    outline: "none",
    border: "none",
    background: "rgb(252 252 252)",
    color: "black"
  };
  const animatorStyle = {
    maxWidth: "600px",
    width: "100%",
    background: "white",
    color: "black",
    borderRadius: "8px",
    overflow: "hidden",
    boxShadow: "0px 4px 4px rgba(0, 0, 0, 0.25)"
  };
  const positionerStyle = {
    opacity: 1,
    transition: "background 0.35s cubic-bezier(0.4, 0, 0.2, 1) 0s",
    backdropFilter: "saturate(180%) blur(1px)",
    background: "rgba(0, 0, 0, 0.1)",
    zIndex: "9999"
  };
  return /* @__PURE__ */ React.createElement(KBarPortal, null, /* @__PURE__ */ React.createElement(KBarPositioner, { style: positionerStyle }, /* @__PURE__ */ React.createElement(KBarAnimator, { style: animatorStyle }, /* @__PURE__ */ React.createElement(KBarSearch, { style: searchStyle }), /* @__PURE__ */ React.createElement(RenderResults, null))));
}, "CommandBar");

// src/components/renderResults/index.tsx
import React2 from "react";
import { KBarResults, useMatches } from "kbar";
var groupNameStyle = {
  padding: "8px 16px",
  fontSize: "14px",
  textTransform: "uppercase",
  fontWeight: "bold",
  opacity: 0.5
};
var RenderResults = /* @__PURE__ */ __name(() => {
  const { results, rootActionId } = useMatches();
  return /* @__PURE__ */ React2.createElement(
    KBarResults,
    {
      items: results,
      onRender: ({ item, active }) => {
        return typeof item === "string" ? /* @__PURE__ */ React2.createElement("div", { style: groupNameStyle }, item) : /* @__PURE__ */ React2.createElement(
          ResultItem,
          {
            action: item,
            active,
            currentRootActionId: rootActionId
          }
        );
      }
    }
  );
}, "RenderResults");

// src/components/resultItem/index.tsx
import React3 from "react";
var ResultItem = React3.forwardRef(
  ({ action, active, currentRootActionId }, ref) => {
    var _a;
    const ancestors = React3.useMemo(() => {
      if (!currentRootActionId)
        return action.ancestors;
      const index = action.ancestors.findIndex(
        (ancestor) => ancestor.id === currentRootActionId
      );
      return action.ancestors.slice(index + 1);
    }, [action.ancestors, currentRootActionId]);
    return /* @__PURE__ */ React3.createElement(
      "div",
      {
        ref,
        style: {
          padding: "12px 16px",
          background: active ? "rgba(0 0 0 / 0.05)" : "transparent",
          borderLeft: `2px solid ${active ? "rgb(28 28 29)" : "transparent"}`,
          display: "flex",
          alignItems: "center",
          justifyContent: "space-between",
          cursor: "pointer"
        }
      },
      /* @__PURE__ */ React3.createElement(
        "div",
        {
          style: {
            display: "flex",
            gap: "8px",
            alignItems: "center",
            fontSize: 14
          }
        },
        action.icon && action.icon,
        /* @__PURE__ */ React3.createElement("div", { style: { display: "flex", flexDirection: "column" } }, /* @__PURE__ */ React3.createElement("div", null, ancestors.length > 0 && ancestors.map((ancestor) => /* @__PURE__ */ React3.createElement(React3.Fragment, { key: ancestor.id }, /* @__PURE__ */ React3.createElement(
          "span",
          {
            style: {
              opacity: 0.5,
              marginRight: 8
            }
          },
          ancestor.name
        ), /* @__PURE__ */ React3.createElement(
          "span",
          {
            style: {
              marginRight: 8
            }
          },
          "\u203A"
        ))), /* @__PURE__ */ React3.createElement(
          "span",
          {
            style: {
              color: action.name.toLocaleUpperCase() === "DELETE" ? "red" : "black"
            }
          },
          action.name
        )), action.subtitle && /* @__PURE__ */ React3.createElement("span", { style: { fontSize: 12 } }, action.subtitle))
      ),
      ((_a = action.shortcut) == null ? void 0 : _a.length) ? /* @__PURE__ */ React3.createElement(
        "div",
        {
          "aria-hidden": true,
          style: {
            display: "grid",
            gridAutoFlow: "column",
            gap: "4px"
          }
        },
        action.shortcut.map((sc) => /* @__PURE__ */ React3.createElement(
          "kbd",
          {
            key: sc,
            style: {
              padding: "4px 6px",
              background: "rgba(0 0 0 / .1)",
              borderRadius: "4px",
              fontSize: 14
            }
          },
          sc
        ))
      ) : null
    );
  }
);
ResultItem.displayName = "ResultItem";

// src/components/refineKbar/index.tsx
import React4, { useContext as useContext2 } from "react";
var RefineKbar = /* @__PURE__ */ __name(({ commandBarProps }) => {
  const context = useContext2(RefineKbarPropsContext);
  useRefineKbar();
  const props = { ...context, ...commandBarProps };
  return /* @__PURE__ */ React4.createElement(CommandBar, { ...props });
}, "RefineKbar");

// src/index.tsx
var RefineKbarPropsContext = createContext({});
var RefineKbarProvider = /* @__PURE__ */ __name(({ children, commandBarProps }) => {
  return /* @__PURE__ */ React5.createElement(RefineKbarPropsContext.Provider, { value: commandBarProps ?? {} }, /* @__PURE__ */ React5.createElement(KBarProvider, null, children));
}, "RefineKbarProvider");
export {
  KBAR_LISTBOX,
  KBarAnimator2 as KBarAnimator,
  KBarContext2 as KBarContext,
  KBarPortal2 as KBarPortal,
  KBarPositioner2 as KBarPositioner,
  KBarProvider2 as KBarProvider,
  KBarResults2 as KBarResults,
  KBarSearch2 as KBarSearch,
  NO_GROUP,
  Priority,
  RefineKbar,
  RefineKbarPropsContext,
  RefineKbarProvider,
  VisualState2 as VisualState,
  createAction2 as createAction,
  getListboxItemId,
  useKBar,
  useMatches2 as useMatches,
  useRefineKbar,
  useRegisterActions2 as useRegisterActions
};
