// @refinedev/ui-types/dist/index.cjs
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// src/index.tsx
var src_exports = {};
__export(src_exports, {
  RefineButtonClassNames: () => RefineButtonClassNames,
  RefineButtonTestIds: () => RefineButtonTestIds,
  RefinePageHeaderClassNames: () => RefinePageHeaderClassNames
});
module.exports = __toCommonJS(src_exports);

// src/ids.tsx
var RefineButtonTestIds = {
  CloneButton: "refine-clone-button",
  DeleteButton: "refine-delete-button",
  EditButton: "refine-edit-button",
  SaveButton: "refine-save-button",
  CreateButton: "refine-create-button",
  ImportButton: "refine-import-button",
  ExportButton: "refine-export-button",
  ListButton: "refine-list-button",
  RefreshButton: "refine-refresh-button",
  ShowButton: "refine-show-button"
};

// src/classNames.ts
var RefinePageHeaderClassNames = {
  Title: "refine-pageHeader-title",
  SubTitle: "refine-pageHeader-subTitle"
};
var RefineButtonClassNames = {
  CloneButton: "refine-clone-button",
  DeleteButton: "refine-delete-button",
  EditButton: "refine-edit-button",
  SaveButton: "refine-save-button",
  CreateButton: "refine-create-button",
  ImportButton: "refine-import-button",
  ExportButton: "refine-export-button",
  ListButton: "refine-list-button",
  RefreshButton: "refine-refresh-button",
  ShowButton: "refine-show-button"
};
